<!doctype html><html><head><title>Annex I - Code</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Annex I - Code"><meta property="og:description" content="This jupyter notebook shall serve as accompanying material to the paper titled &ldquo;Using Computational Methods to Discover Novel Drugs for the Treatment of Androgenetic Alopecia&rdquo;, a report for the &ldquo;Computational Structural Biology for Lead Discovery&rdquo; subject at UPM&rsquo;s Master in Computational Biology
Imports First, we proceed to import the modules we will use:
import json # Lets us work with the json format import requests # Allows Python to make web requests import pandas as pd # Analysis of tabular data import numpy as np # Numerical library import matplotlib."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-01-10T00:00:00+00:00"><meta property="article:modified_time" content="2022-01-10T00:00:00+00:00"><meta name=description content="Annex I - Code"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=B칰squeda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinform치tica</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biolog칤a Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biolog칤a Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estad칤stica</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul class=active><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentaci칩n>Presentaci칩n</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelizaci칩n</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erd칬s-R칠nyi>Erd칬s-R칠nyi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poes칤a</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atr칠vete a Cruzar">Atr칠vete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="A침o Nuevo">A침o Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisi칩n Inminente">Colisi칩n Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmali칩n">Efecto Pigmali칩n</a></li><li><a href=/es/posts/poesia/efimero/ title=Ef칤mero>Ef칤mero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 游댠 游댠">Fuego 游댠 游댠</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocres칤a>Hipocres칤a</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducci칩n>Introducci칩n</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revoluci칩n">La Revoluci칩n</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Monta침a Rusa">Monta침a Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojal치>Ojal치</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 游꿭 游꿬">Orquesta 游꿭 游꿬</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plut칩n>Plut칩n</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=P치jaro>P치jaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulaci칩n>Simulaci칩n</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sue침o>Sue침o</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentaci칩n>Tentaci칩n</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Ego칤smo>Ego칤smo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Ignacio Marcos L칩pez</h5><p>lunes, 10 de enero de 2022</p></div><div class=title><h1>Annex I - Code</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><p>This jupyter notebook shall serve as accompanying material to the paper titled &ldquo;Using Computational Methods to Discover Novel Drugs for the Treatment of Androgenetic Alopecia&rdquo;, a report for the &ldquo;Computational Structural Biology for Lead Discovery&rdquo; subject at UPM&rsquo;s Master in Computational Biology</p><h2 id=imports>Imports</h2><p>First, we proceed to import the modules we will use:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> json                        <span style=color:#75715e># Lets us work with the json format</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> requests                    <span style=color:#75715e># Allows Python to make web requests</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> pandas <span style=color:#66d9ef>as</span> pd                <span style=color:#75715e># Analysis of tabular data</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np                 <span style=color:#75715e># Numerical library</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> matplotlib.pyplot <span style=color:#66d9ef>as</span> plt    <span style=color:#75715e># Static, animated, and interactive visualizations</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> rdkit                       <span style=color:#75715e># Cheminformatics and ML package</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> rdkit.Chem <span style=color:#f92672>import</span> MACCSkeys   <span style=color:#75715e># MACCS fingerprint calculation</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> rdkit.Chem <span style=color:#f92672>import</span> PandasTools <span style=color:#75715e># RDkit interaction with pandas</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> rdkit.Chem <span style=color:#f92672>import</span> rdFingerprintGenerator <span style=color:#75715e>#Generate fingerprints</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> rdkit <span style=color:#f92672>import</span> Chem             <span style=color:#75715e># The chemistry library</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> rdkit.Chem <span style=color:#f92672>import</span> Descriptors, Draw <span style=color:#75715e># Molecule descriptors for the Ro5</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> rdkit.Chem.AllChem <span style=color:#66d9ef>as</span> AllChem <span style=color:#75715e># Import all RDKit chemistry modules</span>
</span></span></code></pre></div><h2 id=initial-analysis>Initial Analysis</h2><p>First, we query the compounds targeting <strong>CHEMBL1856</strong> (3-oxo-5-alpha-steroid 4-dehydrogenase 2, a.k.a. 5-$\alpha$-reductase type II</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Ask for at least 150 compounds</span>
</span></span><span style=display:flex><span>activity_url <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;https://www.ebi.ac.uk/chembl/api/data/activity?target_chembl_id_exact=CHEMBL1856&amp;offset=150&amp;limit=150&#34;</span>
</span></span><span style=display:flex><span><span style=color:#75715e># Get results as JSON</span>
</span></span><span style=display:flex><span>activity_request <span style=color:#f92672>=</span> requests<span style=color:#f92672>.</span>get(activity_url, headers<span style=color:#f92672>=</span>{<span style=color:#e6db74>&#34;Accept&#34;</span>:<span style=color:#e6db74>&#34;application/json&#34;</span>})<span style=color:#f92672>.</span>json()
</span></span><span style=display:flex><span><span style=color:#75715e># Display as table</span>
</span></span><span style=display:flex><span>activity_table <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>DataFrame<span style=color:#f92672>.</span>from_dict(activity_request[<span style=color:#e6db74>&#39;activities&#39;</span>])[[<span style=color:#e6db74>&#39;molecule_chembl_id&#39;</span>,
</span></span><span style=display:flex><span>                                                                         <span style=color:#e6db74>&#39;type&#39;</span>, <span style=color:#e6db74>&#39;standard_value&#39;</span>, <span style=color:#e6db74>&#39;standard_units&#39;</span>]]
</span></span></code></pre></div><p>We then process the data by Ki and standarize the values</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Select only those that have Ki</span>
</span></span><span style=display:flex><span>activity_table_filter <span style=color:#f92672>=</span> activity_table<span style=color:#f92672>.</span>loc[activity_table[<span style=color:#e6db74>&#39;type&#39;</span>]<span style=color:#f92672>==</span><span style=color:#e6db74>&#34;Ki&#34;</span>]<span style=color:#f92672>.</span>copy()<span style=color:#f92672>.</span>dropna()
</span></span><span style=display:flex><span><span style=color:#75715e># Transform the standard_value column to float to be able to work with it</span>
</span></span><span style=display:flex><span>activity_table_filter[<span style=color:#e6db74>&#39;standard_value&#39;</span>] <span style=color:#f92672>=</span> activity_table_filter[<span style=color:#e6db74>&#39;standard_value&#39;</span>]<span style=color:#f92672>.</span>astype(<span style=color:#e6db74>&#34;float&#34;</span>)
</span></span><span style=display:flex><span><span style=color:#75715e># Order the table by value and display first 5 values</span>
</span></span><span style=display:flex><span>activity_table_filter<span style=color:#f92672>.</span>sort_values([<span style=color:#e6db74>&#39;standard_value&#39;</span>])<span style=color:#f92672>.</span>head(<span style=color:#ae81ff>5</span>)
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>molecule_chembl_id</th><th>type</th><th>standard_value</th><th>standard_units</th></tr></thead><tbody><tr><th>143</th><td>CHEMBL336532</td><td>Ki</td><td>7.6</td><td>nM</td></tr><tr><th>21</th><td>CHEMBL3350133</td><td>Ki</td><td>44.0</td><td>nM</td></tr><tr><th>140</th><td>CHEMBL340006</td><td>Ki</td><td>46.0</td><td>nM</td></tr><tr><th>25</th><td>CHEMBL3350133</td><td>Ki</td><td>110.0</td><td>nM</td></tr><tr><th>141</th><td>CHEMBL340006</td><td>Ki</td><td>265.0</td><td>nM</td></tr></tbody></table></div><p>We find the best compound to be CHEMBL296415. After <a href=https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL296415/>checking CHEMBL</a>, we find no information on selectivity; since we want it to be selective, we search for approved, selective drugs:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Specify approved drugs that target CHEMBL1856</span>
</span></span><span style=display:flex><span>mechanism_url <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;https://www.ebi.ac.uk/chembl/api/data/mechanism?target_chembl_id__exact=CHEMBL1856&#34;</span>
</span></span><span style=display:flex><span><span style=color:#75715e># Get data from the URL</span>
</span></span><span style=display:flex><span>mechanism_components <span style=color:#f92672>=</span> requests<span style=color:#f92672>.</span>get(mechanism_url, headers<span style=color:#f92672>=</span>{<span style=color:#e6db74>&#34;Accept&#34;</span>:<span style=color:#e6db74>&#34;application/json&#34;</span>})<span style=color:#f92672>.</span>json()
</span></span><span style=display:flex><span><span style=color:#75715e># Format dataframe</span>
</span></span><span style=display:flex><span>mechanism_table <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>DataFrame<span style=color:#f92672>.</span>from_dict(mechanism_components[<span style=color:#e6db74>&#39;mechanisms&#39;</span>])[[<span style=color:#e6db74>&#39;molecule_chembl_id&#39;</span>, <span style=color:#e6db74>&#39;max_phase&#39;</span>]]
</span></span><span style=display:flex><span>mechanism_table <span style=color:#75715e>#And print</span>
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>molecule_chembl_id</th><th>max_phase</th></tr></thead><tbody><tr><th>0</th><td>CHEMBL710</td><td>4</td></tr></tbody></table></div><p>We find there is only one approved compound, Finasteride (<strong>CHEMBL710</strong>). Our focus will thus be finding drugs similar to Finasteride, but more effective, more selective, and with less side effects.</p><h2 id=similarity-analysis>Similarity Analysis</h2><p>We upload FNS&rsquo;s SMILES TO <a href=http://www.swisssimilarity.ch/>SwissSimilarity</a>, select ChEMBL (activity&lt;10췃M) as subset and combined as method, run the analysis and save the result as &ldquo;ScreeningResultsFinasteride.csv&rdquo;.</p><h2 id=ligand-based-virtual-screening>Ligand Based Virtual Screening</h2><p>We subset this data to get only the results most similar to FNS, using a series of metrics (DICE/Tanimoto, MACCS/Morgan) explained in the paper.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Define Query molecule</span>
</span></span><span style=display:flex><span>smiles <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;CC(C)(C)NC(=O)[C@H]1CC[C@H]2[C@@H]3CC[C@H]4NC(=O)C=C[C@]4(C)[C@H]3CC[C@]12C&#34;</span>
</span></span><span style=display:flex><span>Query <span style=color:#f92672>=</span> rdkit<span style=color:#f92672>.</span>Chem<span style=color:#f92672>.</span>MolFromSmiles(smiles)
</span></span><span style=display:flex><span><span style=color:#75715e># And depict it</span>
</span></span><span style=display:flex><span>rdkit<span style=color:#f92672>.</span>Chem<span style=color:#f92672>.</span>Draw<span style=color:#f92672>.</span>MolToImage(Query, includeAtomNumbers<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span></code></pre></div><p><img src=output_files/output_16_0.png alt=png></p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>raw_database <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#39;./ScreeningResultsFinasteride.csv&#39;</span>, delimiter<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;;&#39;</span>,
</span></span><span style=display:flex><span>                           names <span style=color:#f92672>=</span>(<span style=color:#e6db74>&#39;ChemblID&#39;</span>,<span style=color:#e6db74>&#39;Score&#39;</span>,<span style=color:#e6db74>&#39;Smile&#39;</span>))       <span style=color:#75715e># Read the file and add titles</span>
</span></span><span style=display:flex><span>print(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;The initial compounds database has </span><span style=color:#e6db74>{</span>len(raw_database)<span style=color:#e6db74>}</span><span style=color:#e6db74> molecules&#34;</span>)
</span></span><span style=display:flex><span>PandasTools<span style=color:#f92672>.</span>AddMoleculeColumnToFrame(raw_database, smilesCol<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;Smile&#39;</span>) <span style=color:#75715e># Generate molecule (RoMol) from SMILES</span>
</span></span></code></pre></div><pre><code>The initial compounds database has 400 molecules
</code></pre><p>Now, we will build databases for our fingerprints:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># We build databases for the two methods</span>
</span></span><span style=display:flex><span>MACCSDatabase <span style=color:#f92672>=</span> raw_database<span style=color:#f92672>.</span>ROMol<span style=color:#f92672>.</span>apply(MACCSkeys<span style=color:#f92672>.</span>GenMACCSKeys)
</span></span><span style=display:flex><span>MorganDatabase <span style=color:#f92672>=</span> rdFingerprintGenerator<span style=color:#f92672>.</span>GetFPs(raw_database[<span style=color:#e6db74>&#34;ROMol&#34;</span>]<span style=color:#f92672>.</span>tolist())
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># And prepare the queries:</span>
</span></span><span style=display:flex><span>MorganQuery <span style=color:#f92672>=</span> rdFingerprintGenerator<span style=color:#f92672>.</span>GetFPs([Query])[<span style=color:#ae81ff>0</span>]
</span></span><span style=display:flex><span>MACCQuery <span style=color:#f92672>=</span> MACCSkeys<span style=color:#f92672>.</span>GenMACCSKeys(Query)
</span></span></code></pre></div><p>And we calculate the indices based on those fingerprints:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># We calculate this indices both for the MACCS fingerprints</span>
</span></span><span style=display:flex><span>raw_database[<span style=color:#e6db74>&#34;Tanimoto (MACCS)&#34;</span>] <span style=color:#f92672>=</span> rdkit<span style=color:#f92672>.</span>DataStructs<span style=color:#f92672>.</span>BulkTanimotoSimilarity(MACCQuery, MACCSDatabase)
</span></span><span style=display:flex><span>raw_database[<span style=color:#e6db74>&#34;Dice (MACCS)&#34;</span>] <span style=color:#f92672>=</span> rdkit<span style=color:#f92672>.</span>DataStructs<span style=color:#f92672>.</span>BulkDiceSimilarity(MACCQuery, MACCSDatabase)
</span></span><span style=display:flex><span><span style=color:#75715e># And for the morgan fingerprint too</span>
</span></span><span style=display:flex><span>raw_database[<span style=color:#e6db74>&#34;Tanimoto (Morgan)&#34;</span>] <span style=color:#f92672>=</span> rdkit<span style=color:#f92672>.</span>DataStructs<span style=color:#f92672>.</span>BulkTanimotoSimilarity(MorganQuery, MorganDatabase)
</span></span><span style=display:flex><span>raw_database[<span style=color:#e6db74>&#34;Dice (Morgan)&#34;</span>] <span style=color:#f92672>=</span> rdkit<span style=color:#f92672>.</span>DataStructs<span style=color:#f92672>.</span>BulkDiceSimilarity(MorganQuery, MorganDatabase)
</span></span></code></pre></div><p>We can describe the new database:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>raw_database<span style=color:#f92672>.</span>describe()
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>Score</th><th>Tanimoto (MACCS)</th><th>Dice (MACCS)</th><th>Tanimoto (Morgan)</th><th>Dice (Morgan)</th></tr></thead><tbody><tr><th>count</th><td>400.000000</td><td>400.000000</td><td>400.000000</td><td>400.000000</td><td>400.000000</td></tr><tr><th>mean</th><td>0.055978</td><td>0.482113</td><td>0.640505</td><td>0.157272</td><td>0.258770</td></tr><tr><th>std</th><td>0.099914</td><td>0.130162</td><td>0.115564</td><td>0.108950</td><td>0.140409</td></tr><tr><th>min</th><td>0.025000</td><td>0.205128</td><td>0.340426</td><td>0.035294</td><td>0.068182</td></tr><tr><th>25%</th><td>0.027750</td><td>0.388682</td><td>0.559785</td><td>0.093458</td><td>0.170940</td></tr><tr><th>50%</th><td>0.030000</td><td>0.450806</td><td>0.621456</td><td>0.113636</td><td>0.204082</td></tr><tr><th>75%</th><td>0.038000</td><td>0.600000</td><td>0.750000</td><td>0.172580</td><td>0.294359</td></tr><tr><th>max</th><td>0.998000</td><td>1.000000</td><td>1.000000</td><td>1.000000</td><td>1.000000</td></tr></tbody></table></div><p>As we can see, the simmilarity scores are pretty small, but this is to be expected when selecting &ldquo;Combined - ChEMBL (activity&lt;10췃M)&rdquo; in ENSEMBL, as it takes two methods (electroshape and fingerprint) which are quite different. For our Tanimoto and Dice metrics, we can see that all metrics including MACCS yield way better values than those used with the Morgan fingerprint, and those with Dice show better results than those with Tanimoto. Thus, we use &ldquo;dice_maccs&rdquo; as our parameter, and set the cutoff to 0.621456 (with a length of ~200 elements and representing Quartile 2) and save the data.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>raw_databasefilter <span style=color:#f92672>=</span> raw_database<span style=color:#f92672>.</span>loc[raw_database[<span style=color:#e6db74>&#39;Dice (MACCS)&#39;</span>]<span style=color:#f92672>.</span>astype(<span style=color:#e6db74>&#39;float&#39;</span>) <span style=color:#f92672>&gt;</span> <span style=color:#ae81ff>0.621456</span>]
</span></span><span style=display:flex><span>raw_databasefilter<span style=color:#f92672>.</span>drop(<span style=color:#e6db74>&#39;ROMol&#39;</span>, axis<span style=color:#f92672>=</span><span style=color:#ae81ff>1</span>)<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>&#39;./SimilarityAnalysisFinasteride.csv&#39;</span>, index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span><span style=display:flex><span>print(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;The DICE(MACCS) filtered database has </span><span style=color:#e6db74>{</span>len(raw_databasefilter)<span style=color:#e6db74>}</span><span style=color:#e6db74> molecules&#34;</span>)
</span></span></code></pre></div><pre><code>The DICE(MACCS) filtered database has 200 molecules
</code></pre><h2 id=pharmacophore-based-virtual-screning>Pharmacophore Based Virtual Screning</h2><p>We upload &ldquo;SimilarityAnalysisFinasteride.csv&rdquo; to Pharmit, and generate the &ldquo;Finasteride-Similar 5ARIs KQMJ9Y&rdquo; database. Since PharmIt requires the db format to be &ldquo;.smi&rdquo;, we make the necessary changes:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>pharmit_input <span style=color:#f92672>=</span> raw_databasefilter[<span style=color:#e6db74>&#39;Smile&#39;</span>] <span style=color:#f92672>+</span> <span style=color:#e6db74>&#39; &#39;</span> <span style=color:#f92672>+</span> raw_databasefilter[<span style=color:#e6db74>&#39;ChemblID&#39;</span>]
</span></span><span style=display:flex><span>pharmit_input<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>&#39;./pharmit_input.smi&#39;</span>, index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>, header<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><p>We use our created pharmacophore (see paper) to filter the database, and download the results as &ldquo;pharmit_output.sdf&rdquo;</p><h2 id=admet-properties-analysis>ADMET Properties analysis</h2><p>As a proxy for ADMET properties, we are using Lipinski&rsquo;s rule of five, adding TPSA as an additional method.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>calculate_ro5_properties</span>(smiles): <span style=color:#75715e># Define the funtion to validate Ro5</span>
</span></span><span style=display:flex><span>    molecule <span style=color:#f92672>=</span> Chem<span style=color:#f92672>.</span>MolFromSmiles(smiles)
</span></span><span style=display:flex><span>    molecular_weight <span style=color:#f92672>=</span> Descriptors<span style=color:#f92672>.</span>ExactMolWt(molecule)
</span></span><span style=display:flex><span>    n_hba <span style=color:#f92672>=</span> Descriptors<span style=color:#f92672>.</span>NumHAcceptors(molecule)
</span></span><span style=display:flex><span>    n_hbd <span style=color:#f92672>=</span> Descriptors<span style=color:#f92672>.</span>NumHDonors(molecule)
</span></span><span style=display:flex><span>    logp <span style=color:#f92672>=</span> Descriptors<span style=color:#f92672>.</span>MolLogP(molecule)
</span></span><span style=display:flex><span>    TPSA <span style=color:#f92672>=</span> Descriptors<span style=color:#f92672>.</span>TPSA (molecule)
</span></span><span style=display:flex><span>    conditions <span style=color:#f92672>=</span> [molecular_weight <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>500</span>, n_hba <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>10</span>, n_hbd <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>5</span>, logp <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>5</span>, TPSA <span style=color:#f92672>&lt;</span> <span style=color:#ae81ff>140</span>]
</span></span><span style=display:flex><span>    ro5_fulfilled <span style=color:#f92672>=</span> sum(conditions) <span style=color:#f92672>==</span> <span style=color:#ae81ff>5</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> pd<span style=color:#f92672>.</span>Series(
</span></span><span style=display:flex><span>        [molecular_weight, n_hba, n_hbd, logp, TPSA, ro5_fulfilled],
</span></span><span style=display:flex><span>        index<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#34;molecular_weight&#34;</span>, <span style=color:#e6db74>&#34;n_hba&#34;</span>, <span style=color:#e6db74>&#34;n_hbd&#34;</span>, <span style=color:#e6db74>&#34;logp&#34;</span>, <span style=color:#e6db74>&#34;TPSA&#34;</span>, <span style=color:#e6db74>&#34;ro5_fulfilled&#34;</span>],
</span></span><span style=display:flex><span>    )
</span></span></code></pre></div><p>Read the database we will process:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>MoleculeDatabase <span style=color:#f92672>=</span> PandasTools<span style=color:#f92672>.</span>LoadSDF(<span style=color:#e6db74>&#39;./pharmit_output.sdf&#39;</span>, embedProps<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>, molColName<span style=color:#f92672>=</span><span style=color:#66d9ef>None</span>, smilesName<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;smiles&#39;</span>)
</span></span><span style=display:flex><span>print(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;The pharmit database has </span><span style=color:#e6db74>{</span>len(MoleculeDatabase)<span style=color:#e6db74>}</span><span style=color:#e6db74> molecules&#34;</span>)
</span></span></code></pre></div><pre><code>The pharmit database has 51 molecules
</code></pre><p>And apply the Rule of Five</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Calculate RO5 properties for all molecules</span>
</span></span><span style=display:flex><span>ro5_properties <span style=color:#f92672>=</span> MoleculeDatabase[<span style=color:#e6db74>&#34;smiles&#34;</span>]<span style=color:#f92672>.</span>apply(calculate_ro5_properties)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Concat the properties dataframe with the pharmit output dataframe</span>
</span></span><span style=display:flex><span>MoleculeDatabase_Concat <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>concat([MoleculeDatabase, ro5_properties], axis<span style=color:#f92672>=</span><span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>#And separate by valid and invalid drugs (~ negates boolean values)</span>
</span></span><span style=display:flex><span>MoleculeDatabase_ro5_fulfilled <span style=color:#f92672>=</span> MoleculeDatabase_Concat[MoleculeDatabase_Concat[<span style=color:#e6db74>&#34;ro5_fulfilled&#34;</span>]]
</span></span><span style=display:flex><span>MoleculeDatabase_ro5_violated <span style=color:#f92672>=</span> MoleculeDatabase_Concat[<span style=color:#f92672>~</span>MoleculeDatabase_Concat[<span style=color:#e6db74>&#34;ro5_fulfilled&#34;</span>]]
</span></span><span style=display:flex><span>print(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;</span><span style=color:#e6db74>{</span>len(MoleculeDatabase_ro5_fulfilled)<span style=color:#e6db74>}</span><span style=color:#e6db74> Ro5-following compounds found&#34;</span>)
</span></span></code></pre></div><pre><code>30 Ro5-following compounds found
</code></pre><p>We save Ro5-fulfilling compounds for refference:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>MoleculeDatabase_ro5_fulfilled<span style=color:#f92672>.</span>to_csv(<span style=color:#e6db74>&#34;MolDB_Ro5.csv&#34;</span>, index<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><h2 id=docking-using-dockthor-autodock-vina>Docking using DockThor (Autodock Vina)</h2><p>For autodock vina, we want to screen the most similar molecules to our pharmacophore to see how these bind to our original targets. We can get the 4 best hits like so:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>MoleculeDatabase_ro5_fulfilled<span style=color:#f92672>.</span>sort_values([<span style=color:#e6db74>&#39;rmsd&#39;</span>], ascending<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)<span style=color:#f92672>.</span>head(<span style=color:#ae81ff>4</span>)
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>rmsd</th><th>ID</th><th>smiles</th><th>molecular_weight</th><th>n_hba</th><th>n_hbd</th><th>logp</th><th>TPSA</th><th>ro5_fulfilled</th></tr></thead><tbody><tr><th>33</th><td>0.60144037</td><td>CHEMBL2397139</td><td>CNC(=O)c1ccc(OC2CCC(NC(=O)NC34CC5CC(CC(C5)C3)C...</td><td>425.267842</td><td>3</td><td>3</td><td>4.00420</td><td>79.46</td><td>True</td></tr><tr><th>23</th><td>0.551425219</td><td>CHEMBL1668930</td><td>CC(=O)N1CCC(NC(=O)NC2CCC(C)(C)CC2)CC1</td><td>295.225977</td><td>2</td><td>2</td><td>2.26530</td><td>61.44</td><td>True</td></tr><tr><th>17</th><td>0.547150791</td><td>CHEMBL2282650</td><td>CC1=C2N(C)CC3C(CCC4(C)C(C(=O)NC(C)(C)C)CCC34)C...</td><td>400.308979</td><td>3</td><td>1</td><td>4.54840</td><td>49.41</td><td>True</td></tr><tr><th>46</th><td>0.510232747</td><td>CHEMBL1631395</td><td>CC(=O)Nc1cc2c3c(c1)n(C/C=C/Cn1cc(C)c(=O)[nH]c1...</td><td>409.175004</td><td>7</td><td>2</td><td>1.12242</td><td>110.89</td><td>True</td></tr></tbody></table></div><p>And we process them using Autodock Vina. Unfortunately, this are not valid molecules (see paper). So, we process the full database to get one in SDF format for DockThor</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>pd<span style=color:#f92672>.</span>options<span style=color:#f92672>.</span>mode<span style=color:#f92672>.</span>chained_assignment <span style=color:#f92672>=</span> <span style=color:#66d9ef>None</span> <span style=color:#75715e># Disable warnings</span>
</span></span><span style=display:flex><span><span style=color:#75715e># Subset only the eseential columns</span>
</span></span><span style=display:flex><span>MoleculeDatabase_ro5_fulfilled<span style=color:#f92672>.</span>reset_index(inplace<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>MitDatabase <span style=color:#f92672>=</span> MoleculeDatabase_ro5_fulfilled[[<span style=color:#e6db74>&#39;ID&#39;</span>,<span style=color:#e6db74>&#39;smiles&#39;</span>]]
</span></span><span style=display:flex><span><span style=color:#75715e># We add the molecule column using rdkit</span>
</span></span><span style=display:flex><span>PandasTools<span style=color:#f92672>.</span>AddMoleculeColumnToFrame(MitDatabase,<span style=color:#e6db74>&#39;smiles&#39;</span>,<span style=color:#e6db74>&#39;Molecule&#39;</span>)
</span></span><span style=display:flex><span><span style=color:#75715e># And add Hidrogens to said molecules</span>
</span></span><span style=display:flex><span>MitDatabase[<span style=color:#e6db74>&#39;MoleculeH&#39;</span>] <span style=color:#f92672>=</span> MitDatabase[<span style=color:#e6db74>&#39;Molecule&#39;</span>]<span style=color:#f92672>.</span>apply(Chem<span style=color:#f92672>.</span>AddHs)
</span></span><span style=display:flex><span><span style=color:#75715e># Compute 3D coordinates</span>
</span></span><span style=display:flex><span>MitDatabase[<span style=color:#e6db74>&#39;MoleculeH&#39;</span>]<span style=color:#f92672>.</span>map(AllChem<span style=color:#f92672>.</span>EmbedMolecule);
</span></span></code></pre></div><p>And we save it:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>PandasTools<span style=color:#f92672>.</span>WriteSDF(MitDatabase, <span style=color:#e6db74>&#39;Finasteride_Similar_Ro5.sdf&#39;</span>,
</span></span><span style=display:flex><span>                     molColName<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;MoleculeH&#39;</span>, properties<span style=color:#f92672>=</span>list(MitDatabase<span style=color:#f92672>.</span>columns))
</span></span></code></pre></div><p>We can now read the results and process them:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e># Load dataframes</span>
</span></span><span style=display:flex><span>bestranking_typeII <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#39;./bestranking_typeII.csv&#39;</span>, delimiter<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;;&#39;</span>, header<span style=color:#f92672>=</span><span style=color:#ae81ff>0</span>)
</span></span><span style=display:flex><span>bestranking_typeI  <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>read_csv(<span style=color:#e6db74>&#39;./bestranking_typeI.csv&#39;</span>,  delimiter<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;;&#39;</span>, header<span style=color:#f92672>=</span><span style=color:#ae81ff>0</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># For each database, get the compound ID and select the Name and Score columns</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> i, row <span style=color:#f92672>in</span> bestranking_typeII<span style=color:#f92672>.</span>iterrows(): bestranking_typeII[<span style=color:#e6db74>&#39;Name&#39;</span>][i] <span style=color:#f92672>=</span> bestranking_typeII[<span style=color:#e6db74>&#39;Name&#39;</span>][i]<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;_&#39;</span>)[<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>bestranking_typeII <span style=color:#f92672>=</span> bestranking_typeII[[<span style=color:#e6db74>&#39;Name&#39;</span>, <span style=color:#e6db74>&#39;Score&#39;</span>]]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># For each database, get the compound ID and select the Name and Score columns</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> i, row <span style=color:#f92672>in</span> bestranking_typeI<span style=color:#f92672>.</span>iterrows(): bestranking_typeI[<span style=color:#e6db74>&#39;Name&#39;</span>][i] <span style=color:#f92672>=</span> bestranking_typeI[<span style=color:#e6db74>&#39;Name&#39;</span>][i]<span style=color:#f92672>.</span>split(<span style=color:#e6db74>&#39;_&#39;</span>)[<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>bestranking_typeI <span style=color:#f92672>=</span> bestranking_typeI[[<span style=color:#e6db74>&#39;Name&#39;</span>, <span style=color:#e6db74>&#39;Score&#39;</span>]]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># We rename the columns</span>
</span></span><span style=display:flex><span>bestranking_typeII<span style=color:#f92672>.</span>rename(columns <span style=color:#f92672>=</span> {<span style=color:#e6db74>&#39;Score&#39;</span>:<span style=color:#e6db74>&#39;Score for type II&#39;</span>}, inplace <span style=color:#f92672>=</span> <span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>bestranking_typeI<span style=color:#f92672>.</span>rename(columns <span style=color:#f92672>=</span> {<span style=color:#e6db74>&#39;Score&#39;</span>:<span style=color:#e6db74>&#39;Score for type I&#39;</span>}, inplace <span style=color:#f92672>=</span> <span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Merge the two dataframes</span>
</span></span><span style=display:flex><span>bestranking <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>merge(bestranking_typeI, bestranking_typeII, how<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;inner&#39;</span>, on <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;Name&#39;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Add CHEMBLIDs</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> i, row <span style=color:#f92672>in</span> bestranking<span style=color:#f92672>.</span>iterrows():
</span></span><span style=display:flex><span>    bestranking[<span style=color:#e6db74>&#39;Name&#39;</span>][i] <span style=color:#f92672>=</span> MoleculeDatabase_ro5_fulfilled<span style=color:#f92672>.</span>iloc[int(row[<span style=color:#e6db74>&#34;Name&#34;</span>])][<span style=color:#e6db74>&#39;ID&#39;</span>]
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># And convert scores to Ki</span>
</span></span><span style=display:flex><span>bestranking[<span style=color:#e6db74>&#34;KI for type I&#34;</span>] <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>exp(<span style=color:#f92672>-</span>bestranking[<span style=color:#e6db74>&#34;Score for type I&#34;</span>]<span style=color:#f92672>/</span>(<span style=color:#ae81ff>0.008314</span><span style=color:#f92672>*</span>(<span style=color:#ae81ff>273</span><span style=color:#f92672>+</span><span style=color:#ae81ff>37</span>)))
</span></span><span style=display:flex><span>bestranking[<span style=color:#e6db74>&#34;KI for type II&#34;</span>] <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>exp(<span style=color:#f92672>-</span>bestranking[<span style=color:#e6db74>&#34;Score for type II&#34;</span>]<span style=color:#f92672>/</span>(<span style=color:#ae81ff>0.008314</span><span style=color:#f92672>*</span>(<span style=color:#ae81ff>273</span><span style=color:#f92672>+</span><span style=color:#ae81ff>37</span>)))
</span></span></code></pre></div><p>Now, we select only those CHEMBLIDs that are more effective and selective than finasteride:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>initial_ki <span style=color:#f92672>=</span> float(bestranking<span style=color:#f92672>.</span>loc[bestranking[<span style=color:#e6db74>&#34;Name&#34;</span>]<span style=color:#f92672>==</span><span style=color:#e6db74>&#34;CHEMBL710&#34;</span>][<span style=color:#e6db74>&#34;KI for type I&#34;</span>])
</span></span><span style=display:flex><span>bestranking<span style=color:#f92672>.</span>loc[(bestranking[<span style=color:#e6db74>&#34;KI for type I&#34;</span>] <span style=color:#f92672>&gt;=</span> initial_ki) <span style=color:#f92672>&amp;</span> (bestranking[<span style=color:#e6db74>&#34;KI for type II&#34;</span>] <span style=color:#f92672>&lt;=</span> initial_ki)]
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>Name</th><th>Score for type I</th><th>Score for type II</th><th>KI for type I</th><th>KI for type II</th></tr></thead><tbody><tr><th>0</th><td>CHEMBL2057291</td><td>-10.082</td><td>-8.657</td><td>49.988102</td><td>28.757235</td></tr><tr><th>2</th><td>CHEMBL1343539</td><td>-9.619</td><td>-8.498</td><td>41.768499</td><td>27.036773</td></tr><tr><th>3</th><td>CHEMBL2282650</td><td>-9.608</td><td>-8.746</td><td>41.590612</td><td>29.767617</td></tr><tr><th>4</th><td>CHEMBL2057296</td><td>-9.581</td><td>-8.565</td><td>41.157187</td><td>27.748830</td></tr><tr><th>5</th><td>CHEMBL76192</td><td>-9.507</td><td>-8.255</td><td>39.992294</td><td>24.604138</td></tr><tr><th>6</th><td>CHEMBL2282653</td><td>-9.351</td><td>-8.616</td><td>37.643461</td><td>28.303388</td></tr><tr><th>7</th><td>CHEMBL1762030</td><td>-9.271</td><td>-8.796</td><td>36.492965</td><td>30.350742</td></tr><tr><th>8</th><td>CHEMBL2282652</td><td>-9.252</td><td>-8.281</td><td>36.224930</td><td>24.853599</td></tr><tr><th>9</th><td>CHEMBL1668930</td><td>-9.204</td><td>-7.939</td><td>35.556526</td><td>21.765096</td></tr><tr><th>10</th><td>CHEMBL1631395</td><td>-9.186</td><td>-8.164</td><td>35.309066</td><td>23.750579</td></tr><tr><th>11</th><td>CHEMBL1825149</td><td>-9.137</td><td>-7.898</td><td>34.644117</td><td>21.421599</td></tr><tr><th>12</th><td>CHEMBL3221237</td><td>-9.109</td><td>-8.250</td><td>34.269783</td><td>24.556452</td></tr><tr><th>13</th><td>CHEMBL1800917</td><td>-9.108</td><td>-8.054</td><td>34.256489</td><td>22.758239</td></tr><tr><th>14</th><td>CHEMBL280155</td><td>-8.991</td><td>-8.473</td><td>32.736163</td><td>26.775786</td></tr><tr><th>15</th><td>CHEMBL2282654</td><td>-8.969</td><td>-8.318</td><td>32.457919</td><td>25.212967</td></tr><tr><th>16</th><td>CHEMBL2057293</td><td>-8.937</td><td>-8.796</td><td>32.057416</td><td>30.350742</td></tr><tr><th>17</th><td>CHEMBL2282779</td><td>-8.909</td><td>-8.314</td><td>31.711032</td><td>25.173867</td></tr><tr><th>18</th><td>CHEMBL1668929</td><td>-8.847</td><td>-8.067</td><td>30.957299</td><td>22.873321</td></tr><tr><th>19</th><td>CHEMBL710</td><td>-8.839</td><td>-8.467</td><td>30.861358</td><td>26.713525</td></tr></tbody></table></div><p>Since there are a lot of compounds, we can select those that are better among equals:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>bestranking<span style=color:#f92672>.</span>loc[(bestranking[<span style=color:#e6db74>&#34;KI for type I&#34;</span>] <span style=color:#f92672>&gt;=</span> (initial_ki <span style=color:#f92672>+</span> <span style=color:#ae81ff>0.15</span><span style=color:#f92672>*</span>initial_ki )) <span style=color:#f92672>&amp;</span>
</span></span><span style=display:flex><span>                 (bestranking[<span style=color:#e6db74>&#34;KI for type II&#34;</span>] <span style=color:#f92672>&lt;=</span> (initial_ki <span style=color:#f92672>-</span> <span style=color:#ae81ff>0.15</span><span style=color:#f92672>*</span>initial_ki))]
</span></span></code></pre></div><div><style scoped>.dataframe tbody tr th:only-of-type{vertical-align:middle}<pre><code>.dataframe tbody tr th{vertical-align:top}.dataframe thead th{text-align:right}</code></pre><p></style></p><table border=1 class=dataframe><thead><tr style=text-align:right><th></th><th>Name</th><th>Score for type I</th><th>Score for type II</th><th>KI for type I</th><th>KI for type II</th></tr></thead><tbody><tr><th>5</th><td>CHEMBL76192</td><td>-9.507</td><td>-8.255</td><td>39.992294</td><td>24.604138</td></tr><tr><th>8</th><td>CHEMBL2282652</td><td>-9.252</td><td>-8.281</td><td>36.224930</td><td>24.853599</td></tr><tr><th>9</th><td>CHEMBL1668930</td><td>-9.204</td><td>-7.939</td><td>35.556526</td><td>21.765096</td></tr></tbody></table></div><p>The discussion can be checked in the paper! :P</p><h2 id=code-and-acknowledgements>Code and Acknowledgements</h2><p>The header image for this post is <a href=http://creativecommons.org/licenses/by-sa/3.0/>CC-By-Sa 3.0</a> by <a href=//commons.wikimedia.org/wiki/User:Alberto_Salguero>Alberto Salguero</a> on <a href=https://commons.wikimedia.org/wiki/File:Arabidopsis_thaliana_hojas_basales.jpg#/media/Archivo:Arabidopsis_thaliana_hojas_basales.jpg>Wikimedia Commons</a></p></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f&text=Annex%20I%20-%20Code&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f&title=Annex%20I%20-%20Code" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f&name=Annex%20I%20-%20Code" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f&title=Annex%20I%20-%20Code" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f&title=Annex%20I%20-%20Code" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Annex%20I%20-%20Code&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Annex%20I%20-%20Code - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Annex%20I%20-%20Code https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Annex%20I%20-%20Code&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fannex-i-code%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Lead%20Discovery/Annex%20I%20-%20Code/index.es.md title="Mejorar esta p치gina" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta p치gina</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title="The Link Between Brain and Machine!" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>The Link Between Brain and Machine!</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Designing a pharmacophore and ensuring ADMET properties" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Designing a pharmacophore and ensuring ADMET properties</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#imports>Imports</a></li><li><a href=#initial-analysis>Initial Analysis</a></li><li><a href=#similarity-analysis>Similarity Analysis</a></li><li><a href=#ligand-based-virtual-screening>Ligand Based Virtual Screening</a></li><li><a href=#pharmacophore-based-virtual-screning>Pharmacophore Based Virtual Screning</a></li><li><a href=#admet-properties-analysis>ADMET Properties analysis</a></li><li><a href=#docking-using-dockthor-autodock-vina>Docking using DockThor (Autodock Vina)</a></li><li><a href=#code-and-acknowledgements>Code and Acknowledgements</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegaci칩n</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre m칤</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su direcci칩n de correo electr칩nico, acepta recibir el bolet칤n de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">춸 2020 Pablo Ignacio Marcos L칩pez</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Ap칩yame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>