<!doctype html><html><head><title>The Kuramoto Oscilator Model</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="The Kuramoto Oscilator Model"><meta property="og:description" content="The Kuramoto Model is a mathematical model used to describe the synchronization of a large set of coupled oscillators. This formulation, which was motivated by the behaviour of both physical and biological oscilators, has found widespread application in neuroscience.
The model makes several assumptions, including that there is weak coupling, that the oscillators are identical or nearly identical, and that interactions depend sinusoidally on the phase difference between each pair of objects."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-11-17T00:00:00+00:00"><meta property="article:modified_time" content="2021-11-17T00:00:00+00:00"><meta name=description content="The Kuramoto Oscilator Model"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=B칰squeda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinform치tica</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biolog칤a Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biolog칤a Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estad칤stica</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentaci칩n>Presentaci칩n</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelizaci칩n</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erd칬s-R칠nyi>Erd칬s-R칠nyi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poes칤a</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atr칠vete a Cruzar">Atr칠vete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="A침o Nuevo">A침o Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisi칩n Inminente">Colisi칩n Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmali칩n">Efecto Pigmali칩n</a></li><li><a href=/es/posts/poesia/efimero/ title=Ef칤mero>Ef칤mero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 游댠 游댠">Fuego 游댠 游댠</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocres칤a>Hipocres칤a</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducci칩n>Introducci칩n</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revoluci칩n">La Revoluci칩n</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Monta침a Rusa">Monta침a Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojal치>Ojal치</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 游꿭 游꿬">Orquesta 游꿭 游꿬</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plut칩n>Plut칩n</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=P치jaro>P치jaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulaci칩n>Simulaci칩n</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sue침o>Sue침o</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentaci칩n>Tentaci칩n</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Ego칤smo>Ego칤smo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Marcos</h5><p>mi칠rcoles, 17 de noviembre de 2021</p></div><div class=title><h1>The Kuramoto Oscilator Model</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><div style=text-align:center><a href="./Kuramoto Oscilators.ipynb" target=_parent><img src=/posts/Imagenes/jupyter-badge.svg align=center width=20%></a></div></br><div class="cell markdown"><p>The Kuramoto Model is a mathematical model used to describe the
synchronization of a large set of coupled oscillators. This formulation,
which was motivated by the behaviour of both physical and biological
oscilators, has found widespread application in neuroscience.</p><p>The model makes several assumptions, including that there is weak
coupling, that the oscillators are identical or nearly identical, and
that interactions depend sinusoidally on the phase difference between
each pair of objects.</p></div><div class="cell markdown"><p>For the exercises, we need to simulate some kuramoto oscilators coupled
using an external forcing: for one of the systems, this external forcing
will be the average phase of all the oscilators, while, for the other,
we will be using a uniform distribution between 0 and $2 \cdot \pi$.
Thus:</p></div><div class="cell code" data-execution_count=1><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np                   <span style=color:#75715e>#Supports common numeric operations</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> numpy <span style=color:#f92672>import</span> random             <span style=color:#75715e>#To create sets of random numbers</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> matplotlib.pyplot <span style=color:#66d9ef>as</span> plt      <span style=color:#75715e>#For graphs</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> colour <span style=color:#f92672>import</span> Color             <span style=color:#75715e>#Make color gradients in a programmatic way</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> seaborn <span style=color:#66d9ef>as</span> sns                <span style=color:#75715e>#Statistical data visualization</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> pandas <span style=color:#66d9ef>as</span> pd                  <span style=color:#75715e>#Data Visualization library</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> matplotlib <span style=color:#f92672>import</span> animation, rc <span style=color:#75715e>#Make anymations</span>
</span></span><span style=display:flex><span><span style=color:#f92672>from</span> IPython.display <span style=color:#f92672>import</span> HTML     <span style=color:#75715e>#Convert to HTML (display animations)</span>
</span></span></code></pre></div></div><div class="cell code" data-execution_count=2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e>#A function to calculate 풪(t)</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>get_new_theta</span>(theta,omega,K,dt, method):
</span></span><span style=display:flex><span>    new_theta <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>copy(theta); oscilators_diff <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>    dtheta <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>zeros(len(theta))
</span></span><span style=display:flex><span>    psi <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>arctan2(sum(np<span style=color:#f92672>.</span>sin(theta)), sum(np<span style=color:#f92672>.</span>cos(theta)))<span style=color:#f92672>%</span>(<span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi)  <span style=color:#75715e>#Average phase</span>
</span></span><span style=display:flex><span>    forcing <span style=color:#f92672>=</span> random<span style=color:#f92672>.</span>uniform(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi) <span style=color:#75715e>#Uniform dist between 0 and 2*pi</span>
</span></span><span style=display:flex><span>    <span style=color:#75715e>#Putting it here makes r calculated with starting values but not ending values</span>
</span></span><span style=display:flex><span>    r <span style=color:#f92672>=</span> (<span style=color:#ae81ff>1</span><span style=color:#f92672>/</span>len(theta))<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>sqrt(sum(np<span style=color:#f92672>.</span>sin(theta))<span style=color:#f92672>**</span><span style=color:#ae81ff>2</span><span style=color:#f92672>+</span>sum(np<span style=color:#f92672>.</span>cos(theta))<span style=color:#f92672>**</span><span style=color:#ae81ff>2</span>) 
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> j <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1</span>,len(theta)): <span style=color:#75715e>#for each pair of oscilators</span>
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>if</span> method <span style=color:#f92672>==</span> <span style=color:#e6db74>&#34;average&#34;</span>:
</span></span><span style=display:flex><span>            dtheta <span style=color:#f92672>+=</span> K<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>sin(psi <span style=color:#f92672>-</span> theta) <span style=color:#75715e>#Adding to whole arrays is easier</span>
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>elif</span> method <span style=color:#f92672>==</span> <span style=color:#e6db74>&#34;random&#34;</span>:
</span></span><span style=display:flex><span>            dtheta <span style=color:#f92672>+=</span> K<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>sin(forcing <span style=color:#f92672>-</span> theta) 
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>elif</span> method <span style=color:#f92672>==</span> <span style=color:#e6db74>&#34;oscilators&#34;</span>:
</span></span><span style=display:flex><span>            shift_theta <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>roll(theta,j)  <span style=color:#75715e># periodic boundary</span>
</span></span><span style=display:flex><span>            dtheta <span style=color:#f92672>+=</span> K<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>sin(shift_theta <span style=color:#f92672>-</span> theta)
</span></span><span style=display:flex><span>            oscilators_diff <span style=color:#f92672>=</span> shift_theta <span style=color:#f92672>-</span> theta
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>else</span>: <span style=color:#75715e>#Custom forcing</span>
</span></span><span style=display:flex><span>            dtheta <span style=color:#f92672>+=</span> K<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>sin(method <span style=color:#f92672>-</span> theta)
</span></span><span style=display:flex><span>    dtheta <span style=color:#f92672>+=</span> omega <span style=color:#75715e>#Adds intrinsic frequencies; they could be int or array</span>
</span></span><span style=display:flex><span>    new_theta <span style=color:#f92672>+=</span> dt<span style=color:#f92672>*</span>dtheta
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> r, new_theta<span style=color:#f92672>%</span>(<span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi), oscilators_diff  <span style=color:#75715e>#modulo 2*pi</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e>#Run new_theta recursively nstep times</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>do_nsteps</span>(theta,omega,K,dt,nsteps, method):
</span></span><span style=display:flex><span>    r_array <span style=color:#f92672>=</span> []; oscilators_diff <span style=color:#f92672>=</span> []; new_theta <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>copy(theta) <span style=color:#75715e>#Initializations</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(nsteps): 
</span></span><span style=display:flex><span>        r, new_theta, osc <span style=color:#f92672>=</span> get_new_theta(new_theta,omega,K,dt, method)
</span></span><span style=display:flex><span>        r_array<span style=color:#f92672>.</span>append(r); oscilators_diff<span style=color:#f92672>.</span>append(osc)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> r_array, new_theta, oscilators_diff
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>fill2d</span>(theta,omega,K,dt,nsteps,method):
</span></span><span style=display:flex><span>    zeta <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>zeros((len(theta),nsteps))   <span style=color:#75715e># our 2d array</span>
</span></span><span style=display:flex><span>    zeta[:,<span style=color:#ae81ff>0</span>] <span style=color:#f92672>=</span> theta  <span style=color:#75715e># initial conditions</span>
</span></span><span style=display:flex><span>    new_theta <span style=color:#f92672>=</span> theta
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1</span>,nsteps):
</span></span><span style=display:flex><span>        r, new_theta, oscilators_diff <span style=color:#f92672>=</span> get_new_theta(new_theta,omega,K,dt,method)
</span></span><span style=display:flex><span>        zeta[:,i] <span style=color:#f92672>=</span> new_theta
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> zeta
</span></span></code></pre></div></div><div class="cell markdown"><p>This code, which has been adapted from <a href=astro.pas.rochester.edu/~aquillen/phy256/problems/ps5+sol.html>a course by Rochester
University</a>,
implements the two methods we have been asked to implement and can work
with a number of N oscilators. Now, we would like to study the
syncronization transition through order parameter $r$ as $K$ varies
in the two ensembles. To do this, we first calculate r for a range of
100 values of k between 0 and 10.</p></div><div class="cell markdown"><p>Define the plotting function:</p></div><div class="cell code" data-execution_count=3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plot_r_values</span>(mymethod, legend<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>):
</span></span><span style=display:flex><span>    r_dict<span style=color:#f92672>=</span> {}
</span></span><span style=display:flex><span>    colors <span style=color:#f92672>=</span> list(Color(<span style=color:#e6db74>&#34;red&#34;</span>)<span style=color:#f92672>.</span>range_to(Color(<span style=color:#e6db74>&#34;blue&#34;</span>),<span style=color:#ae81ff>36</span>))
</span></span><span style=display:flex><span>    theta <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span>    omega <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> k <span style=color:#f92672>in</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>0.07</span>,<span style=color:#ae81ff>36</span>): <span style=color:#75715e>#Used cutre trial and error to settle on this values</span>
</span></span><span style=display:flex><span>        r_dict[k], theta, osc <span style=color:#f92672>=</span> do_nsteps(theta,omega,k,<span style=color:#ae81ff>0.1</span>,<span style=color:#ae81ff>100</span>, mymethod)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i, element <span style=color:#f92672>in</span> enumerate(r_dict):
</span></span><span style=display:flex><span>        plt<span style=color:#f92672>.</span>plot(np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>100</span>,<span style=color:#ae81ff>100</span>),r_dict[element], label<span style=color:#f92672>=</span>round(element,<span style=color:#ae81ff>2</span>), color<span style=color:#f92672>=</span><span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;</span><span style=color:#e6db74>{</span>colors[i]<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>) 
</span></span><span style=display:flex><span>        plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;Time&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>        plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#39;Order parameter&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>if</span> isinstance(mymethod, float):
</span></span><span style=display:flex><span>            plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;External coupling with $\Omega$ = </span><span style=color:#e6db74>{</span>round(mymethod, <span style=color:#ae81ff>3</span>)<span style=color:#e6db74>}</span><span style=color:#e6db74>&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>) 
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>else</span>:
</span></span><span style=display:flex><span>            plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;Evolution of the order parameter for </span><span style=color:#e6db74>{</span>mymethod<span style=color:#e6db74>}</span><span style=color:#e6db74> coupling&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>)
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>if</span> legend <span style=color:#f92672>==</span> <span style=color:#66d9ef>True</span>: plt<span style=color:#f92672>.</span>legend(ncol<span style=color:#f92672>=</span><span style=color:#ae81ff>2</span>)
</span></span></code></pre></div></div><div class="cell markdown"><p>Plot for average forcing and for external forcing of
<code>random.uniform(0, 2*np.pi)</code></p></div><div class="cell code" data-execution_count=4 data-scrolled=false><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plot_r_values(<span style=color:#e6db74>&#34;average&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plot_r_values(<span style=color:#e6db74>&#34;random&#34;</span>)
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Kuramoto%20Oscilators_8_0.png></figure></div></div><div class="cell markdown"><p>As we can see, the system coupled by average internal phase experiences
pretty fast syncronization, almost too fast to bee seen on the graph
(hum! suspicious&mldr; something might be wrong?). For the systems that
reach complete syncronization, that happens after only 2 to 3
oscilations, while the ones that reach partial sync take a bit more to
reach their maximum r, with that r being clearly dependent on K (as one
can see with the gradient) It should also be noted that the values of K
tested (between 0 and 0.07) are quite, quite low, so we can say this
system is <strong>highly sensible to coupling</strong>. Looking at the legends, the
transition to partial syncronization (total syncronization is more
difficult to study as previously explained) seems to occur after about 5
repetitions for k = 0.04; or after 15 rounds for k = 0.02</p><p>The system coupled using a random distribution, however, never reaches a
point where it becames stable; and this makes sense! By applying a
random external force in each oscilation, we are just &ldquo;driving the
system crazy&rdquo;, making it impossible for it to reach a steady state.
Thus, we will try to apply a more &ldquo;coherent&rdquo; external force, one which
will act as a &ldquo;master ocilator&rdquo; that helps all oscilators syncronize; in
this case, this forces will be $\pi$ and $\frac{\pi}{2}$</p></div><div class="cell code" data-execution_count=5 data-scrolled=true><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plot_r_values(np<span style=color:#f92672>.</span>pi, legend<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>) <span style=color:#75715e>#Already know the legend, make it clearer</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plot_r_values(np<span style=color:#f92672>.</span>pi<span style=color:#f92672>/</span><span style=color:#ae81ff>2</span>, legend<span style=color:#f92672>=</span><span style=color:#66d9ef>False</span>)
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Kuramoto%20Oscilators_10_0.png></figure></div></div><div class="cell markdown"><p>As we can see, both graphs are pretty simmilar: They have some complete
coupling, some partial coupling and some random noise (no coupling at
all). When compared with the previous graph, we can see than, unline
with external random force, coupling does appear; but, the coupling is
always less intense (there is less lines in the upper part of the graph)
than when using $\Psi$ as the coupling method. Thus, it can be said
that <strong>interal average phase is a better coupling method than an
external forcing</strong></p><p>Once again, the color gradient clearly shows how higher k values lead to
higher r values (makes sense)</p></div><div class="cell markdown"><p>Now, we would like to plot the distribution of phase differences between
all pairs of oscillators for different values of K, before and after the
transition. To do this, we obtain the values from get_new_theta
function. We use k = 0.05, and 5 vs 20 rounds of change:</p><p>(To avoid repeating work, I will do this for Average Phase Coupling
only)</p></div><div class="cell markdown"><p>First, we set the plotting function and the initial (must be common)
values for plotting:</p></div><div class="cell code" data-execution_count=6><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>theta <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>); omega <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>)
</span></span></code></pre></div></div><div class="cell code" data-execution_count=7 data-scrolled=false><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plot_phase_diffs</span>(k, numosc):
</span></span><span style=display:flex><span>    r, new_theta, osc <span style=color:#f92672>=</span> do_nsteps(theta,omega,k,<span style=color:#ae81ff>0.1</span>,numosc,<span style=color:#e6db74>&#34;oscilators&#34;</span>) <span style=color:#75715e>#Get things to plot</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;Oscilators&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#39;Phase&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;Phase of oscilators for k = </span><span style=color:#e6db74>{</span>k<span style=color:#e6db74>}</span><span style=color:#e6db74> &amp; </span><span style=color:#e6db74>{</span>numosc<span style=color:#e6db74>}</span><span style=color:#e6db74> oscilations&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(new_theta,<span style=color:#e6db74>&#39;bo&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;Number of oscilatos&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#39;Phase difference&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;Phase difference for k = </span><span style=color:#e6db74>{</span>k<span style=color:#e6db74>}</span><span style=color:#e6db74> &amp; </span><span style=color:#e6db74>{</span>numosc<span style=color:#e6db74>}</span><span style=color:#e6db74> oscilations&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>hist(osc[<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>], color<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;#0000ff&#34;</span>);
</span></span></code></pre></div></div><div class="cell markdown"><p>And then, we plot:</p></div><div class="cell code" data-execution_count=8><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plot_phase_diffs(<span style=color:#ae81ff>0.05</span>, <span style=color:#ae81ff>5</span>)
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Kuramoto%20Oscilators_17_0.png></figure></div></div><div class="cell markdown"><p>As we can see, after 5 oscilations, most oscilators stand completely
separated and different. Lets see after 20:</p></div><div class="cell code" data-execution_count=9 data-scrolled=true><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plot_phase_diffs(<span style=color:#ae81ff>0.05</span>, <span style=color:#ae81ff>20</span>)
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Kuramoto%20Oscilators_19_0.png></figure></div></div><div class="cell markdown"><p>Wow! As we can see, the oscilators have converged clearly: The
histogram shows a distribution much more centered than the previous one:
this is also visible at the left, where oscilators now seem much more
&ldquo;aligned!&rdquo; (its bad for me to say it, but it is definetely an amazing
data visualization hum)</p></div><div class="cell markdown"><p>Finally, we would like to vary $\omega$ and $K$ to stimate where the
arnold tongues are situated. Lets have a quick recap: arnold tongues
are, essentially, <strong>areas of entrainment</strong>, that is, the regions where
the oscilators tend to syncronize with each other. We can approximate
arnold tongues by using the order parameter $r$, given that, the
higher the r, the more syncronized; another, more accurate way of doing
it would be by plotting the same heatmap but with a comparison of the
oscillator frequencies $\theta$ and the external forcing $\Omega$.
To do this, we can define the following plotting function:</p></div><div class="cell code" data-execution_count=10><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>theta <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>); omega <span style=color:#f92672>=</span> <span style=color:#ae81ff>2</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi<span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>random(<span style=color:#ae81ff>100</span>) <span style=color:#75715e>#Reset initial values</span>
</span></span></code></pre></div></div><div class="cell code" data-execution_count=11><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plot_heatmap</span>(mymethod):
</span></span><span style=display:flex><span>    my_omegas <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>4</span><span style=color:#f92672>*</span>np<span style=color:#f92672>.</span>pi, <span style=color:#ae81ff>25</span>); my_ks <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>0.07</span>,<span style=color:#ae81ff>36</span>)
</span></span><span style=display:flex><span>    omega_axis <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>around(my_omegas, <span style=color:#ae81ff>2</span>); k_axis <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>around(my_ks, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>    r_matrix <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>DataFrame(index<span style=color:#f92672>=</span>my_ks, columns<span style=color:#f92672>=</span>my_omegas)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> omega <span style=color:#f92672>in</span> my_omegas:
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>for</span> k <span style=color:#f92672>in</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>0.07</span>,<span style=color:#ae81ff>36</span>):
</span></span><span style=display:flex><span>            r, new_theta, osc <span style=color:#f92672>=</span> do_nsteps(theta,omega,k,<span style=color:#ae81ff>0.1</span>,<span style=color:#ae81ff>10</span>,mymethod) 
</span></span><span style=display:flex><span>            r_matrix<span style=color:#f92672>.</span>at[k,omega] <span style=color:#f92672>=</span> round(r[<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>], <span style=color:#ae81ff>4</span>) <span style=color:#75715e>#Append r after n oscillations (n-1 actually)</span>
</span></span><span style=display:flex><span>    sns<span style=color:#f92672>.</span>heatmap(r_matrix<span style=color:#f92672>.</span>to_numpy()<span style=color:#f92672>.</span>tolist(), 
</span></span><span style=display:flex><span>                cmap <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;viridis&#34;</span>, xticklabels <span style=color:#f92672>=</span> omega_axis, yticklabels <span style=color:#f92672>=</span> k_axis)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>if</span> isinstance(mymethod, float):
</span></span><span style=display:flex><span>        plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;Heatmap of $r$ with $\Omega$ = </span><span style=color:#e6db74>{</span>round(mymethod, <span style=color:#ae81ff>2</span>)<span style=color:#e6db74>}</span><span style=color:#e6db74>&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>) 
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>else</span>:
</span></span><span style=display:flex><span>        plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#39;Heatmap of $r$ with </span><span style=color:#e6db74>{</span>mymethod<span style=color:#e6db74>}</span><span style=color:#e6db74> coupling&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>18</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;External forcing $\omega$&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#39;Coupling coefficient $k$&#39;</span>, fontsize<span style=color:#f92672>=</span><span style=color:#ae81ff>15</span>)
</span></span></code></pre></div></div><div class="cell markdown"><p>And plot it:</p></div><div class="cell code" data-execution_count=12><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>20</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plot_heatmap(<span style=color:#e6db74>&#34;average&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plot_heatmap(<span style=color:#e6db74>&#34;random&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plot_heatmap(np<span style=color:#f92672>.</span>pi)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plot_heatmap(np<span style=color:#f92672>.</span>pi<span style=color:#f92672>/</span><span style=color:#ae81ff>2</span>)
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Kuramoto%20Oscilators_25_0.png></figure></div></div><div class="cell markdown"><p>In the 4 graphs plotted above, we can see the following:</p><ul><li>The average-phase coupled map presents arnold tongues for all k &lt;
0.03 (approximately). Of course, $\omega$ does nothing here: this
map is not coupled by external force, but just by average phase!!</li><li>The randomly-coupled map has a completely random, pixelated
behaviour: we cannot find <strong>any</strong> arnold tongues (or any pattern
really)</li><li>For both of the externally coupled maps, arnold tongues are found,
and in a pretty simmilar way: the higher the k, the more likely the
oscilators will be coupled. However, with regards to omega, there
seems to be a treshold around $2 \cdot \pi$, after which no
coupling is found; in fact, there seems to be a point of &ldquo;maximum
disorder&rdquo; around $3 \cdot \pi$, after which syncronization seems
to get a bit better, probably doe to random syncronizations (noise)</li></ul></div><div class="cell markdown"><p>And, 춰that would mostly be it! As a bonus, if you wish, there is a way
to plot a dynamic phase map, 춰just as in <a href=https://www.complexity-explorables.org/slides/ride-my-kuramotocycle/>Ride my
Kuramotorcicle</a>!
using the same code from <a href=astro.pas.rochester.edu/~aquillen/phy256/problems/ps5+sol.html>Rochester
University</a>.
Please, refer to their course for full details.</p><p>As a closing note, one way to improve the code in this Jupyter Notebook
would be to rename the <code>theta</code> and <code>omega</code> vectors, which currently have
to be updated anytime a function is called, to avoid conflict; however,
I didnt have enough time for that :(</p></div></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f&text=The%20Kuramoto%20Oscilator%20Model&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f&title=The%20Kuramoto%20Oscilator%20Model" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f&name=The%20Kuramoto%20Oscilator%20Model" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f&title=The%20Kuramoto%20Oscilator%20Model" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f&title=The%20Kuramoto%20Oscilator%20Model" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=The%20Kuramoto%20Oscilator%20Model&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=The%20Kuramoto%20Oscilator%20Model - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=The%20Kuramoto%20Oscilator%20Model https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=The%20Kuramoto%20Oscilator%20Model&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fkuramoto-oscilators%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Modelizaci%c3%b3n%20de%20Sistemas%20Biol%c3%b3gicos/Kuramoto%20Oscilators/index.es.md title="Mejorar esta p치gina" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta p치gina</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Is free will or its counterpart, determinism, just a delusion?" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>Is free will or its counterpart, determinism, just a delusion?</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="El mapa Log칤stico" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>El mapa Log칤stico</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegaci칩n</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre m칤</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su direcci칩n de correo electr칩nico, acepta recibir el bolet칤n de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">춸 2020 Pablo Ignacio Marcos L칩pez</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Ap칩yame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>