<!doctype html><html><head><title>Designing a pharmacophore and ensuring ADMET properties</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Designing a pharmacophore and ensuring ADMET properties"><meta property="og:description" content="In the previous assignment, we managed, using ligand Based Virtual Screening, to obtain a list of those molecules that we considered to have &#34;high affinity&#34; with finasteride, a selective inhibitor of 5α-reductase type II, the enzyme that converts testosterone into dihydrotestosterone in the scalp and, therefore, that causes baldness. The idea was thus to find a molecule similar to finasteride and with the same properties, but even more selective and with greater affinity, in order to improve its therapeutic properties."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-12-12T00:00:00+00:00"><meta property="article:modified_time" content="2021-12-12T00:00:00+00:00"><meta name=description content="Designing a pharmacophore and ensuring ADMET properties"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=Búsqueda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinformática</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biología Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biología Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estadística</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentación>Presentación</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelización</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erdös-Rényi>Erdös-Rényi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poesía</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atrévete a Cruzar">Atrévete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="Año Nuevo">Año Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisión Inminente">Colisión Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmalión">Efecto Pigmalión</a></li><li><a href=/es/posts/poesia/efimero/ title=Efímero>Efímero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 🔥 🔥">Fuego 🔥 🔥</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocresía>Hipocresía</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducción>Introducción</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revolución">La Revolución</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Montaña Rusa">Montaña Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojalá>Ojalá</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 🎻 🎺">Orquesta 🎻 🎺</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plutón>Plutón</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=Pájaro>Pájaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulación>Simulación</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sueño>Sueño</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentación>Tentación</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Egoísmo>Egoísmo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Ignacio Marcos López</h5><p>domingo, 12 de diciembre de 2021</p></div><div class=title><h1>Designing a pharmacophore and ensuring ADMET properties</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><div style=text-align:center><a href="./Pharmacophore and ADMET.ipynb" target=_parent><img src=/posts/Imagenes/jupyter-badge.svg align=center width=20%></a></div></br><p>In the previous assignment, we managed, using l<strong>igand Based Virtual Screening</strong>, to obtain a list of those molecules that we considered to have "high affinity" with finasteride, a selective inhibitor of 5α-reductase type II, the enzyme that converts testosterone into dihydrotestosterone in the scalp and, therefore, that causes baldness. The idea was thus to find a molecule similar to finasteride and with the same properties, but even more selective and with greater affinity, in order to improve its therapeutic properties.</p><p>In this assignment, the idea is to perform a <strong>pharmacophore-based virtual screening</strong>. In our case, the pharmacophore we design will already be selective, as it is based on the properties of finasteride; therefore, what we are looking for is to improve as much as possible the <strong>affinity</strong>, which we described in the initial assignment through its IC~50~ of 16 nM.</p><p>To do this, the first step is to enter PDB and search for structures of our target protein (5α-reductase) associated with the selective inhibitor we have previously chosen (finasteride). In this case, we have only managed to find <strong>only one result</strong>,<sup id=fnref:1><a href=#fn:1 class=footnote-ref role=doc-noteref>1</a></sup> which is coupled with NADPH as a reaction intermediary.</p><div style=text-align:center><img src=./Finasteride-Coupled.png alt=sometext align=center width=50%><p>Figure 1. The Finasteride-coupled-to-NADPH complex I found in PDB. In yellow, with the joining Carbon atom highlighted in green, the bond between FNS (right, front) and NADPH (back). Nottice some SO4-2 ions on the back: I believe they act as coadyuvants.</p></div><p>This complex structure has raised several serious issues for me. Is it appropriate to cut the NADPH bond, and recreate the double bond in order to have a "guide molecule" that has at least the right structure to bind to the 5-aR? Or, on the contrary, is this a bad idea since, although the structure of the complex may be adequate to interact with the enzyme, it may not be, for example, adequate to access the active center? Does it make sense to cut the molecule in this way, when surely the binding of NADPH itself modifies its spatial conformation?</p><p>It is because of all these doubts that I have decided to dispense with this molecule. I am aware that, in doing so, I give up the possibility of having a guide molecule with the appropriate conformational structure to bind to the enzyme in question, but, in exchange, I make sure that I am not touching parameters that I do not fully understand and that may end up messing up the pharmacophore even more. In addition, PDB as a format does not support the double bonds<sup id=fnref:2><a href=#fn:2 class=footnote-ref role=doc-noteref>2</a></sup> that Finasteride presents, and, as much as I have tried with Chimera, I have not found a way to simulate them.</p><p>Therefore, to make up for this lack of valid crystallographic structures, I will take the 4 best hits from the file 'SimilarityAnalysisFinasteride.csv', measured by Dice (MACCS), which recall that in the previous assignment we had defined as the best similarity measurement method. Excluding <em>CHEMBL710</em> (which has an score of 1.0, since it is our initial drug, finasteride), we get the following list:</p><p style=text-align:center><strong>CHEMBL2282655 CHEMBL2282779 CHEMBL2282654 CHEMBL2282782</strong></p><p>These four compounds all come from the same paper,<sup id=fnref:3><a href=#fn:3 class=footnote-ref role=doc-noteref>3</a></sup> have the same mechanism as Finasteride (competitive inhibition) and use the same &ldquo;bolsillo protéico&rdquo;, which means they will be quite similar, and thus good at predicting a pharmacophore.<sup id=fnref1:3><a href=#fn:3 class=footnote-ref role=doc-noteref>3</a></sup></p><p>Once we have added the three compounds + the Finasteride to Chimera, we add Hydrogens, add Charges and minimize the structure five times, in order to provide as accurate a result as possible, and save as mol2.</p><div style=text-align:center><img src=./Pharmacophore-Precursors.png alt=sometext align=center width=50%><p>Figure 2. Molecule’s structure before (L) and after (R) minimization. Notice how the minimized structure is much more “loose”, letting molecules have their way.</p></div><p>For the next step, we would like to create a pharmacophore out of these molecules. For this, we merge the 4 individual mol2 files into one, and we upload it to PharmaGist.<sup id=fnref:4><a href=#fn:4 class=footnote-ref role=doc-noteref>4</a></sup> To account for its hight lipophilicity, the &ldquo;Hydrophobic&rdquo; advanced option could have been upped a bit, but, since I didn&rsquo;t know by how much, I prefered to leave the default values. Once we get the results, we select the target with the best score from the alignments that include the 4 molecules, and download the JMOL file, which includes <strong>our pharmacophore</strong>.</p><p>Now that we have created our pharmacophore, we can proceed with the part of interest in our analysis: the <strong>pharmacophore based virtual screening</strong>. For this, we will use Pharmit: first, we create a new database from the compounds discovered in Assignment 1 (SimilarityAnalysisFinasteride.csv), which we convert to .msi format by removing all columns except for SMILES and CHEMBLID and concatenating both of them with an space in between. Once we are sure the database has been correctly created (in our case, <strong>Finasteride-Similar 5ARIs KQMJ9Y</strong>, with 2,832 conformers and192 compounds), we can proceed with the scanning: we enter pharmit search, search for our database from the &ldquo;contributed libraries&rdquo; panel and add our pharmacores under the &ldquo;Load features&mldr;&rdquo; button.</p><p>Unfortunately, there are no hits for our chosen pharmacophore: this most likely means its structure is too restrictive, and thus that pharmit has problems finding matches. No problem! We can continue searching for matches using PharmaGist&rsquo;s list of suitable pharmacophores, descending one-by-one in order of Score: for #6, we found 6 matches, and 9 matches for #13; still a bit too little for our analysis. However, when we go for #10, ¡bingo! 65 matches are found, which reduce to 25 when we select 1 hit per configuration and per molecule. After ordering the matches in order of decreasing RMSD, we can save the .sdf results file.</p><p>And, just like this, ¡we have some drug candidates! Now, to make sure this drugs are good enough for clinical use, we would like to try and predict some of their ADME properties. The term ADME, which stands for Absorption, Distribution, Metabolism, and Excretion, describes how the drug interacts with the target organism, since the levels and kinetics of drug exposure to different tissues influence the degradation and processing of the drug, and thus its activity and efficiency. For this, we will use the accompanying Jupyter Notebook, which was kindly provided to us by the teacher.</p><p>This notebook takes the .sdf file and transforms it to CSV, which is more manageable, and evaluates whether the compounds fulfill Lipinski&rsquo;s rule of five. That rule, developed at Pfizer in 1997, evaluates how likely a drug candidate is to be an effective oral drug in humans, based on the idea that most orally-active compounds are small, lipofillic molecules.<sup id=fnref:5><a href=#fn:5 class=footnote-ref role=doc-noteref>5</a></sup> It thus demands that the mollecular weight is &lt; 500, that there are no more than 10 Hydrogen Bond acceptors and no more than 5 HB donors, and that its <em>n</em>-octanol-water partition coefficient (a measure of the relationship between lipophilicity and hydrophilicity) does not exceed 5 (high lipophilicity, but not too high).</p><p>If all 5 rules (or 4, since Lipinski originally merged the HB rules in one) are met, and if the Topological Polar Surface Area is less than 140 (since polar mollecules are bad at permeating cell membranes), the data is saved into a new file, &lsquo;MolDB_Ro5.csv&rsquo;, which you can find attached. We could also have allowed a rule to be broken without exclusion.; however, I found that too permissive, as it left out only 2 molecules (full rule-following left 13 in and 12 out, a ~50/50 split).</p><p>To better understand the statistical distribution of our drugs, we made the following graphs:</p><ul><li>A radar plot of the values of our selected &ldquo;drug candidates&rdquo;: here, we can see that none exceed the &ldquo;blue area&rdquo; of accepted values under the rule of five, although the more &ldquo;dangerous&rdquo; values (i.e. those that risk causing an exclusion the most) are molecular weight and <em>n</em>-octanol-water partition coefficient (here specified as LogP in log scale)</li></ul><div style=text-align:center><img src=./Radar-Plot.png alt=sometext align=center width=50%><p>Figure 3. Radar plot of the Ro5 (and therefore ADME) classification of those drugs that previously met the Rule of Five.Note how the characteristics with the highest deviation are molecular weight and logP</p></div><ul><li>Some histograms of the Ro5 properties of our drugs. Since we only found 13, it is a bit more difficult to extract conclussions from a histogram, but we can still see that the parameters with the greatest variance are in the number of hidrogen bond donors and acceptors, which can also be seen in the previous graphs. Molecular weight is actually pretty similar, which must mean that those molecules with more HBD/A are smaller, since hidrogen-bond formating atoms (usually N and O) are bigger than Carbon</li></ul><div style=text-align:center><img src=./Histograms.png alt=sometext align=center width=50%><p>Figure 4. Histogram of the Ro5 properties of our compounds. As can be seen, the greatest variance is in the number of hydrogen bond acceptors and donors, which indicates, together with the low variance in molecular mass, that the molecules must vary in size.</p></div><p>And, ¡that would be it! With the rule-of-five acting as a proxi for ADME properties acting as a proxy to ensure compliance with ADMET properties, these would be possible drugs to study as a substitute for finasteride. It remains for future researchers both to analyze the in vivo inhibitory capacity, side effects and viability of the drug, and to find more crystallographic structures for finasteride itself, which currently, to my surprise, is not as extensively studied as one would expect for one of the most important enzymes in the human (or at least in the male) body.</p><h2 id=references>References</h2><div class=footnotes role=doc-endnotes><hr><ol><li id=fn:1><p>Bank, RCSB Protein Data. RCSB PDB - 7BW1: Crystal Structure of Steroid 5-Alpha-Reductase 2 in Complex with Finasteride. <a href=https://www.rcsb.org/structure/7BW1>https://www.rcsb.org/structure/7BW1</a>. Accessed 12 december 2021.&#160;<a href=#fnref:1 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:2><p>«PDB File Format and Double Bonds». Avogadro Discussion, 19 de febrero de 2016, <a href=https://discuss.avogadro.cc/t/pdb-file-format-and-double-bonds/2293>https://discuss.avogadro.cc/t/pdb-file-format-and-double-bonds/2293</a>.&#160;<a href=#fnref:2 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:3><p>Kumar, Rajnish, y Manoj Kumar. «3D-QSAR CoMFA and CoMSIA Studies for Design of Potent Human Steroid 5α-Reductase Inhibitors». Medicinal Chemistry Research, vol. 22, n.o 1, enero de 2013, pp. 105-14. Springer Link, <a href=https://doi.org/10.1007/s00044-012-0006-1>https://doi.org/10.1007/s00044-012-0006-1</a>.&#160;<a href=#fnref:3 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a>&#160;<a href=#fnref1:3 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:4><p>Schneidman-Duhovny, D., et al. «PharmaGist: A Webserver for Ligand-Based Pharmacophore Detection». Nucleic Acids Research, vol. 36, n.o Web Server, mayo de 2008, pp. W223-28. DOI.org (Crossref), <a href=https://doi.org/10.1093/nar/gkn187>https://doi.org/10.1093/nar/gkn187</a>.&#160;<a href=#fnref:4 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:5><p>Lipinski, Christopher A., et al. «Experimental and Computational Approaches to Estimate Solubility and Permeability in Drug Discovery and Development Settings1PII of Original Article: S0169-409X(96)00423-1. The Article Was Originally Published in Advanced Drug Delivery Reviews 23 (1997) 3&ndash;25.1». Advanced Drug Delivery Reviews, vol. 46, n.o 1, marzo de 2001, pp. 3-26. ScienceDirect, <a href=https://doi.org/10.1016/S0169-409X(00)00129-0>https://doi.org/10.1016/S0169-409X(00)00129-0</a>.&#160;<a href=#fnref:5 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li></ol></div></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f&text=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f&title=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f&name=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f&title=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f&title=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Designing%20a%20pharmacophore%20and%20ensuring%20ADMET%20properties&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2flead-discovery%2fassignment-2%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Lead%20Discovery/Assignment%202/index.es.md title="Mejorar esta página" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta página</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Designing a pharmacophore and ensuring ADMET properties" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>Designing a pharmacophore and ensuring ADMET properties</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Using Computational Methods to discover novel drugs for the treatment of Androgenetic Alopecia" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Using Computational Methods to discover novel drugs for the treatment of Androgenetic Alopecia</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#references>References</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegación</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre mí</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su dirección de correo electrónico, acepta recibir el boletín de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">© 2020 Pablo Ignacio Marcos López</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Apóyame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>