<!doctype html><html><head><title>Preferential Attachment Networks</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Preferential Attachment Networks"><meta property="og:description" content="In the last exercise, we used an Erdös-Rényi random attachment model to analyze how interconnections influence the spread of illness over space. Erdös-Rényi, however, is not the only model that exists, and others, such as Preferential Attachment Networks, have the advantage of being more true to reality and an even closer representation of reality.
In these models, unlike in the random models, there is a preference when forming links between nodes: normally, nodes tend to cluster around the nodes with the largest number of linked nodes, promoting centralization and improving the connection between nodes; in this way, for example, the diameter of the network tends to decrease, since most nodes can be connected by traveling to the &ldquo;mother node&rdquo;."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-12-15T00:00:00+00:00"><meta property="article:modified_time" content="2021-12-15T00:00:00+00:00"><meta name=description content="Preferential Attachment Networks"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=Búsqueda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinformática</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biología Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biología Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estadística</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentación>Presentación</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelización</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erdös-Rényi>Erdös-Rényi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poesía</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atrévete a Cruzar">Atrévete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="Año Nuevo">Año Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisión Inminente">Colisión Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmalión">Efecto Pigmalión</a></li><li><a href=/es/posts/poesia/efimero/ title=Efímero>Efímero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 🔥 🔥">Fuego 🔥 🔥</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocresía>Hipocresía</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducción>Introducción</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revolución">La Revolución</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Montaña Rusa">Montaña Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojalá>Ojalá</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 🎻 🎺">Orquesta 🎻 🎺</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plutón>Plutón</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=Pájaro>Pájaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulación>Simulación</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sueño>Sueño</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentación>Tentación</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Egoísmo>Egoísmo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Marcos</h5><p>miércoles, 15 de diciembre de 2021</p></div><div class=title><h1>Preferential Attachment Networks</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><div style=text-align:center><a href="./Beyond Erdös-Rényi - Preferential Attachment Networks.ipynb" target=_parent><img src=/posts/Imagenes/jupyter-badge.svg align=center width=20%></a></div></br><div class="cell markdown"><p>In the last exercise, we used an Erdös-Rényi random attachment model to
analyze how interconnections influence the spread of illness over space.
Erdös-Rényi, however, is not the only model that exists, and others,
such as Preferential Attachment Networks, have the advantage of being
more true to reality and an even closer representation of reality.</p><p>In these models, unlike in the random models, there is a preference when
forming links between nodes: normally, nodes tend to cluster around the
nodes with the largest number of linked nodes, promoting centralization
and improving the connection between nodes; in this way, for example,
the diameter of the network tends to decrease, since most nodes can be
connected by traveling to the &ldquo;mother node&rdquo;.</p><p>This model, theorized by Barabási and Albert in 2000, works with an
initial grid full of vertices, which are connected either to each other
or to a larger node according to a series of possibilities.</p><p>To begin with, we will import the necessary modules:</p></div><div class="cell code" data-execution_count=1><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np                  <span style=color:#75715e>#Supports common numeric operations</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> matplotlib.pyplot <span style=color:#66d9ef>as</span> plt     <span style=color:#75715e>#For graphs</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> networkx <span style=color:#66d9ef>as</span> nx               <span style=color:#75715e>#For the creation, manipulation, and study of complex networks.</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> time<span style=color:#f92672>,</span> datetime               <span style=color:#75715e>#To check how long the script takes to run</span>
</span></span></code></pre></div></div><div class="cell markdown"><p>To make the task easier to work with, we will also import some of the
functions from the previous section:</p></div><div class="cell code" data-execution_count=2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>erdos_renyi</span>(nodes, link_prob):
</span></span><span style=display:flex><span>    x <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>zeros((nodes, nodes),int) <span style=color:#75715e>#start with n vertices</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(nodes):
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>for</span> j <span style=color:#f92672>in</span> range(nodes):
</span></span><span style=display:flex><span>            <span style=color:#66d9ef>if</span> np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>uniform(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>1</span>) <span style=color:#f92672>&lt;</span> link_prob:
</span></span><span style=display:flex><span>                x[i,j] <span style=color:#f92672>=</span> x[j,i] <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span>(x)
</span></span></code></pre></div></div><div class="cell code" data-execution_count=3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plot_adjacency</span>(adjacency_matrix):
</span></span><span style=display:flex><span>    gr <span style=color:#f92672>=</span> nx<span style=color:#f92672>.</span>from_numpy_matrix(adjacency_matrix)
</span></span><span style=display:flex><span>    mylabels <span style=color:#f92672>=</span> dict(zip(range(len(adjacency_matrix)), range(len(adjacency_matrix))))
</span></span><span style=display:flex><span>    nx<span style=color:#f92672>.</span>draw(gr, node_size<span style=color:#f92672>=</span><span style=color:#ae81ff>200</span>, edge_cmap <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Greys&#34;</span>, labels<span style=color:#f92672>=</span>mylabels, with_labels<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> nx<span style=color:#f92672>.</span>diameter(gr)
</span></span></code></pre></div></div><div class="cell markdown"><p>And now, we can define the functions that will help us generate the
preferential attachment network:</p></div><div class="cell code" data-execution_count=4><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>preferential_attachment</span>(nodes, links_per_newnode):
</span></span><span style=display:flex><span>    x <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>zeros((nodes, nodes),int) <span style=color:#75715e>#start with n vertices</span>
</span></span><span style=display:flex><span>    bonds <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>zeros(nodes,int)
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1</span>, links_per_newnode):
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>for</span> j <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1</span>, nodes<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>):
</span></span><span style=display:flex><span>            x[i,j] <span style=color:#f92672>=</span> x[j,i] <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>            bonds[i] <span style=color:#f92672>+=</span> <span style=color:#ae81ff>1</span>; bonds[j] <span style=color:#f92672>+=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>            
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(links_per_newnode <span style=color:#f92672>+</span> <span style=color:#ae81ff>1</span>, nodes):
</span></span><span style=display:flex><span>        prob <span style=color:#f92672>=</span> bonds <span style=color:#f92672>/</span> np<span style=color:#f92672>.</span>sum(bonds)
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>for</span> n <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1</span>, links_per_newnode):
</span></span><span style=display:flex><span>            r <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>uniform(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>            cum_p <span style=color:#f92672>=</span> <span style=color:#ae81ff>0</span>; j <span style=color:#f92672>=</span> <span style=color:#ae81ff>0</span>
</span></span><span style=display:flex><span>            <span style=color:#66d9ef>while</span> cum_p <span style=color:#f92672>&lt;</span> r:
</span></span><span style=display:flex><span>                j <span style=color:#f92672>+=</span> <span style=color:#ae81ff>1</span>; cum_p <span style=color:#f92672>+=</span> prob[j]
</span></span><span style=display:flex><span>            x[i,j] <span style=color:#f92672>=</span> x[j,i] <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>            bonds[i] <span style=color:#f92672>+=</span> <span style=color:#ae81ff>1</span>; bonds[j] <span style=color:#f92672>+=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>            prob[j] <span style=color:#f92672>=</span> <span style=color:#ae81ff>0</span>
</span></span><span style=display:flex><span>            prob <span style=color:#f92672>=</span> prob <span style=color:#f92672>/</span> np<span style=color:#f92672>.</span>sum(prob <span style=color:#f92672>+</span> <span style=color:#ae81ff>1e-7</span>) <span style=color:#75715e>#+ 1e-7 so that it doesnt produce an error</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span>(x)
</span></span></code></pre></div></div><div class="cell markdown"><p>And a 2D-lattice network; this I have done using only networkx, since I
am so short on time I cant afford to think how to do it.</p></div><div class="cell markdown"><p>To see how our graphs went, we would like to plot them; but, first, we
need to select only the biggest component:</p></div><div class="cell code" data-execution_count=5><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>purge</span>(adjacency_matrix):
</span></span><span style=display:flex><span>    graph <span style=color:#f92672>=</span> nx<span style=color:#f92672>.</span>from_numpy_matrix( adjacency_matrix )
</span></span><span style=display:flex><span>    sub_graph <span style=color:#f92672>=</span> graph<span style=color:#f92672>.</span>subgraph( (sorted(nx<span style=color:#f92672>.</span>connected_components(graph), key<span style=color:#f92672>=</span>len, reverse<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>))[<span style=color:#ae81ff>0</span>] )
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> nx<span style=color:#f92672>.</span>to_numpy_matrix( sub_graph )
</span></span></code></pre></div></div><div class="cell markdown"><p>And now, we can plot:</p></div><div class="cell code" data-execution_count=11><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>lattice_graph <span style=color:#f92672>=</span> nx<span style=color:#f92672>.</span>grid_2d_graph(<span style=color:#ae81ff>10</span>,<span style=color:#ae81ff>10</span>) 
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>d <span style=color:#f92672>=</span> plot_adjacency( purge(erdos_renyi(<span style=color:#ae81ff>35</span>, <span style=color:#ae81ff>0.1</span>)) ); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;Network diameter: </span><span style=color:#e6db74>{</span>d<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>d <span style=color:#f92672>=</span> plot_adjacency( purge(preferential_attachment(<span style=color:#ae81ff>35</span>, <span style=color:#ae81ff>2</span>)) ); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;Network diameter: </span><span style=color:#e6db74>{</span>d<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>d <span style=color:#f92672>=</span> plot_adjacency( purge(nx<span style=color:#f92672>.</span>to_numpy_matrix( lattice_graph ))  ); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;Network diameter: </span><span style=color:#e6db74>{</span>d<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Beyond%20Erd%c3%b6s-R%c3%a9nyi%20-%20Preferential%20Attachment%20Networks_11_0.png></figure></div></div><div class="cell markdown"><p>In the above graphs, one can clearly see the difference betweeen the
three topologies: one has completely random edges, with most nodes
exhibiting a similar number of conections and none clearly
predominating. The network diameter is 4, which is small but not as
small as in the preferential attachment network (diameter 2), where most
nodes can be easily connected through the middle one. Finally, the
2D-lattice has the biggest diameter (18) as there are no shortcuts and
one must traverse lots of nodes to get from one place to another.</p></div><div class="cell markdown"><p>To better understand how the structure affects outcome, lets carry out a
numerical simulation of a rumor propagation process that starts with a
single individual carrying a hot piece of news. At each time step,
individuals will pass the rumour to their neighbors with probability 𝛽
(per neighbor), and they will stop propagating the network with
probability 𝜇. We will compare these three structures with the
mean-field solution, which is propagation at a constant speed given by
the average number of contacts.</p></div><div class="cell markdown"><p>We define the propagating function:</p></div><div class="cell code" data-execution_count=6><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>propagate_rumour</span>(adjacency_matrix, pass_it <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3</span>, ignore <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.001</span>, iterations <span style=color:#f92672>=</span> <span style=color:#ae81ff>1000</span>):
</span></span><span style=display:flex><span>    my_graph <span style=color:#f92672>=</span> nx<span style=color:#f92672>.</span>from_numpy_matrix(adjacency_matrix)
</span></span><span style=display:flex><span>    new_infected <span style=color:#f92672>=</span> dict(zip(range(len(my_graph)), np<span style=color:#f92672>.</span>zeros(len(my_graph))))
</span></span><span style=display:flex><span>    infected_overtime <span style=color:#f92672>=</span> [<span style=color:#ae81ff>1</span>]; new_infected[np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>randint(len(my_graph))] <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> t <span style=color:#f92672>in</span> range(iterations<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>):
</span></span><span style=display:flex><span>        infected <span style=color:#f92672>=</span> new_infected
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>for</span> element <span style=color:#f92672>in</span> range(len(my_graph)): <span style=color:#75715e>#For each of the elemets of the graph</span>
</span></span><span style=display:flex><span>            <span style=color:#66d9ef>for</span> neighbor <span style=color:#f92672>in</span> my_graph<span style=color:#f92672>.</span>neighbors(element): <span style=color:#75715e>#Find its neighbors</span>
</span></span><span style=display:flex><span>                p <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>random<span style=color:#f92672>.</span>uniform(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>                <span style=color:#66d9ef>if</span> p <span style=color:#f92672>&gt;=</span> ignore:
</span></span><span style=display:flex><span>                    <span style=color:#66d9ef>if</span> infected[element] <span style=color:#f92672>==</span> <span style=color:#ae81ff>1</span> <span style=color:#f92672>and</span> p <span style=color:#f92672>&lt;</span> pass_it <span style=color:#f92672>and</span> infected[neighbor] <span style=color:#f92672>!=</span> <span style=color:#66d9ef>None</span>:
</span></span><span style=display:flex><span>                        new_infected[neighbor] <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>
</span></span><span style=display:flex><span>                <span style=color:#66d9ef>else</span>:
</span></span><span style=display:flex><span>                    new_infected[element] <span style=color:#f92672>=</span> <span style=color:#66d9ef>None</span>
</span></span><span style=display:flex><span>                    infected[element] <span style=color:#f92672>=</span> <span style=color:#66d9ef>None</span>
</span></span><span style=display:flex><span>        counter <span style=color:#f92672>=</span> [<span style=color:#ae81ff>1</span> <span style=color:#66d9ef>if</span> x <span style=color:#f92672>==</span> <span style=color:#ae81ff>1</span> <span style=color:#66d9ef>else</span> <span style=color:#ae81ff>0</span> <span style=color:#66d9ef>for</span> x <span style=color:#f92672>in</span> new_infected<span style=color:#f92672>.</span>values()] <span style=color:#75715e>#Cannot sum &#34;None&#34;</span>
</span></span><span style=display:flex><span>        infected_overtime<span style=color:#f92672>.</span>append(sum(counter))
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>if</span> sum(counter) <span style=color:#f92672>==</span> <span style=color:#ae81ff>0</span>: <span style=color:#66d9ef>return</span> infected_overtime, t
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> infected_overtime
</span></span></code></pre></div></div><div class="cell markdown"><p>And we run the simulations 1000 times for up to 1000 steps. Please note
that this cell can take suuuuuuper long to run, since it is doing
1000x1000 repetitions. Please, bear in mind that, in my computer, it
took 1 full hour to run.</p></div><div class="cell code" data-execution_count=7><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>erdos_endpoint <span style=color:#f92672>=</span> preferential_endpoint <span style=color:#f92672>=</span> lattice_endpoint <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>print(datetime<span style=color:#f92672>.</span>datetime<span style=color:#f92672>.</span>now())
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> repetition <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>1000</span>):
</span></span><span style=display:flex><span>    print(<span style=color:#e6db74>f</span><span style=color:#e6db74>&#34;Repetition #</span><span style=color:#e6db74>{</span>repetition<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>, end <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;</span><span style=color:#ae81ff>\r</span><span style=color:#e6db74>&#34;</span>, flush<span style=color:#f92672>=</span><span style=color:#66d9ef>True</span>)
</span></span><span style=display:flex><span>    erdos_endpoint<span style=color:#f92672>.</span>append(propagate_rumour(purge(erdos_renyi(<span style=color:#ae81ff>100</span>, <span style=color:#ae81ff>0.1</span>)))[<span style=color:#ae81ff>1</span>])
</span></span><span style=display:flex><span>    preferential_endpoint<span style=color:#f92672>.</span>append(propagate_rumour(purge(preferential_attachment(<span style=color:#ae81ff>100</span>, <span style=color:#ae81ff>3</span>)))[<span style=color:#ae81ff>1</span>])
</span></span><span style=display:flex><span>    lattice_endpoint<span style=color:#f92672>.</span>append(propagate_rumour(nx<span style=color:#f92672>.</span>to_numpy_matrix(nx<span style=color:#f92672>.</span>grid_2d_graph(<span style=color:#ae81ff>10</span>,<span style=color:#ae81ff>10</span>)))[<span style=color:#ae81ff>1</span>])
</span></span><span style=display:flex><span>print(datetime<span style=color:#f92672>.</span>datetime<span style=color:#f92672>.</span>now())
</span></span></code></pre></div><div class="output stream stdout"><pre><code>2021-12-14 16:22:29.960008
2021-12-14 17:12:37.746489
</code></pre></div></div><div class="cell markdown"><p>And we plot some histograms. Please note that I reduced the number of
bins from 1000 to 100, since what I want to look at is not the precise
distribution by exact time of ending, but rather a general idea about
whether the simulation ends prematurely, towards the middle, at the end,
or does not finish at all by the time of reporting.</p></div><div class="cell code" data-execution_count=37><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>1</span>); plt<span style=color:#f92672>.</span>ylim(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>hist(erdos_endpoint, bins <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#34;Time of ending for 1000 runs on Erdös-Rényi&#34;</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>2</span>); plt<span style=color:#f92672>.</span>ylim(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>hist(preferential_endpoint, bins <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#34;Time of ending for 1000 runs on Preferential Endoint&#34;</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>3</span>); plt<span style=color:#f92672>.</span>ylim(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>hist(lattice_endpoint, bins <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>); plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#34;Time of ending for 1000 runs on a 2D-Lattice&#34;</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Beyond%20Erd%c3%b6s-R%c3%a9nyi%20-%20Preferential%20Attachment%20Networks_19_0.png></figure></div></div><div class="cell markdown"><p>What we can see is that, in fact, the three gaphs are quite similar, if
not identical. Part of this has to do with the reduction in the number
of bins (grouping in groups of 10 necessarily reduces resolution), but
part of this must also have to do with the underlying mechanisms of the
data. This has greatly surprised me: I would have expected the
Preferentially-attached network to be at least more rumour-prone than
the erdos-renyi one, and I would have expected the 2D-lattice to provide
a middle ground between them, maybe with rumours spreading for longer
given that the network diameter is so large. That this didn&rsquo;t happen
seems to contradict my hypothesis from the last exercise, that network
topology is determinist towards what will happen; of course, this could
also be just an artifact due to the different enunciados. I dont know.</p></div><div class="cell markdown"><p>Anyways, with regards to the found results, we can clearly identify
three groups:</p><ul><li><strong>Early finishers</strong>: Actually, most of the trajectories in all
networks end at some point between 1 and ~10 repetitions, so much
so that I have been forced to cut the y-axis so that the rest of the
endpoints could be seen. This has surprised me, but it actually
makes sense: I selected such an small transmission value <em>0.3</em>, that
it is actually quite difficult for local <em>cotillas</em> to spread the
word, specially since most of the nodes have an average of ~3
joined nodes. This was made on purpouse, no make the networks
comparable, but, since $3 \cdot 0.3 = 1$, it is actually difficult
to spread things over.</li><li><strong>The fading fashion</strong>: For those rumours that manage to break out,
there seems to be a normal-like distribution centered around
$t = 300$: The rumour spreads for some time, until all individuals
become inmune, and then it dies out.</li><li><strong>The enduring cult</strong>: Some rumours manage to fo even further,
trespassing a &ldquo;barrier&rdquo; around 800 to live for longer, and maybe
even perdure over time: around 20 simulations hadn&rsquo;t actually ended
by the time of reporting.</li></ul></div><div class="cell markdown"><p>Since the initial individual was selected at random, we can for sure say
that this is not a relevant factor on the outcome: what, then, decides
whether a rumour spreads out or not? What defines, then, wether an
individual survives or not? With the graphs and the information that I
have, I am sadly forced to say that this must be pure chance: if it was
network topology, some differences would have been observed between the
three histograms.</p></div><div class="cell markdown"><p>Another option is that network topology <strong>is relevant</strong> on the small
scale, but that, with such a big number of repetitions, that variability
becomes less relevant, forcing the model to &ldquo;collapse&rdquo; around the
explained groups.</p></div></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f&text=Preferential%20Attachment%20Networks&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f&title=Preferential%20Attachment%20Networks" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f&name=Preferential%20Attachment%20Networks" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f&title=Preferential%20Attachment%20Networks" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f&title=Preferential%20Attachment%20Networks" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Preferential%20Attachment%20Networks&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Preferential%20Attachment%20Networks - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Preferential%20Attachment%20Networks https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Preferential%20Attachment%20Networks&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fbeyond-erdos-renyi%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Modelizaci%c3%b3n%20de%20Sistemas%20Biol%c3%b3gicos/Beyond%20Erdos-Renyi/index.es.md title="Mejorar esta página" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta página</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="El mapa Logístico" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>El mapa Logístico</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Modelling the syncronization pattern of Photinus carolinus in silico using cellular automata" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Modelling the syncronization pattern of Photinus carolinus in silico using cellular automata</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegación</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre mí</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su dirección de correo electrónico, acepta recibir el boletín de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">© 2020 Pablo Ignacio Marcos López</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Apóyame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>