<!doctype html><html><head><title>GDAV Single Cell - Ex 3</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="GDAV Single Cell - Ex 3"><meta property="og:description" content="Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data. In this exercise, we will learn to use it!
## Set up images knitr::opts_chunk$set(dev = &#34;png&#34;, dpi = 1000, echo = FALSE, cache = TRUE) Using the raw data in file “SCOplanaria."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/"><meta property="og:image" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ex3_feature_selection.png"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-02-01T00:00:00+00:00"><meta property="article:modified_time" content="2022-02-01T00:00:00+00:00"><meta name=description content="GDAV Single Cell - Ex 3"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=Búsqueda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinformática</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biología Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biología Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estadística</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentación>Presentación</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelización</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erdös-Rényi>Erdös-Rényi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poesía</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atrévete a Cruzar">Atrévete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="Año Nuevo">Año Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisión Inminente">Colisión Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmalión">Efecto Pigmalión</a></li><li><a href=/es/posts/poesia/efimero/ title=Efímero>Efímero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 🔥 🔥">Fuego 🔥 🔥</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocresía>Hipocresía</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducción>Introducción</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revolución">La Revolución</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Montaña Rusa">Montaña Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojalá>Ojalá</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 🎻 🎺">Orquesta 🎻 🎺</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plutón>Plutón</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=Pájaro>Pájaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulación>Simulación</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sueño>Sueño</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentación>Tentación</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Egoísmo>Egoísmo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Ignacio Marcos López</h5><p>martes, 1 de febrero de 2022</p></div><div class=title><h1>GDAV Single Cell - Ex 3</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><p>Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data. In this exercise, we will learn to use it!</p><pre tabindex=0><code class=language-{r data-lang={r>## Set up images
knitr::opts_chunk$set(dev = &#34;png&#34;, dpi = 1000, echo = FALSE, cache = TRUE)
</code></pre><p>Using the raw data in file “SCOplanaria.txt” we will:</p><pre tabindex=0><code class=language-{r, data-lang={r,>## First, we will import the necessary packages
packages &lt;- c(&#34;Matrix&#34;, &#34;dplyr&#34;, &#34;Seurat&#34;)
##for (x in packages) {install.packages(x)}
lapply(packages, require, character.only = TRUE)
</code></pre><h2 id=initialize-a-seurat-object>Initialize a Seurat object</h2><p>The steps can be consulted below:</p><pre tabindex=0><code class=language-{r, data-lang={r,>
## First, we read the data from the .txt file:
table &lt;- read.table(&#34;./SCOplanaria.txt&#34;, row.names = 1, header = FALSE)

## To save memory, and since the current dataset has lots of 0&#39;s, we convert
## the table -&gt; sparse matrix
table &lt;- as.matrix(table, sparse=TRUE)

## And, we create the Seurat object
SCOplanaria &lt;- CreateSeuratObject(counts = table, project = &#34;SCOPlanaria&#34;)

## We have created an RNA assay with 24883 features for 1871 cells.
## We do not want to filter, but, rather, to use all the available data
## We can inspect both the data propper
SCOplanaria[[&#34;RNA&#34;]]@counts[1:5,1:5] 
</code></pre><pre tabindex=0><code class=language-{r, data-lang={r,>## And the meta-data
SCOplanaria@meta.data [1:5,] 
## Here, we can see the read counts per cell (nCount_RNA) and the total
## number of expressed genes per cell (nFeature_RNA)
</code></pre><h2 id=pre-process-the-data>Pre-process the data</h2><h3 id=perform-quality-control>Perform quality control</h3><p>For the QC, we would like to select only those samples that are acceptable in quality. For this, we can do a number of data visualizations: for example, we can show the meta.data matrix using a violin plot like this:</p><pre tabindex=0><code class=language-{r, data-lang={r,># Violin plot and scatter plots are for data discovery only
png(file=&#34;ex3_violin_pre.png&#34;,res=300, width = 5400, height = 5200, units = &#34;px&#34;)
VlnPlot(object = SCOplanaria, features= c(&#34;nFeature_RNA&#34;, &#34;nCount_RNA&#34;), ncol = 2)
dev.off()
</code></pre><img src=./unnamed-chunk-4-1.png width=70% class=center><p>In this plot, each point represents an individual read, with the red area being the rotated kernel density plot for the values. As one can read in the title, the left plot represents the number of genes expressed per cell, with the one in the one on the right showing the number of reads per cell. At first sight, there seems to be more counts than features; this is because each gene can have one or more reads.</p><p>Now, we can visualize the metadata with the GenePlot function, to try and see if there are any outliers which we should remove.</p><pre tabindex=0><code class=language-{r, data-lang={r,>png(file=&#34;ex3_scatter_pre.png&#34;,res=300, width = 5400, height = 5200, units = &#34;px&#34;)
FeatureScatter(SCOplanaria, feature1 = &#34;nFeature_RNA&#34;, feature2 = &#34;nCount_RNA&#34;)
dev.off()
</code></pre><img src=./unnamed-chunk-5-1.png width=70% class=center><p>This function generates an scatter plot for our data, in which we can see that there is a 95% positive correlation, which is pretty high and actually statistically significant. We can try to improve this by removing the values with a nFeature_RNA over 1800, and those with a nCount_RNA &lt; 4500.</p><pre tabindex=0><code class=language-{r, data-lang={r,># Quality control
png(file=&#34;ex3_scatter_post.png&#34;,res=300, width = 5400, height = 5200, units = &#34;px&#34;)
SCOplanaria &lt;- subset(SCOplanaria, subset = nFeature_RNA &lt; 1500 &amp; nCount_RNA &lt; 4000)
FeatureScatter(SCOplanaria, feature1 = &#34;nFeature_RNA&#34;, feature2 = &#34;nCount_RNA&#34;)
dev.off()

# Normalization
png(file=&#34;ex3_violin_post.png&#34;,res=300, width = 5400, height = 5200, units = &#34;px&#34;)
SCOplanaria &lt;- NormalizeData(SCOplanaria, normalization.method = &#34;LogNormalize&#34;)
normaldata &lt;- colSums(SCOplanaria[[&#34;RNA&#34;]]@data)
SCOplanaria &lt;- AddMetaData (SCOplanaria, normaldata, col.name=&#34;Normalized_Data&#34;)
VlnPlot(object = SCOplanaria, features= c(&#34;nCount_RNA&#34;, &#34;Normalized_Data&#34;), ncol = 2)
dev.off()
</code></pre><img src=./unnamed-chunk-6-1.png width=70% class=center><p>Here, we see that, while the correlation does not get any better, it also does not get worse, and there are in fact way less outliers, which is always desirable.</p><h3 id=normalization>Normalization</h3><p>We can normalize the gene expression for all cells by multiplying each cell&rsquo;s baseline expression by a scale factor of 10,000, and log-transforming the result. The new values will be stored in [[&ldquo;RNA&rdquo;]]@data (as Seurat forces us to use an existing slot for the data), and can be visualized, using its equivalent violin plot(s), as follows:</p><pre tabindex=0><code class=language-{r, data-lang={r,>SCOplanaria &lt;- NormalizeData(SCOplanaria, normalization.method = &#34;LogNormalize&#34;)
normaldata &lt;- colSums(SCOplanaria[[&#34;RNA&#34;]]@data)
SCOplanaria &lt;- AddMetaData (SCOplanaria, normaldata, col.name=&#34;Normalized_Data&#34;)
VlnPlot(object = SCOplanaria, features= c(&#34;nCount_RNA&#34;, &#34;Normalized_Data&#34;), ncol = 2)
</code></pre><img src=./unnamed-chunk-7-1.png width=70% class=center><p>As we can see, due to the normalization, the data is much more evenly split over the values.</p><h3 id=feature-selecion>Feature selecion</h3><p>We would also like to separate highly variable genes (those that show differential expression across tissues, for example), focusing on them for downstream analysis; seurat does this by modelling the mean-variance relationship od single-cell data, returning 2000 of the most important features. Using this, we will identify the top10 most important genes, and plot them to better understand which they are:</p><pre tabindex=0><code class=language-{r, data-lang={r,>png(file=&#34;ex3_feature_selection.png&#34;,res=300, width = 2000, height = 2000, units = &#34;px&#34;)
SCOplanaria &lt;- FindVariableFeatures(SCOplanaria, selection.method = &#34;vst&#34;)
best_hits &lt;- head(VariableFeatures(SCOplanaria), 10)
VariableFeaturePlot(SCOplanaria)
# One can also add the labels using the following code; NoLegend because it was ugly
LabelPoints(plot = VariableFeaturePlot(SCOplanaria), points = best_hits, repel = TRUE
            ) + NoLegend()
dev.off()
</code></pre><img src=./ex3_feature_selection.png width=50% class=center><h3 id=scaling-of-the-data>Scaling of the data</h3><p>Finally, we can center and scale the data into a linear model, assuring that the mean expression across cells is 0 and the variance is 1 and savung the results in SCOplanaria[[&ldquo;RNA&rdquo;]]@scale.data. This gives equal weight to all cells, ensuring there is no over-domination by highly-expressed cells.</p><pre tabindex=0><code class=language-{r, data-lang={r,>SCOplanaria &lt;- ScaleData(SCOplanaria, features = rownames(SCOplanaria))
</code></pre><h2 id=pca-analysis>PCA Analysis</h2><p>After pre-processing the data, we can finally run the PCA analysis! To understand the results, first, we will group the genes based on behaviour and plot a heatmap with the 15 most variable genes. We can also plot the cells based on which component they belong to, and show the scores for each variant and component, with the following code:</p><pre tabindex=0><code class=language-{r, data-lang={r,># Run the PCA itself
SCOplanaria &lt;- RunPCA(SCOplanaria, features = VariableFeatures(object = SCOplanaria))
# Some graphs we dont care much about
DimHeatmap(SCOplanaria, dims = 1:9, cells = 500, balanced = TRUE)
VizDimLoadings(SCOplanaria, dims = 1, reduction = &#34;pca&#34;)
DimPlot(SCOplanaria, reduction = &#34;pca&#34;)
</code></pre><div class=row><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-10-1.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-10-2.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-10-3.png alt></div></div><p>Seurat may also plot the cells by their PCA scores, which would help us group genes by common characteristics (for example, gene families). To determine the number of components, seurat automatically calculates significancy values for N components and reruns the analysis until there isa good distribution of gene scores, in a method that was inspired by the JackStraw algorithm. There, significant components would be those with a strong enrichment of low p-values, represented as dots over a curved line.</p><pre tabindex=0><code class=language-{r, data-lang={r,>SCOplanaria &lt;- JackStraw(SCOplanaria, num.replicate = 100)
SCOplanaria &lt;- ScoreJackStraw(SCOplanaria, dims = 1:20)
# Draw JackStraw and ElbowPlot
png(file=&#34;ex3_jackstraw_1.png&#34;,res=300, width = 2000, height = 2000, units = &#34;px&#34;)
JackStrawPlot(SCOplanaria, dims = 1:15)
dev.off()
png(file=&#34;ex3_elbowplot_1.png&#34;,res=300, width = 1300, height = 2000, units = &#34;px&#34;)
ElbowPlot(SCOplanaria)
dev.off()
</code></pre><div class=row><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-11-1.png alt></div><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-12-1.png alt></div></div><p>We may also plot a traditional elbow plot, in which we show the standard deviation against the number of Principal Components. As it turns out, a sharp drop in SD occurs after PC 19, which means that, at said point of inflexion, adding an additional PC does not add much to the data; thus, we can keep 10 as our number of PCs.</p><h2 id=cell-clustering>Cell clustering</h2><p>Now that we have defined an adecuate number of Principal Components, we can run a clustering to see if, in fact, we can generate real groups of families of genes. This clusters can be visualized through non-linear dimensional reduction, and then plotted, like so:</p><pre tabindex=0><code class=language-{r, data-lang={r,>SCOplanaria &lt;- FindNeighbors(SCOplanaria, dims = 1:10)
SCOplanaria &lt;- FindClusters(SCOplanaria, resolution = 0.5)
SCOplanaria &lt;- RunUMAP(SCOplanaria, dims = 1:10)
DimPlot(SCOplanaria, reduction = &#34;umap&#34;)
SCOplanaria &lt;- RunTSNE(SCOplanaria, dims = 1:10)

png(file=&#34;ex3_dimpolot_10PCs_2.png&#34;,res=300, width = 1300, height = 2000, units = &#34;px&#34;)
DimPlot(SCOplanaria, reduction = &#34;tsne&#34;)
dev.off()
</code></pre><div class=row><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-13-1.png alt></div><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-13-2.png alt></div></div><h2 id=new-parameters>New Parameters</h2><p>To validate our results, we will reproduce the analysis by keeping only genes expressed in at least 3 cells and cells with at least 200 and at most 2500 genes, lognormalizing with a scale of 10000 and selecting the 3000 most variable features. We will also regress out any variability, and use PCs 1-5 with a resolution of 0.6</p><pre tabindex=0><code class=language-{r, data-lang={r,>New_plan &lt;- CreateSeuratObject(counts = table, project = &#34;Planaria&#34;, 
                               min.cells = 3, min.features = 200)
New_plan &lt;- subset(New_plan, subset = nFeature_RNA &lt; 2500 &amp; nFeature_RNA &gt; 200)
New_plan &lt;- NormalizeData(New_plan, normalization.method = &#34;LogNormalize&#34;, 
                          scale.factor = 10000)
New_plan &lt;- AddMetaData(New_plan,colSums(New_plan[[&#34;RNA&#34;]]@data),col.name=&#34;Norma Data&#34;)
New_plan &lt;- FindVariableFeatures(New_plan, selection.method = &#34;vst&#34;, nfeatures = 3000)
New_plan &lt;- ScaleData(New_plan, features = rownames(New_plan))
New_plan &lt;- RunPCA(New_plan, features = VariableFeatures(object = New_plan))

png(file=&#34;ex3_elbowplot_2.png&#34;,res=300, width = 1300, height = 2000, units = &#34;px&#34;)
ElbowPlot(New_plan)
dev.off()

New_plan &lt;- FindNeighbors(New_plan, dims = 1:5)
New_plan &lt;- FindClusters(New_plan, resolution = 0.6)
New_plan &lt;- RunUMAP(New_plan, dims = 1:5)
DimPlot(New_plan, reduction = &#34;umap&#34;)
New_plan &lt;- RunTSNE(New_plan, dims = 1:5)
DimPlot(New_plan, reduction = &#34;tsne&#34;)
</code></pre><div class=row><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-14-1.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-14-2.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./unnamed-chunk-14-3.png alt></div></div><p>As we can see in the elbow plot, there is not much of a difference when we account for the small changes introduced now; in both cases, there are two possible elbows, at 5 and 10 clusters, respectively. This is good! It means our results are robust, and do not depend so much in discretionary parameters which could be misinterpreted. If we look at the graph itself, we will see that its shape is quite different, but mostly in the group shapes, since their size is quite similar. The two options, however, seem good enough, which reinforces the idea that our results are robust independently of the number of PCs chosen.</p><h2 id=finding-biomarkers-through-differential-gene-expression>Finding Biomarkers through Differential Gene Expression</h2><p>For each of the gene expression clusters that we have previously defined, we would like to find a series of associated biomarkers, to try and see whether they relate to a specific cellular function or not. For this, Seurat has a built-in function that analyzes each cluster with regards to all others, analyzing whether some biomarkers show a disproportionate expression in each cluster or not. Since, as we have already discussed in the previous section, 5 PCs seem to show better adjustment to the data than 10 PCs, we will use the data from our repetition for this analysis.</p><pre tabindex=0><code class=language-{r, data-lang={r,>## Calculate biomarkers
New_plan.biomarkers &lt;- FindAllMarkers(New_plan, only.pos = TRUE,
                                      min.pct = 0.25, logfc.threshold = 0.25)

## Select the best hits (we chose 5)
best_hit_biomarkers &lt;- (New_plan.biomarkers %&gt;% group_by(cluster) 
                        %&gt;% slice_max(n = 5, order_by = avg_log2FC))

## And show them as a heatmap
# And show them as a heatmap
png(file=&#34;ex3_heatmap.png&#34;,res=300, width = 2000, height = 1500, units = &#34;px&#34;)
DoHeatmap(New_plan, features = best_hit_biomarkers$gene) + NoLegend()
dev.off()
</code></pre><img src=./unnamed-chunk-15-1.png width=70% class=center><p>The results seem promising: for each gene cluster, from 0 to 9, we find that each one has a series of clearly differentiated markers (in yellow) and that, except in clusters 4-8-9 and 5-7, there are no duplicities, as they are marked on the diagonal. This indicates that we will surely be able to differentiate the gene clusters by their markers, and, therefore, that they probably have a differentiated biological function.</p><h2 id=showing-the-expression-distribution-of-these-markers>Showing the expression distribution of these markers</h2><p>Finally, we can directly show the distribution of the expression of these markers between the different clusters, by means of both violin plots and feature plots. For them, we can choose between t-SNE and uMap as methods; since t-SNE very good at creating a single map that reveals structure at many different scales, we decided to use that algorithm for our dimensionality reduction.</p><pre tabindex=0><code class=language-{r, data-lang={r,>## First, we set the col names. Since R does not let us create a dictionary,
## we must declare two vectors:
gene_names = c(&#39;dd-Smed-v6-61-0&#39;, &#39;dd-Smed-v6-2178-0&#39;, &#39;dd-Smed-v6-298-0&#39;,
               &#39;dd-Smed-v6-1410-0&#39;, &#39;dd-Smed-v6-702-0&#39;, &#39;dd-Smed-v6-2548-0&#39;,
               &#39;dd-Smed-v6-9977-0&#39;, &#39;dd-Smed-v6-48-0&#39;, &#39;dd-Smed-v6-175-0&#39;,
               &#39;dd-Smed-v6-1161-1&#39;)

cell_identities = c(&#39;Early epidermal progenitors&#39;, &#39;Late epidermal progenitors&#39;,
                    &#39;Epidermis&#39;, &#39;Muscle progenitors&#39;, &#39;Muscle body&#39;,
                    &#39;Neural progenitors&#39;, &#39;GABA neurons&#39;, &#39;Phagocytes&#39;,
                    &#39;Parenchymal cells&#39;, &#39;Pigment&#39;)

# The full plots. Not of much use, but, if I dont save them
# R just makes them hyper mega o sea super-super ugly
# (gosh I hate R)
png(file=&#34;ex3_full_violinplots.png&#34;, res=300, width = 5000, height = 4000, units = &#34;px&#34;)
VlnPlot(New_plan, features = gene_names)
dev.off()
png(file=&#34;ex3_full_featureplots.png&#34;, res=300, width = 5000, height = 4000, units = &#34;px&#34;)
FeaturePlot(New_plan,  features = gene_names, reduction = &#34;tsne&#34;)
dev.off()
</code></pre><p><img src=./ex3_full_violinplots.png alt> <img src=./ex3_full_featureplots.png alt></p><p>We can also show our renamed clusters based on the information we now know; since all of the data-processing steps are already done, this is just a visualization step, in which we get to see how our analysis went:</p><pre tabindex=0><code class=language-{r, data-lang={r,>
## Rename the columns
names(cell_identities) &lt;- levels(New_plan)
New_plan &lt;- RenameIdents(New_plan, cell_identities)

## Or the feature/violin plot for some of the elements
VlnPlot(New_plan, features = &#39;dd-Smed-v6-9977-0&#39;)
FeaturePlot(New_plan, features = &#39;dd-Smed-v6-9977-0&#39;, reduction = &#34;tsne&#34;)

VlnPlot(New_plan, features = &#39;dd-Smed-v6-1999-0&#39;)
FeaturePlot(New_plan, features = &#39;dd-Smed-v6-1999-0&#39;, reduction = &#34;tsne&#34;)
</code></pre><p><div class=row><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-17-2.png alt></div><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-17-3.png alt></div></div><div class=row><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-17-4.png alt></div><div class="col col-sm-12 col-lg-6"><img src=./unnamed-chunk-17-5.png alt></div></div></p><p>As we can see, it seems that the biomarker <em>dd-Smed-v6-9977-0</em>, related to cluster 4, is clearly related to muscle progenitors, its expression being highly localised in the lower left side of the image, as can be seen in the individual feature plot. However, for <em>dd-Smed-v6-1999-0</em>, mainly related to epidermal, phagocyte and GABA neuron progenitors, it seems that its expression is much more diffuse, which may be due both to the fact that it is a poorer quality cluster and that, as it contains so many features, it is easier to find dispersion.</p><p>This analysis could be done individually for each of the clusters, but for the purpose of the exercise at hand, I believe that this short description serves to show how to proceed in the clusters.</p><pre tabindex=0><code class=language-{r, data-lang={r,># We can show the labeled dimplot
names(cell_identities) &lt;- levels(New_plan)
New_plan &lt;- RenameIdents(New_plan, cell_identities)
png(file=&#34;ex3_labeled_dimplot.png&#34;, res=300, width = 3000, height = 2000, units = &#34;px&#34;)
DimPlot(New_plan, reduction = &#34;tsne&#34;)
dev.off()
</code></pre><p><img src=./unnamed-chunk-17-1.png alt></p></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f&text=GDAV%20Single%20Cell%20-%20Ex%203&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f&title=GDAV%20Single%20Cell%20-%20Ex%203" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f&name=GDAV%20Single%20Cell%20-%20Ex%203" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f&title=GDAV%20Single%20Cell%20-%20Ex%203" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f&title=GDAV%20Single%20Cell%20-%20Ex%203" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=GDAV%20Single%20Cell%20-%20Ex%203&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=GDAV%20Single%20Cell%20-%20Ex%203 - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=GDAV%20Single%20Cell%20-%20Ex%203 https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=GDAV%20Single%20Cell%20-%20Ex%203&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fgdav%2fsingle-cell-genomics%2fexercise-3%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/GDAV/Single%20Cell%20Genomics/Exercise%203/index.es.md title="Mejorar esta página" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta página</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="GDAV Single Cell - Ex 2" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>GDAV Single Cell - Ex 2</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Report on Single Cell Genomics" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Report on Single Cell Genomics</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#initialize-a-seurat-object>Initialize a Seurat object</a></li><li><a href=#pre-process-the-data>Pre-process the data</a><ul><li><a href=#perform-quality-control>Perform quality control</a></li><li><a href=#normalization>Normalization</a></li><li><a href=#feature-selecion>Feature selecion</a></li><li><a href=#scaling-of-the-data>Scaling of the data</a></li></ul></li><li><a href=#pca-analysis>PCA Analysis</a></li><li><a href=#cell-clustering>Cell clustering</a></li><li><a href=#new-parameters>New Parameters</a></li><li><a href=#finding-biomarkers-through-differential-gene-expression>Finding Biomarkers through Differential Gene Expression</a></li><li><a href=#showing-the-expression-distribution-of-these-markers>Showing the expression distribution of these markers</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegación</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre mí</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su dirección de correo electrónico, acepta recibir el boletín de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">© 2020 Pablo Ignacio Marcos López</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Apóyame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>