<!doctype html><html><head><title>Studying the SIR Model</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Studying the SIR Model"><meta property="og:description" content="A basic SIR model, such as the one that we will now proceed to study, includes three cathegories:
Susceptibles: The individuals that may be infected Infected: The individuals currently carrying the virus, which can pass it on to the next generation Recovered: Those that have been infected, have recovered, and cannot pass on the infection (this model assumes permanent inmunity) Which relate among them using the following parameters:
$\beta$: number of infected persons per sick individual per day."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-11-02T00:00:00+00:00"><meta property="article:modified_time" content="2021-11-02T00:00:00+00:00"><meta name=description content="Studying the SIR Model"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=B칰squeda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinform치tica</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biolog칤a Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biolog칤a Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estad칤stica</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentaci칩n>Presentaci칩n</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelizaci칩n</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erd칬s-R칠nyi>Erd칬s-R칠nyi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poes칤a</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atr칠vete a Cruzar">Atr칠vete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="A침o Nuevo">A침o Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisi칩n Inminente">Colisi칩n Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmali칩n">Efecto Pigmali칩n</a></li><li><a href=/es/posts/poesia/efimero/ title=Ef칤mero>Ef칤mero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 游댠 游댠">Fuego 游댠 游댠</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocres칤a>Hipocres칤a</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducci칩n>Introducci칩n</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revoluci칩n">La Revoluci칩n</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Monta침a Rusa">Monta침a Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojal치>Ojal치</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 游꿭 游꿬">Orquesta 游꿭 游꿬</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plut칩n>Plut칩n</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=P치jaro>P치jaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulaci칩n>Simulaci칩n</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sue침o>Sue침o</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentaci칩n>Tentaci칩n</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Ego칤smo>Ego칤smo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/hero.png)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Marcos</h5><p>martes, 2 de noviembre de 2021</p></div><div class=title><h1>Studying the SIR Model</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><div style=text-align:center><a href="./Studying the SIR Model.ipynb" target=_parent><img src=/posts/Imagenes/jupyter-badge.svg align=center width=20%></a></div></br><div class="cell markdown"><p>A basic SIR model, such as the one that we will now proceed to study,
includes three cathegories:</p><ul><li>Susceptibles: The individuals that may be infected</li><li>Infected: The individuals currently carrying the virus, which can
pass it on to the next generation</li><li>Recovered: Those that have been infected, have recovered, and cannot
pass on the infection (this model assumes permanent inmunity)</li></ul><p>Which relate among them using the following parameters:</p><ul><li>$\beta$: number of infected persons per sick individual per day.</li><li>$\mu$: 1/period of infection. This is the rate at which infected
individuals recover.</li></ul><p>Thus, we get the following set of equations that define the model:</p><p>$$\frac{dS}{dt} = -\beta S I \qquad \frac{dI}{dt} = \beta S I - \mu I \qquad \frac{dR}{dt} = \mu I$$</p></div><div class="cell markdown"><p>As we can see, the system does not explicitly reference time, and rather
works indirectly with the variation in the population of the different
epidemiological groups. Given that $\mu$ is defined as the time an
individual takes to recover, we can define the parameter
$\tau = \mu^{-1}$ as the average time an individual remains infected.
To set an epidemic treshold, we can use $R_{0}$, a <a href=https://pubmed.ncbi.nlm.nih.gov/25365599/>tried and tested
parameter</a> defined as:</p><p>$$R_{0} = \frac{\beta \cdot S_{0}}{\mu}$$</p></div><div class="cell markdown"><p>Here, if $R_{0}$ > 1, the infection propagates, growing
exponentially; if, however, $R_{0}$ &lt; 1, the infection progressively
deflates, ending eventually. In this model, the longer an individual
spends Infected ($\uparrow \tau$, and thus $\downarrow \mu$), the
higher the $R_{0}$, and, thus, the more contagious the infection is.
For $\beta$, the effect is the opposite: the more individuals one is
able to infect, the more contagious the disease gets.</p></div><div class="cell markdown"><p>To understand the effecfts of the initial conditions, we can use the
plane phase:</p></div><div class="cell code" data-execution_count=1><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#75715e>#Imports module</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> matplotlib.pyplot <span style=color:#66d9ef>as</span> plt
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np
</span></span></code></pre></div></div><div class="cell code" data-execution_count=2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plane_phase</span>(b, u):
</span></span><span style=display:flex><span>    S, I <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>meshgrid(np<span style=color:#f92672>.</span>arange(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>10</span>,<span style=color:#ae81ff>1</span>), np<span style=color:#f92672>.</span>arange(<span style=color:#ae81ff>0</span>,<span style=color:#ae81ff>10</span>,<span style=color:#ae81ff>1</span>))
</span></span><span style=display:flex><span>    f <span style=color:#f92672>=</span> <span style=color:#f92672>-</span>b<span style=color:#f92672>*</span>S<span style=color:#f92672>*</span>I
</span></span><span style=display:flex><span>    g <span style=color:#f92672>=</span> b<span style=color:#f92672>*</span>S<span style=color:#f92672>*</span>I<span style=color:#f92672>-</span>u<span style=color:#f92672>*</span>I
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>quiver(S,I,f,g)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Quiver graph for  = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74> &amp; u = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(b, u))
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;Susceptible&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#39;Infected&#39;</span>)
</span></span></code></pre></div></div><div class="cell markdown"><p>We can plot it for different parameter values that generate an $R_{0}$
above the epidemic value of 1:</p></div><div class="cell code" data-execution_count=3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>12</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plane_phase(<span style=color:#ae81ff>100</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plane_phase(<span style=color:#ae81ff>5</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Studying%20the%20SIR%20Model_7_0.png></figure></div></div><div class="cell markdown"><p>As we can see, despite the different set values, if $R_{0}$ > 1, the
infected population grows quickly, making the susceptible population
decrease as fast (since N = I + S + R = constant). If, however, we
select values which get us to an $R_{0}$ &lt; 1, we get an increasingly
smaller infected population, with susceptibles decreasing or even
staying the same as the epidemic extinguishes so fastly it cannot even
turn Susceptibles to Infected an Recovered at a rate big enough to show
in the graph. As we can see in the last example, values of $\beta$ = 1
and $\mu$ = 1 do not necessarily produce neither $R_{0}$ = 1 (as it
is also dependent on $S_{0}$) nor stability (as will be discussed
below).</p></div><div class="cell code" data-execution_count=4 data-scrolled=true><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>20</span>, <span style=color:#ae81ff>5</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plane_phase(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>75</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plane_phase(<span style=color:#ae81ff>0.3</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>3</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plane_phase(<span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Studying%20the%20SIR%20Model_9_0.png></figure></div></div><div class="cell markdown"><p>With regards to stability, we can mathematically analyze the system.
Since the population, N, remains constant (as the model ignores new
births and deaths), we can skip the Recovered class in order to
understand how the system behaves (See &ldquo;Mathematical Models in Biology,
by Leah Edelstein, page 247): it would be simply equivalent to saying
that all Recovered patients go outside the scope of the model, since,
assuming they are inmune,they can no longer go back to any other
compartiment, and stop being of interest to us.</p></div><div class="cell code" data-execution_count=5><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> sympy <span style=color:#66d9ef>as</span> sp
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> warnings
</span></span><span style=display:flex><span>warnings<span style=color:#f92672>.</span>filterwarnings(<span style=color:#e6db74>&#34;ignore&#34;</span>)
</span></span></code></pre></div></div><div class="cell code" data-execution_count=6><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>S, I, R, b, u <span style=color:#f92672>=</span> sp<span style=color:#f92672>.</span>symbols(<span style=color:#e6db74>&#39;S I R  洧랞&#39;</span>)
</span></span></code></pre></div></div><div class="cell code" data-execution_count=7><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>f <span style=color:#f92672>=</span> <span style=color:#f92672>-</span>b<span style=color:#f92672>*</span>S<span style=color:#f92672>*</span>I
</span></span><span style=display:flex><span>g <span style=color:#f92672>=</span> b<span style=color:#f92672>*</span>S<span style=color:#f92672>*</span>I<span style=color:#f92672>-</span>u<span style=color:#f92672>*</span>I
</span></span><span style=display:flex><span>f, g
</span></span></code></pre></div><div class="output execute_result" data-execution_count=7><pre><code>(-I*S*, I*S* - I*洧랞)
</code></pre></div></div><div class="cell code" data-execution_count=8><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>sp<span style=color:#f92672>.</span>solve([f, g], [S, I])
</span></span></code></pre></div><div class="output execute_result" data-execution_count=8><pre tabindex=0><code>[]
</code></pre></div></div><div class="cell markdown"><p>As we can see, the model has no solution, no steady states and no points
where the population of Infected and Susceptible individuals remain
unchanged. This makes sense: the model assumes a constant population N =
I + S + R, and with S only able to decrease and R only able to increase,
as new births and deaths are not accounted for. This result is
consistent with what we found when playing with the plane phase: there
are no points of stability, only inevitable derive towards extinction or
full infection and recovery</p></div><div class="cell markdown"><p>The maximum incidence, i.e. the moment in time with the most infected
individuals, can be graphically plotted as:</p></div><div class="cell code" data-execution_count=9><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>from</span> scipy.integrate <span style=color:#f92672>import</span> odeint
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> mpld3
</span></span><span style=display:flex><span>mpld3<span style=color:#f92672>.</span>enable_notebook()
</span></span></code></pre></div></div><div class="cell code" data-execution_count=10><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>deriv</span>(y, t, N, beta, u):
</span></span><span style=display:flex><span>    S, I, R <span style=color:#f92672>=</span> y
</span></span><span style=display:flex><span>    dSdt <span style=color:#f92672>=</span> <span style=color:#f92672>-</span>beta <span style=color:#f92672>*</span> S <span style=color:#f92672>*</span> I <span style=color:#f92672>/</span> N
</span></span><span style=display:flex><span>    dIdt <span style=color:#f92672>=</span> beta <span style=color:#f92672>*</span> S <span style=color:#f92672>*</span> I <span style=color:#f92672>/</span> N <span style=color:#f92672>-</span> u <span style=color:#f92672>*</span> I
</span></span><span style=display:flex><span>    dRdt <span style=color:#f92672>=</span> u <span style=color:#f92672>*</span> I
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> dSdt, dIdt, dRdt
</span></span></code></pre></div></div><div class="cell code" data-execution_count=11><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>N <span style=color:#f92672>=</span> <span style=color:#ae81ff>1000</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>999</span>, <span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>0</span>  <span style=color:#75715e># initial conditions: one infected, rest susceptible</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>t <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>, <span style=color:#ae81ff>39</span>, <span style=color:#ae81ff>40</span>) <span style=color:#75715e># Grid of time units</span>
</span></span><span style=display:flex><span>y0 <span style=color:#f92672>=</span> S0, I0, R0 <span style=color:#75715e># Initial conditions vector</span>
</span></span></code></pre></div></div><div class="cell code" data-execution_count=12><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plotsir</span>(t, S, I, R):
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(t, S, <span style=color:#e6db74>&#39;b&#39;</span>, alpha<span style=color:#f92672>=</span><span style=color:#ae81ff>0.7</span>, linewidth<span style=color:#f92672>=</span><span style=color:#ae81ff>2</span>, label<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;Susceptible&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(t, I, <span style=color:#e6db74>&#39;y&#39;</span>, alpha<span style=color:#f92672>=</span><span style=color:#ae81ff>0.7</span>, linewidth<span style=color:#f92672>=</span><span style=color:#ae81ff>2</span>, label<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;Infected&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(t, R, <span style=color:#e6db74>&#39;g&#39;</span>, alpha<span style=color:#f92672>=</span><span style=color:#ae81ff>0.7</span>, linewidth<span style=color:#f92672>=</span><span style=color:#ae81ff>2</span>, label<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;Recovered&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>legend()
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#39;Time (days)&#39;</span>)
</span></span><span style=display:flex><span>    <span style=color:#75715e>#legend = ax.legend()</span>
</span></span><span style=display:flex><span>    <span style=color:#75715e>#legend.get_frame().set_alpha(0.5)</span>
</span></span></code></pre></div></div><div class="cell code" data-execution_count=13><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure A1&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1.0</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.25</span> <span style=color:#75715e>#Infection lasts 4 time units (days)</span>
</span></span><span style=display:flex><span><span style=color:#75715e># Integrate the SIR equations over the time grid, t.</span>
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure B1&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>7</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.5</span> 
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure C1&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.4</span> 
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure D1&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.7</span>
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Studying%20the%20SIR%20Model_21_0.png></figure></div></div><div class="cell markdown"><p>In <strong>Figure A</strong>, we see that the maximum ratio of infections is reached
after about 12 days; we can modify this to fit a different, more
contagious epidemic, such as in <strong>Figure B</strong>. As we can see, in this
crazy scenario, the peak is reached really abruptly, since the infection
is incredibly contagious. The peak of infected individuals is reached
afyer only 3 days, and, after 12 days, all the population (N=1000) is
recovered (since this model does not account for deaths). With a more
mild disease, the infected curve can be less and less steep, making the
epidemic last longer, but probably overcharging the sanitary system
less: see <strong>Figure C</strong>. We can even have a disease so mild it fails to
infect all the population! In <strong>Figure D</strong>, a form of &ldquo;herd immunity&rdquo;
is reached after 30 days.</p></div><div class="cell markdown"><p>We can also use this to confirm there are no endemic states when
variating the initial conditions:</p></div><div class="cell code" data-execution_count=14><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1.0</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.25</span> <span style=color:#75715e>#Set params as in Figure A</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure A2&#39;</span>); S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>999</span>, <span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>0</span>; y0 <span style=color:#f92672>=</span> S0, I0, R0
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure B2&#39;</span>); S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>700</span>, <span style=color:#ae81ff>300</span>, <span style=color:#ae81ff>0</span>; y0 <span style=color:#f92672>=</span> S0, I0, R0
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure C2&#39;</span>); S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>500</span>, <span style=color:#ae81ff>500</span>, <span style=color:#ae81ff>0</span> ; y0 <span style=color:#f92672>=</span> S0, I0, R0
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure D2&#39;</span>); S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>, <span style=color:#ae81ff>900</span>, <span style=color:#ae81ff>0</span> ; y0 <span style=color:#f92672>=</span> S0, I0, R0
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Studying%20the%20SIR%20Model_24_0.png></figure></div></div><div class="cell markdown"><p>As can be seen, the system behaves the same, changing the initial
conditions is equivalent to moving the start of the epidemic over the
&ldquo;days&rdquo; axis</p></div><div class="cell markdown"><p>Finally, we can fix <strong>both</strong> the initial conditions <strong>and</strong> the value of
the epidemic threshold, and change $\beta$ and $\mu$, such that we
can better understand the qualitative role of such parameters. If we
fix, say, $R_{0}$ = 1.5, $S_{0}$ = 999 and $I_{0}$ = 1, all
possible combinations of $\beta$ and $\mu$ might given by
$\beta = \mu \cdot 1.5015$. Thus:</p></div><div class="cell code" data-execution_count=15><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>S0, I0, R0 <span style=color:#f92672>=</span> <span style=color:#ae81ff>999</span>, <span style=color:#ae81ff>1</span>, <span style=color:#ae81ff>0</span>
</span></span><span style=display:flex><span>y0 <span style=color:#f92672>=</span> S0, I0, R0
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure A4&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>10.5105</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>7</span> <span style=color:#75715e>#Infection lasts 4 time units (days)</span>
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure B4&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.75075</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.5</span>
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure C4&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>4.5045</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>3</span> 
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Figure D4&#39;</span>); beta <span style=color:#f92672>=</span> <span style=color:#ae81ff>150.15</span>; mu <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>
</span></span><span style=display:flex><span>ret <span style=color:#f92672>=</span> odeint(deriv, y0, t, args<span style=color:#f92672>=</span>(N, beta, mu)); S, I, R <span style=color:#f92672>=</span> ret<span style=color:#f92672>.</span>T
</span></span><span style=display:flex><span>plotsir(t,S,I,R)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./Studying%20the%20SIR%20Model_27_0.png></figure></div></div><div class="cell markdown"><p>Here, we can see that, despite keeping $R_{0}$ stable, the widely
different combinations of $\beta$ and $\mu$ produce quite different
results; after all, it is $\beta$ and $\mu$ that define the
evolution of the epidemic, with $R_{0}$ only an indirect way to know
wether the population will keep rising or not; here, since $R_{0}$ was
not that big (only 1.5) and the population was quite small (only 1000
individuals) the herd inmunity treshold is easy to reach. Nonetheless,
despite the big visual differences among the graphs, we can see that all
of them show quite similar evolutions, only with different time scales;
thus, $R_{0}$ remains an interesting estimator of the general
evolution of an epidemic.</p><p>With regards to the <strong>qualitative role of the parameters</strong>, $\beta$
represents the number of infected persons per sick individual per day,
and $\mu$ the rate at which infected individuals recover. The bigger
the first and the smallest the second, the more agressive the epidemic
is.</p></div><p><a href=https://commons.wikimedia.org/wiki/File:Simple_SIR_model.svg>Post header</a> by ARAKI Satoru at <a href=https://commons.wikimedia.org/>Wikimedia Commons</a> - Own work, <a href=https://creativecommons.org/licenses/by-sa/4.0/deed.en>CC BY-SA 4.0</a>.</p></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f&text=Studying%20the%20SIR%20Model&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f&title=Studying%20the%20SIR%20Model" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f&name=Studying%20the%20SIR%20Model" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f&title=Studying%20the%20SIR%20Model" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f&title=Studying%20the%20SIR%20Model" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Studying%20the%20SIR%20Model&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Studying%20the%20SIR%20Model - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Studying%20the%20SIR%20Model https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Studying%20the%20SIR%20Model&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fsir-model%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Modelizaci%c3%b3n%20de%20Sistemas%20Biol%c3%b3gicos/SIR%20Model/index.es.md title="Mejorar esta p치gina" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta p치gina</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Modelling the syncronization pattern of Photinus carolinus in silico using cellular automata" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>Modelling the syncronization pattern of Photinus carolinus in silico using cellular automata</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/poesia/abrazo/ title="Un abrazo" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Un abrazo</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegaci칩n</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre m칤</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su direcci칩n de correo electr칩nico, acepta recibir el bolet칤n de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">춸 2020 Pablo Ignacio Marcos L칩pez</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Ap칩yame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>