<!doctype html><html><head><title>Finding Orthologues in Arabidopsis Thaliana</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Finding Orthologues in Arabidopsis Thaliana"><meta property="og:description" content="The idea for this assignment is find the orthologue pairs between species Arabidopsis and S. pombe using BioRuby to do BLAST and parse the BLAST-generated reports. We will use use &ldquo;reciprocal best BLAST&rdquo;, scanning for Protein A of Species A in the whole proteome of Species B, and then BLAST the top hit against all proteins in Species A. If the top hits match, the protein is a good Orthologue candidate."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-12-07T00:00:00+00:00"><meta property="article:modified_time" content="2021-12-07T00:00:00+00:00"><meta name=description content="Finding Orthologues in Arabidopsis Thaliana"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=B√∫squeda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinform√°tica</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biolog√≠a Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biolog√≠a Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estad√≠stica</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentaci√≥n>Presentaci√≥n</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelizaci√≥n</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erd√∂s-R√©nyi>Erd√∂s-R√©nyi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poes√≠a</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atr√©vete a Cruzar">Atr√©vete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="A√±o Nuevo">A√±o Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisi√≥n Inminente">Colisi√≥n Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmali√≥n">Efecto Pigmali√≥n</a></li><li><a href=/es/posts/poesia/efimero/ title=Ef√≠mero>Ef√≠mero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego üî• üî•">Fuego üî• üî•</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocres√≠a>Hipocres√≠a</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducci√≥n>Introducci√≥n</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revoluci√≥n">La Revoluci√≥n</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Monta√±a Rusa">Monta√±a Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojal√°>Ojal√°</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta üéª üé∫">Orquesta üéª üé∫</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plut√≥n>Plut√≥n</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=P√°jaro>P√°jaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulaci√≥n>Simulaci√≥n</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sue√±o>Sue√±o</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentaci√≥n>Tentaci√≥n</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Ego√≠smo>Ego√≠smo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/hero.jpg)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Ignacio Marcos L√≥pez</h5><p>martes, 7 de diciembre de 2021</p></div><div class=title><h1>Finding Orthologues in Arabidopsis Thaliana</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><p>The idea for this assignment is find the orthologue pairs between species Arabidopsis and S. pombe using BioRuby to do BLAST and parse the BLAST-generated reports. We will use use &ldquo;reciprocal best BLAST&rdquo;, scanning for Protein A of Species A in the whole proteome of Species B, and then BLAST the top hit against all proteins in Species A. If the top hits match, the protein is a good Orthologue candidate.</p><p>To decide on &ldquo;sensible&rdquo; BLAST parameters, we have done a bit of online reading, and cited the papers or websites that provided the information. Also, we have writen a few sentences describing how we would continue to analyze the putative orthologues we just discovered, to prove that they really are orthologues. We WILL NOT write the code - just describe in words what that code would do.</p><div style=text-align:center><a href=https://codeberg.org/FlyingFlamingo/assignment-answers/src/branch/main/Assignment%204 target=_parent><img src=/posts/Imagenes/codeberg-badge.svg align=center width=20%></a></div><p>Documentation can be accessed using YarDoc:</p><ul><li><code>yardoc README.md blast_orthologues.rb ./Clases/Embl_to_GFF3.rb ./Clases/ImportExportModule.rb ./Clases/BlastCmdEnhanced.rb</code></li><li><code>yard server</code></li></ul><p>I have knowingly gitignored the html files from the repo to make it more legible, since they can be easily generated.</p><p>To run the script, use: <code>ruby blast_orthologues.rb ./SampleData/target_genome.fa ./SampleData/protein.fa report_file.txt</code></p><h2 id=design-choices>Design choices</h2><h3 id=blast-parameters>BLAST parameters</h3><ul><li>For the <strong>e-value</strong> (measures similarity beyond randomness), I choose 0.000001 as my cutoff, following the advice given by <em>Ward and Moreno-Hagelsieb</em> and standard practice.</li><li>This paper, <em>Ward and Moreno-Hagelsieb</em> also suggest that we add a <strong>Smith-Waterman final alignment</strong> (-F ‚Äúm S‚Äù -s T) to our BLASTP options; however, this would mean changing the code, rendering <code>BlastCmdEnhanced..makeblast_besthits()</code> useless and greatly increasing computing time, which is already quire high (around 7 hrs). So, we will just keep a soft filter (-F ‚Äúm S‚Äù), which masks low information segments only during the search phase, reducing the values of masking to increase the accuracy scores of the alignment.</li><li>In normal, command-line, bash BLAST, there are more detailed formatting options: <code>"blastp -query {0} -subject {1} -evalue {2} -outfmt '6 qseqid qseq sseqid sseq qcovs pident evalue' > {3}"</code>, which can be used to filter for <strong>coverage and identity</strong>, as well as for e-value; unfortunately, BioRuby does seems to be <strong>really bad</strong> at managing this, so I have been forced to design the following backstop solution:<ul><li>For <strong>coverage</strong>, which is the percent of the query length that is included in the aligned segments, bioruby only returns absolute values (in total nucleotides), and calls it &ldquo;overlap&rdquo; (in violation of existing naming conventions and for no reason at all), so I decided to divide it by the total length of the sequence and multiplying by 100 to get the percentage value, which PAPER A suggest should be of more than 50%. This makes sense: we dont want a sequence with a really good e-value (i.e. really good alignment) but with, for instance only 10 nucleotides known out of a total length of 1000.</li><li>For <strong>identity</strong>, <a href=http://bioruby.org/rdoc/Bio/Blast/Report/Hit.html>bioruby is supposed to provide</a> a <code>percent_identity()</code> method; however, whenever I have called it in the program it refused to work, always returning 0 when <code>identity()</code> did return non-zero values. Thus, I had to get the percentage value myself, once again by dividing by <code>.query_len</code> and multiplying by 100. According to <em>Ostlund et al</em>, this is usually higher than 60% for eukaryotes, but I decided to go with a value of 30% as <a href=https://www.researchgate.net/post/Minimum-Nucleotide-Percent-Identity-for-Homologous-Orthologous-Sequences>it seems to be more commonplace</a>; after all, if the genes are orthologues, we would expect at least some of the bases to be identical, although others might have changed due to neutral evolution and mutations.</li></ul></li><li>This large amount of conditions greatly reduces the amount of positives, from 2400 with e-value only to&mldr; actually 0!! After doing some tests, I have found that this is due mainly to identity, so I have deactivated that parameter (setting the treshold to 0). This is not that big of a deal: identity should be smaller the closer two samples are in time, but orthologues can be far, far away! With the new treshold, I get way less matches (1800), but, given the high coverage, we can be <strong>pretty sure</strong> they are good putative orthologues.</li><li>Some problems might arise if two or more FASTA sequences with non-unique headers are provided as either query or target. Fortunately, none of the files provided here exhibited this problem (and they shouldn&rsquo;t! its malpractice on the user&rsquo;s end to provide with non-standard files, let alone non-unique FASTA headers!)</li></ul><h3 id=file-input>File input</h3><p>I have decided against allowing tar.gz or any compressed files as an input; I believe file decompression is better left to the OS and the user, which I have thus forced to use fasta files.</p><h3 id=guessing-blast-type>Guessing BLAST type</h3><p>Some internet solutions suggested creating a function for automatically guessing the BLAST types. It would go along the following lines:</p><pre><code>```
def self.guess_blast_type(query, target)
    if self.guess_fastatype(target) =='dna' and self.guess_fastatype(query) =='dna'
        return 'blastn'
    elsif self.guess_fastatype(target) =='prot' and self.guess_fastatype(query) =='prot'
        return 'blastp'
    elsif self.guess_fastatype(target) =='prot' and self.guess_fastatype(query) =='dna'
        return 'blastx'
    elsif self.guess_fastatype(target) =='dna' and self.guess_fastatype(query) =='prot'
        return 'tblastn'
    elsif self.guess_fastatype(target) =='dna' and self.guess_fastatype(query) =='dna'
        return 'tblastx'
    else
        abort(&quot;[ERROR]: Something went wrong when trying to detect BLAST type&quot;)
    end
end
```
</code></pre><p>However, as you can see, there is a conflict between <strong>tblastx</strong> and <strong>blastn</strong>: they both use DNA fastas as source type, and choosing between them depends essentially on the desires of the user running the program. Thus, I have decided against including such function in my code: I could ignore <strong>tblastx</strong> and create a function only useful for this assignment, but, given that the idea, as I see it, is to create as general and as useful and reusable classes as possible, I believe it is better to ask the user to do this manually.</p><h2 id=what-to-do-next-bonus-1>What to do next? (Bonus 1%)</h2><p>To confirm whether the list of putative orthologues found in report.txt is, in fact, made up of real orthologues or not, <em>Trachana et al.</em> suggest that we use the genome of a third species, ideally related both to <em>A. Thaliana</em> and <em>S. pombe</em> (our target and query organisms) to perform a Cluster of Orthologous Genes, a clustering method based on the results of our Best Reciprocal BLASTs. By selecting only as valid those clusters of orthologues with hits for the three species, we would be a little more sure that we are dealing with genuinely orthologous genes</p><p>Another option would be to use the mentioned third specied to generate a phylogenetic tree usign alignment algorithms such as ClustalW, and infering orthology from speciation events; this is not only less computationally efficient than the clustering algorithms presented in the last paragraph, but, as <em>Bapteste et al.</em> suggest, it is not even a failproof way of showing gene-to-gene relation.</p><h2 id=bibliography>Bibliography</h2><ul><li><p>Bapteste, E., et¬†al. ¬´Do orthologous gene phylogenies really support tree-thinking?¬ª BMC Evolutionary Biology, vol. 5, mayo de 2005, p. 33. PubMed Central, <a href=https://doi.org/10.1186/1471-2148-5-33>https://doi.org/10.1186/1471-2148-5-33</a>.</p></li><li><p>Moreno-Hagelsieb, G., y K. Latimer. ¬´Choosing BLAST Options for Better Detection of Orthologs as Reciprocal Best Hits¬ª. Bioinformatics, vol. 24, n.¬∫ 3, febrero de 2008, pp. 319-24. DOI.org (Crossref), <a href=https://doi.org/10.1093/bioinformatics/btm585>https://doi.org/10.1093/bioinformatics/btm585</a>.</p></li><li><p>On the definition of sequence identity. (s. f.). Recuperado 20 de diciembre de 2021, de <a href=https://lh3.github.io/2018/11/25/on-the-definition-of-sequence-identity>https://lh3.github.io/2018/11/25/on-the-definition-of-sequence-identity</a></p></li><li><p>Ostlund, G., Schmitt, T., Forslund, K., Kostler, T., Messina, D. N., Roopra, S., Frings, O., & Sonnhammer, E. L. L. (2010). InParanoid 7: New algorithms and tools for eukaryotic orthology analysis. Nucleic Acids Research, 38(Database), D196-D203. <a href=https://doi.org/10.1093/nar/gkp931>https://doi.org/10.1093/nar/gkp931</a></p></li><li><p>Smith, E. (s.¬†f.). Library guides: Ncbi bioinformatics resources: an introduction: blast: compare & identify sequences. Recuperado 20 de diciembre de 2021, de <a href=https://guides.lib.berkeley.edu/ncbi/blast>https://guides.lib.berkeley.edu/ncbi/blast</a></p></li><li><p>Trachana, Kalliopi, et¬†al. ¬´Orthology prediction methods: A quality assessment using curated protein families¬ª. Bioessays, vol. 33, n.¬∫ 10, octubre de 2011, pp. 769-80. PubMed Central, <a href=https://doi.org/10.1002/bies.201100062>https://doi.org/10.1002/bies.201100062</a>.</p></li><li><p>Ward, Natalie, y Gabriel Moreno-Hagelsieb. ¬´Quickly Finding Orthologs as Reciprocal Best Hits with BLAT, LAST, and UBLAST: How Much Do We Miss?¬ª PLOS ONE, vol. 9, n.¬∫ 7, julio de 2014, p. e101850. PLoS Journals, <a href=https://doi.org/10.1371/journal.pone.0101850>https://doi.org/10.1371/journal.pone.0101850</a>.</p></li><li><p>Yuan, Y. P., et¬†al. ¬´Towards Detection of Orthologues in Sequence Databases¬ª. Bioinformatics (Oxford, England), vol. 14, n.¬∫ 3, 1998, pp. 285-89. PubMed, <a href=https://doi.org/10.1093/bioinformatics/14.3.285>https://doi.org/10.1093/bioinformatics/14.3.285</a>.</p></li><li><p>Zhou, Y., y L. F. Landweber. ¬´BLASTO: A Tool for Searching Orthologous Groups¬ª. Nucleic Acids Research, vol. 35, n.¬∫ Web Server, mayo de 2007, pp. W678-82. DOI.org (Crossref), <a href=https://doi.org/10.1093/nar/gkm278>https://doi.org/10.1093/nar/gkm278</a>.</p></li></ul><h2 id=code-and-acknowledgements>Code and Acknowledgements</h2><p>Source code for the assignment can be found <a href=https://codeberg.org/FlyingFlamingo/assignment-answers/src/branch/main/Assignment%204>here</a></p><p>The header image for this post is <a href=http://creativecommons.org/licenses/by-sa/3.0/>CC-By-Sa 3.0</a> by <a href=//commons.wikimedia.org/wiki/User:Alberto_Salguero>Alberto Salguero</a> on <a href=https://commons.wikimedia.org/wiki/File:Arabidopsis_thaliana_hojas_basales.jpg#/media/Archivo:Arabidopsis_thaliana_hojas_basales.jpg>Wikimedia Commons</a></p><hr><p>This document can easily be converted to PDF format using pandoc:</p><p><code>pandoc --pdf-engine=xelatex README.md -o README.pdf</code></p></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f&text=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f&title=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f&name=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f&title=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f&title=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Finding%20Orthologues%20in%20Arabidopsis%20Thaliana&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fbioinfo-programming-challenges%2fassignment-4%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/BioInfo%20Programming%20Challenges/Assignment%204/index.es.md title="Mejorar esta p√°gina" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta p√°gina</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="GFF feature files and visualization" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>GFF feature files and visualization</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="SPARQL Queries" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>SPARQL Queries</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#design-choices>Design choices</a><ul><li><a href=#blast-parameters>BLAST parameters</a></li><li><a href=#file-input>File input</a></li><li><a href=#guessing-blast-type>Guessing BLAST type</a></li></ul></li><li><a href=#what-to-do-next-bonus-1>What to do next? (Bonus 1%)</a></li><li><a href=#bibliography>Bibliography</a></li><li><a href=#code-and-acknowledgements>Code and Acknowledgements</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegaci√≥n</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre m√≠</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su direcci√≥n de correo electr√≥nico, acepta recibir el bolet√≠n de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">¬© 2020 Pablo Ignacio Marcos L√≥pez</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Ap√≥yame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>