<!doctype html><html><head><title>Practical Assignment 3</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Practical Assignment 3"><meta property="og:description" content="by Yaiza ARNAIZ ALCACER (251), Pablo MARCOS LOPEZ (269), Alexandre VERGNAUD (178) and Lexane LOUIS (179)
Introduction Advanced statistics, through methods such as Principal Component Analysis, Factor Analysis and Correspondence Analysis, allows us to find the most important variables in a dataset, facilitating its understanding and the making of decisions and conclusions. Throughout the following 3 exercises, we will demonstrate the use of these three methods in different fields, showing their usefulness and applications."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2022-01-17T00:00:00+00:00"><meta property="article:modified_time" content="2022-01-17T00:00:00+00:00"><meta name=description content="Practical Assignment 3"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=Búsqueda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinformática</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biología Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biología Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estadística</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentación>Presentación</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelización</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erdös-Rényi>Erdös-Rényi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poesía</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atrévete a Cruzar">Atrévete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="Año Nuevo">Año Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisión Inminente">Colisión Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmalión">Efecto Pigmalión</a></li><li><a href=/es/posts/poesia/efimero/ title=Efímero>Efímero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego 🔥 🔥">Fuego 🔥 🔥</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocresía>Hipocresía</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducción>Introducción</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revolución">La Revolución</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Montaña Rusa">Montaña Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojalá>Ojalá</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta 🎻 🎺">Orquesta 🎻 🎺</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plutón>Plutón</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=Pájaro>Pájaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulación>Simulación</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sueño>Sueño</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentación>Tentación</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Egoísmo>Egoísmo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/hero.jpg)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Ignacio Marcos López</h5><p>lunes, 17 de enero de 2022</p></div><div class=title><h1>Practical Assignment 3</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><p><em>by Yaiza ARNAIZ ALCACER (251), Pablo MARCOS LOPEZ (269), Alexandre VERGNAUD (178) and Lexane LOUIS (179)</em></p><div style=text-align:center><a href=https://colab.research.google.com/drive/1mY1a1nMiMhO8qyW3vitl6votLYQH-A_Q target=_parent><img src=https://colab.research.google.com/assets/colab-badge.svg alt="Open In Colab" align=center width=15%></a></div></br><h2 id=introduction>Introduction</h2><p>Advanced statistics, through methods such as Principal Component Analysis, Factor Analysis and Correspondence Analysis, allows us to find the most important variables in a dataset, facilitating its understanding and the making of decisions and conclusions. Throughout the following 3 exercises, we will demonstrate the use of these three methods in different fields, showing their usefulness and applications.</p><h2 id=exercise-1>Exercise 1</h2><p>In this first exercise, our aim is to analyze the different species of phytoplankton algae present in lakes of different regions at two different times each year. Using PCA and descriptive statistics, we want to find out if the different parameters present in our data are good predictors species type distribution over time, and, if possible, to determine which of these components are the most important in this distribution.</p><p>To do this, the first step is to detect whether the different species are correlated; and, if so, to see how significant said correlation is. To do this, we created Table 1, which shows the correlation matrix between the different species using the p-value of for the pearson correlation quotient adjusted by the Holm-Bonferroni method, which corrects the family-wise error rate (the probability of finding false positives) in multiple hypothesis tests; helping us discern only trully correlated variables. Estimating an acceptable p-value of &lt; 0.05, as is usual in statistics, we obtain a correlation, in decreasing order of significance, between the following species: Cianophyceae and Chlorocococales, Zygophyceae and Chlorococales, Euglenophyceae and Cianophyceae; and Euglenophyceae and Bacillariophyceae.</p><table><thead><tr><th style=text-align:center></th><th style=text-align:center>Bacillariophyceae</th><th style=text-align:center>Chlorococales</th><th style=text-align:center>Cianophyceae</th><th style=text-align:center>Euglenophyceae</th><th style=text-align:center>Zygophyceae</th></tr></thead><tbody><tr><td style=text-align:center>Bacillariophyceae</td><td style=text-align:center></td><td style=text-align:center>1.0000</td><td style=text-align:center>1.0000</td><td style=text-align:center>0.0049</td><td style=text-align:center>0.0853</td></tr><tr><td style=text-align:center>Chlorococales</td><td style=text-align:center>1.0000</td><td style=text-align:center></td><td style=text-align:center>&lt;.0001</td><td style=text-align:center>0.7839</td><td style=text-align:center>0.0269</td></tr><tr><td style=text-align:center>Cianophyceae</td><td style=text-align:center>1.0000</td><td style=text-align:center>&lt;.0001</td><td style=text-align:center></td><td style=text-align:center>0.0343</td><td style=text-align:center>1.0000</td></tr><tr><td style=text-align:center>Euglenophyceae</td><td style=text-align:center>0.0049</td><td style=text-align:center>0.7839</td><td style=text-align:center>0.0343</td><td style=text-align:center></td><td style=text-align:center>1.0000</td></tr><tr><td style=text-align:center>Zygophyceae</td><td style=text-align:center>0.0853</td><td style=text-align:center>0.0269</td><td style=text-align:center>1.0000</td><td style=text-align:center>1.0000</td><td style=text-align:center></td></tr></tbody></table><p style=text-align:center>Matrix of Pearson correlations showing the pairwise two-sided p-values adjusted by Holm's method. As it is shown, the correlation is significant for the following pairs: Euglenophyceae and Bacillariophyceae, Euglenophyceae and Cianophyceae and Zygophyceae and Chlorococales; and highly significant for the pair Cianophyceae and Chlorococales.</p><p>Now that we are sure about possible correlations, we would like to know about the dimensionality of our data, and, therefore, its complexity. Thus, we performed a Principal Component Analysis using the covariance matrix of our different types of bacteria. In Table 2, we can see that, indeed, 71.6% of the variability in our data can be explained by looking at just 2 components. Now, what are these 2 components? One possible theory is that they are, for example, the 2 annual periods in which we have sampled the data. Therefore, we will work only with these components, trying to clarify whether their relevance is indeed due to the fact that they represent the different periods of the year or not.</p><table><thead><tr><th style=text-align:center></th><th style=text-align:center>Component 1</th><th style=text-align:center>Component 2</th><th style=text-align:center>Component 3</th><th style=text-align:center>Component 4</th><th style=text-align:center>Component 5</th></tr></thead><tbody><tr><td style=text-align:center>Proportion of Variance</td><td style=text-align:center>0.4903891</td><td style=text-align:center>0.2258744</td><td style=text-align:center>0.1574135</td><td style=text-align:center>0.08693947</td><td style=text-align:center>0.03938336</td></tr></tbody></table><p style=text-align:center>Proportion of Variance explainaible by each component after PCA of the data.</p><p>Another way to approach this problem is through a sedimentation plot. These figures, which represent the eigenvalues, usually show a "bend", a bend between the most important components and those that are not so interesting. Such a bend is, of course, a relative thing, as there is no "set in stone" definition;<sup id=fnref:1><a href=#fn:1 class=footnote-ref role=doc-noteref>1</a></sup> but, in Figure 1, we would argue that it is precisely the first two components (especially the first, although the second is more subjective) that are above the curve. We therefore consider our selection validated.</p><div style=text-align:center><img src=./exercise1_1.png alt=sometext align=center width=30%><p>Figure 1. The PCA barplot (principal component analysis) of the different components, shows how the 2 first PCs (principal components) explain the 71.6% of the variability of the data.</p></div><p>We can also assess which are the most interesting variables for each component using a loading plot, which interprets the values of the coefficients (loadings), so that those variables furthest away from the origin of coordinates and the centre of the graph are those that will provide more information for each component. In Figure <a href=#first_second>[first_second]</a>{reference-type=&ldquo;ref&rdquo; reference=&ldquo;first_second&rdquo;}, we see that the most important variables for the first component selected are Chlorococales and Euglenophycea, while for the second, these would be Bacilloriaphyceae and Chlorocales.</p><div style=text-align:center><img src=./exercise1_2.png alt=sometext align=center width=30%><p>Figure 1. The PCA plot of the loadings shows the variables more related to: A) The first PC: Chlorococales, Cianophyceae and Euglenophyceae; B) The second PC: Bacilloriaphyceae and Chlorocales.</p></div><p>Finally, we can use various scatter plots to represent the selected components across regions, lakes and time-spans. In Figure 3 a), we can see that Component 1 of the PCA explains the difference between two biomes: on the one hand, the Andean and Caribbean regions; and, on the other hand, the Amazon region. In subfigure c) we see that a clear differentiation for PC2, which allows us to separate the two time periods (high and low tides), as we had previously predicted. For subgraph b), the interpretation is more complicated: it seems that the separations between lakes Iguaque and Tota, Purisima, Momil and Fuquequeque are explained by PC2, while the difference between lakes Purisima and Tota and lake Tarapoto is explained by PC1. This, on the one hand, agrees with our initial hypothesis, since PC1 and PC2 allow us to distinguish the time of sampling; but, on the other hand, these components allow us to explain even more differences: after all, as we have explained, they account for 71% of the total variance.</p><div class=row><div class="col col-sm-12 col-lg-4"><img src=./exercise1_3.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./exercise1_4.png alt></div><div class="col col-sm-12 col-lg-4"><img src=./exercise1_5.png alt></div></div><h2 id=exercise-2>Exercise 2</h2><p>For this second exercise, we would like to analyze the limnological properties of several neotropical lakes in order to find out their productivity degree and adequacy for life, which, a priori, seem like they might be related to values such as nutrient concentration, depth, pH or amount of dissolved oxygen. We therefore seek to use a multifactor analysis to check if we can find a suitable model to explain the data.
 </p><p>To do this, we will, again, see if the variables correlate with each other. Thus, we made a new table, Table 3, of the p-values of the correlation coefficients fitted by Holm&rsquo;s method. Although the correlations are too numerous to detail in writing in this report, the correlation values between the variables are, in general, rather poor, which is to be expected: it is unlikely, for example, that the amount of a certain gas in the lake influences the others, beyond certain isolated cases. <sup id=fnref:2><a href=#fn:2 class=footnote-ref role=doc-noteref>2</a></sup></p><div style=text-align:center><img src=./Table2.png alt=sometext align=center width=70%></div><p>To understand how these variables influence the productivity degree and the adequacy for life, we carried out a Factor Analysis, which works in the opposite way to PCA: instead of seeking to create "new components", as if it were the greatest common divisor, it seeks to find a least common multiple, a subset of factors common to all the data which are implicit, neither observable nor directly measurable. In this way, we hope to see whether the model we have devised, in which different lake concentration variables are related to fitness to support life, fits reality or not. We would expect concentration of nutrients and dissolved organic carbon to be correlated with productivity degree, with the "chemicals" determining if the lake is adequate for life or not.</p><p>One of the ways we can find out whether or not the system does indeed fit the two-factor model we have chosen is to use the Kaiser-Meyer-Olkin, or sample adequacy, index. In our case, this takes a value of 0.66, which means that the model, while not unacceptable, is only an approximate fit to the data.</p><p>Next, we can analyse the results of trying this model, which a priori we have seen is not very good, on our data. To do this, we have two parameters: the loadings, which are the values of the fit coefficients; and the communalities, the sum of the squares of these loadings. According to the result of our analysis, which can be consulted in detail in <em>Annex I - Code</em>, there are two variables with low uniqueness: Conductivity (0.005) and Temperature (0.084), which will therefore have high communality, and respond satisfactorily to our model. However, the other nine variables have high uniqueness, especially pH (87.6%), NH4 (74.4%), Depth (74.2%), NO3 (72.4%) and PO4 (59.4%) which means that they will not be adequately explained by the factorial model because of their low uniqueness.</p><p>About the loadings, we found that they can help us correlate Factor 1 with adequacy, and Factor 2 with productivity. However, pH doesn&rsquo;t seem to correlate well with Factor 1, and the carbon, NO2, P04, Si02 seem to be more linked with the Factor 2 than with the second. This does not fit our previous hypothesis, and makes the model even more messy and difficult to understand.</p><p>With regards to the percentage of variability explained by our factors, we found that, collectively, Factors 1 and 2 only account for 55.9% of our model&rsquo;s cumulative variability, which is too little to be acceptable. What is more, the p-value of the contrast is approximately zero (1.68e-03), which forces us to reject the hypthesis that two factors are enough, and, thus, that only the productivity degree and the habitat&rsquo;s adequacy for life are correlated with the provided variables.</p><p>Finally, as we did in the previous exercise, we will build and interpret the loading plot, availaible in Figure [[second_first]]{#second_first label=&ldquo;second_first&rdquo;}. There, we can see that Oxygen, Depth, PO4, Carbon, Conductivity, SiO2, NO2 and Temperature are mostly correlated to Factor 1, whereas Conductivity, Carbon, NO3 and NH4 are mostly correlated to Factor 2. These correlations do not match our pre-assumptions that concentration of nutrients and dissolved organic carbon would be mostly correlated with productivity degree, with the "chemical compounds data" being mostly connected with habitat adequacy for life, and instead mix together in strange ways, which suggests that the model is neither good not coherent for this kind of analysis</p><div style=text-align:center><img src=./exercise2.png alt=sometext align=center width=30%><p>Figure 1. The Factor Analysis plot of the loadings shows that the variables more related to Factor 1 are Oxygen, Depth, PO4, Carbon, Conductivity, SiO2, NO2 and Temperature; whereas Factor 2 is mostly related to Conductivity, Carbon, NO3 and NH4.</p></div><p>In any case, and as we have seen, a more complete model would be required, perhaps with three factors or more. This conclusion is consistent with the previous observation that nine of the 11 variables are not sufficiently explained by the two factors due to their low communality.</p><h2 id=exercise-3>Exercise 3</h2><p>Finally, we will use cars.xls, a database of random samples of car brands in different European countries, with the aim of discovering whether or not there is an association between the different brands and countries. For this, we will use Correspondence Analysis, a dimension reduction technique that allows users to visualize a cloud of multidimensional points in two dimensions respecting the relative positions of the points in the original point cloud.</p><p>To do so, the first step is to perform the Correspondent Analysis analysis and its factor map, which shows the points distributed in the different dimensions. Thus, in Figure [[third_first]]{#third_first label=&ldquo;third_first&rdquo;}, we can see how brands A and B seem to be clustered quite close to each other, together with most of the Central European countries; while brands C and D are the most distant from the distribution, with C appearing mostly in Scadinavia and Russia (perhaps Volvo) and D in Ireland and the UK, as might be the case for Rolls-Roice. For B, it seems to be mostly associated with Central European countries, such as Germany or Czech Republic, while A is closer to Spain and France.</p><div style=text-align:center><img src=./exercise3.png alt=sometext align=center width=50%><p>Figure 1. Correspondent analysis factor map. The plot shows the associations between the brands A and B, while the huge different between the brands D and C. Regarding the countries, Portugal, Spain, Italy, Greece, Serbia and France are associated with the brand A; Holland, Germany, Switzerland, Belgium, Czech Republic and Hungary are distributed between A and B; Sweden, Norway, Poland and Finland are mainly distributed between B and C; finally, UK and Ireland are associated to brand D.</p></div><p>The next step, to understand whether the model is valid or not, is to analyse the variances of the axes, the absolute and relative contributions, and the inertia:</p><ul><li><strong>Variances of the axes</strong>: As seen on the graph, we have close coordinates between Portugal, Spain, Italy, Greece, Serbia and France especially along the dimension 1, which might mean that their auto industries are closely intertwined together. UK and Ireland, two island nations partially decoupled from the EU&rsquo;s single market and Schengen Area, are far from the others.</li><li><strong>Absolute contributions</strong>: Dimension 1 is most closely related to Finland, Norway, Uk, Ireland and Russia. Dimension 2 explains UK and Ireland, and Dimension 3 works for mostly Greece, the Czeck Republic, Holland and Belgium.</li><li><strong>Relative contributions</strong>: Dimension 1 is good at explaining variations in car buying in Finland, Sweden, Norway, Russia and Poland with a relative value of more than 90%. This increases for Dimension 2, which explains Hungary with a 95% value, although this only reached 70% for the UK and Ireland. Dimension 3 is good at explaining variance in Holland and Belgium: around 80%.</li><li><strong>Inertia</strong>: The most important countries in the CA analysis are Russia (20,6%), UK (17,6%) and Ireland (18,4%). Overall, these values make a lot of sense, and are consistent with what was obtained in Figure [\</li></ul><p>Brands A and B are thus much more closely related to each other and to Dimension 3, with Dimension 2 mostly related to Brand D and Dimension 1 related to Brand C. Moreover, the relative correlation values are very high, always above 50% (see <em>Annex I - Code</em>). Moreover, Dimensions 1 and 2 seem to explain up to 87,5% percent of the variance on our data (see Table 4), being the most important ones, with Dimension 3 only explaining 12,5% of the variance. This makes sense: automotive industries are generally highly correlated with the countries of origin of their parent groups, especially in Europe, one of the world&rsquo;s largest automotive production regions. <sup id=fnref:3><a href=#fn:3 class=footnote-ref role=doc-noteref>3</a></sup></p><div style=text-align:center><table><thead><tr><th style=text-align:center></th><th style=text-align:center>Eigenvalue</th><th style=text-align:center>% of Variance</th><th style=text-align:center>Cumulative % of Variance</th></tr></thead><tbody><tr><td style=text-align:center>Dimension 1</td><td style=text-align:center>0.493421000637901</td><td style=text-align:center>48.6932328057494</td><td style=text-align:center>48.6932328057494</td></tr><tr><td style=text-align:center>Dimension 2</td><td style=text-align:center>0.393197841420016</td><td style=text-align:center>38.802714124917</td><td style=text-align:center>87.4959469306664</td></tr><tr><td style=text-align:center>Dimension 3</td><td style=text-align:center>0.126706772625117</td><td style=text-align:center>12.5040530693336</td><td style=text-align:center>100</td></tr></tbody></table><p>Eigenvalues and variance of each dimension. As shown in the table, the dimension 1 and 2 explain the 87,5% percent of the variance while the dimension 3 only explain the 12,5%.</p></div><h2 id=conclusions>Conclusions</h2><p>Using Principal Component Analysis, Factor Analysis and Correspondence Analysis, we have been able to solve a number of statistical problems in fields as far apart as automobiles, biology and the emergence of life in lakes. Although it has not always produced perfect results, we believe that the use of these models has, in general, been very useful, and has allowed us to solve the problems we faced in a creative and satisfactory way.</p><h2 id=bibliography>Bibliography</h2><p>This document, and the accompanying code, is availaible under the <a href=https://creativecommons.org/licenses/by-sa/4.0/>CC By SA 4.0</a> License</p><div class=footnotes role=doc-endnotes><hr><ol><li id=fn:1><p>S. E. Jørgensen. Application of models in limnological research. SIL Proceedings, 1922-2010, 24(1) :61–67, Dec. 1990.&#160;<a href=#fnref:1 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:2><p>L. C. Kristiansen, S. Jacobsen, F. Jessen, and B. M. Jørgensen. Using a cross-model loadings plot to identify protein spots causing 2-DE gels to become outliers in PCA. PROTEOMICS, 10(8) :1721–1723, 2010.&#160;<a href=#fnref:2 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li><li id=fn:3><p>S. Radosevic and A. Rozeik. Foreign Direct Investment and Restructuring in the Automotive Industry in Central and East Europe, Mar. 2005.&#160;<a href=#fnref:3 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li></ol></div></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f&text=Practical%20Assignment%203&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f&title=Practical%20Assignment%203" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f&name=Practical%20Assignment%203" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f&title=Practical%20Assignment%203" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f&title=Practical%20Assignment%203" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Practical%20Assignment%203&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Practical%20Assignment%203 - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Practical%20Assignment%203 https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Practical%20Assignment%203&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2festad%25C3%25ADstica%2fassignment-3%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Estad%c3%adstica/Assignment%203/index.es.md title="Mejorar esta página" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta página</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Practical Assignment 2" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>Practical Assignment 2</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>Notas de Clase</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#introduction>Introduction</a></li><li><a href=#exercise-1>Exercise 1</a></li><li><a href=#exercise-2>Exercise 2</a></li><li><a href=#exercise-3>Exercise 3</a></li><li><a href=#conclusions>Conclusions</a></li><li><a href=#bibliography>Bibliography</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegación</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre mí</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su dirección de correo electrónico, acepta recibir el boletín de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">© 2020 Pablo Ignacio Marcos López</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Apóyame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>