<!doctype html><html><head><title>Analysis of the efects of Lockdowns on CoViD</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href=/google-fonts/Mulish/mulish.css><link rel=stylesheet href=/fontawesome/css/all.min.css><link rel=stylesheet href=/css/colortheme/colortheme.css><link rel=icon type=image/png href=/images/new-favicon_hucdd21eecb8229141dd7886643601b174_45610_42x0_resize_box_3.png><meta property="og:title" content="Analysis of the efects of Lockdowns on CoViD"><meta property="og:description" content="During the COVID-19 outbreak in spring 2020, preventive confinement at day D for susceptible individuals was implemented in several countries to ‚Äúbend‚Äù the infection curve and control the spread of the virus. Let us assume that confined individuals are effectively protected from infection. A ‚Äúbending‚Äù or flattening of the (log-scale) curve representing the number of active cases was widely publicized as a sign that the epidemics was under control."><meta property="og:type" content="article"><meta property="og:url" content="https://www.pablomarcos.me/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-11-03T00:00:00+00:00"><meta property="article:modified_time" content="2021-11-03T00:00:00+00:00"><meta name=description content="Analysis of the efects of Lockdowns on CoViD"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css><script src=https://cdn.counter.dev/script.js data-id=counter-analytics@loreak.org data-utcoffset=1></script>
<script data-goatcounter=https://n9zp7.goatcounter.com/count async src=/js/goat-counter.js></script></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/es><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png alt=Logo>
Pablo Marcos' Blog</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"><li class="nav-item dropdown"><div id=theme-initialization style=display:none default-theme=system></div><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# onclick=enableLightTheme()><img class=menu-icon-center src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=enableDarkTheme()><img class=menu-icon-center src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a>
<a class="dropdown-item nav-link" href=# onclick=useSystemTheme()><img class=menu-icon-center src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/main-logo_hu47450349114826056e2be89059cb78d7_20725_42x0_resize_box_3.png class=d-none id=main-logo alt=Logo>
<img src=/images/inverted-logo_hu54cab9b231ca4636ef2fd30ef75af799_11321_42x0_resize_box_3.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/es/search><input type=text name=keyword placeholder=B√∫squeda data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/es/posts/ data-filter=all>Posts</a></li><div class=subtree><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/>Concursos</a><ul><li><a href=/es/posts/concursos/centrodelenguas/ title="Centro de Lenguas 2020">Centro de Lenguas 2020</a></li><li><a href=/es/posts/concursos/vinoyaceite2020/ title="Etiquetas ETSIAAB 2020">Etiquetas ETSIAAB 2020</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/concursos/season-of-kde-2021/>Season of KDE 2021</a><ul><li><a href=/es/posts/concursos/season-of-kde-2021/sok-final-status-report/ title="Informe Final">Informe Final</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-april/ title="SoK 2021 Abril">SoK 2021 Abril</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-january/ title="SoK 2021 Enero">SoK 2021 Enero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-february/ title="SoK 2021 Febrero">SoK 2021 Febrero</a></li><li><a href=/es/posts/concursos/season-of-kde-2021/sok-report-march/ title="SoK 2021 Marzo">SoK 2021 Marzo</a></li></ul></li><li><a href=/es/posts/concursos/season-of-kde-2022/ title="Season of KDE 2022">Season of KDE 2022</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/>Master en Bioinform√°tica</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/readme/ title=README>README</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/>Big Data</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-dog-food-spoiling/ title="Dog Food Spoiling">Dog Food Spoiling</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/dog-food-day/ title="Extra Point">Extra Point</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/hack-data/ title="Hack Data">Hack Data</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-churn-risk/ title="Predicting Churn Risk">Predicting Churn Risk</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/predicting-crew-members/ title="Predicting Crew Members">Predicting Crew Members</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/walmart-stock-exercise/ title="Walmart Stock Exercise">Walmart Stock Exercise</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/>Final Project</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/big-data/final-project/paper/ title=ReadMe>ReadMe</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/>Biolog√≠a Evolutiva</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-evolutiva/exercise-2/ title="Exercise 2">Exercise 2</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/>Biolog√≠a Programable</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/essay/ title=Essay>Essay</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/biolog%C3%ADa-programable/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/>Estad√≠stica</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/estad%C3%ADstica/lecture-notes/ title="Notas de Clase">Notas de Clase</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/>GAB</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gab/warning/ title=WARNING>WARNING</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/>GDAV</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/final-project/ title="Final Project">Final Project</a></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/>Single Cell Genomics</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-1/ title="Exercise 1">Exercise 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-2/ title="Exercise 2">Exercise 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/exercise-3/ title="Exercise 3">Exercise 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/gdav/single-cell-genomics/final-report/ title="Final Report">Final Report</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/>Knowledge Representation</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/paper/ title=Paper>Paper</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/knowledge-representation/presentation/ title=Presentation>Presentation</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/>Lead Discovery</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/annex-i-code/ title="Annex I - Code">Annex I - Code</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/lead-discovery/assignment-3/ title="Assignment 3">Assignment 3</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/>Retos en Ruby</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-1/ title="Assignment 1">Assignment 1</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-2/ title="Assignment 2">Assignment 2</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-3/ title="Assignment 3">Assignment 3</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-4/ title="Assignment 4">Assignment 4</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/bioinfo-programming-challenges/assignment-5/ title="Assignment 5">Assignment 5</a></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/>Trabajo Fin de Master</a><ul><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-documentation/ title="CanGraph Documentation">CanGraph Documentation</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/cangraph-update/ title="CanGraph Update">CanGraph Update</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/ecsa-day-2022/ title="ECSA Day 2022">ECSA Day 2022</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/presentation/ title=Presentaci√≥n>Presentaci√≥n</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/tfm/thesis/ title=Tesis>Tesis</a></li></ul></li><li><i class="fas fa-minus-circle"></i><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/>Modelizaci√≥n</a><ul class=active><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="Cellular Automata">Cellular Automata</a></li><li><a class=active href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/ title="CoViD and Lockdowns">CoViD and Lockdowns</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title=Erd√∂s-R√©nyi>Erd√∂s-R√©nyi</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/essay-on-free-will/ title="Essay on Free Will">Essay on Free Will</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/kuramoto-oscilators/ title="Kuramoto Oscilators">Kuramoto Oscilators</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/logistic-map/ title="Logistic Map">Logistic Map</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/beyond-erdos-renyi/ title="Preferential Attachment">Preferential Attachment</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/synchronization-of-fireflies/ title="Synchronization of Fireflies">Synchronization of Fireflies</a></li><li><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/sir-model/ title="The SIR Model">The SIR Model</a></li></ul></li></ul></li><li><i class="fas fa-plus-circle"></i><a href=/es/posts/poesia/>Poes√≠a</a><ul><li><a href=/es/posts/poesia/abrazo/ title=Abrazo>Abrazo</a></li><li><a href=/es/posts/poesia/invierno/ title="Ahora soy Invierno">Ahora soy Invierno</a></li><li><a href=/es/posts/poesia/amor/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/despedida/ title=Amor>Amor</a></li><li><a href=/es/posts/poesia/simetria/ title=Asimetrico>Asimetrico</a></li><li><a href=/es/posts/poesia/paso-de-cebra/ title="Atr√©vete a Cruzar">Atr√©vete a Cruzar</a></li><li><a href=/es/posts/poesia/a%C3%B1o-nuevo/ title="A√±o Nuevo">A√±o Nuevo</a></li><li><a href=/es/posts/poesia/cerezo/ title=Cerezo>Cerezo</a></li><li><a href=/es/posts/poesia/locos/ title="Ciegos y Locos">Ciegos y Locos</a></li><li><a href=/es/posts/poesia/ciudad/ title=Ciudad>Ciudad</a></li><li><a href=/es/posts/poesia/coincidencia/ title=Coincidencia>Coincidencia</a></li><li><a href=/es/posts/poesia/peligro/ title="Colisi√≥n Inminente">Colisi√≥n Inminente</a></li><li><a href=/es/posts/poesia/courage/ title=Coraje>Coraje</a></li><li><a href=/es/posts/poesia/david-revoy/ title="Cute Cats">Cute Cats</a></li><li><a href=/es/posts/poesia/eclipse/ title=Eclipse>Eclipse</a></li><li><a href=/es/posts/poesia/efecto-pigmalion/ title="Efecto Pigmali√≥n">Efecto Pigmali√≥n</a></li><li><a href=/es/posts/poesia/efimero/ title=Ef√≠mero>Ef√≠mero</a></li><li><a href=/es/posts/poesia/espinas/ title=Espinas>Espinas</a></li><li><a href=/es/posts/poesia/estro/ title=Estro>Estro</a></li><li><a href=/es/posts/poesia/fuego/ title="Fuego üî• üî•">Fuego üî• üî•</a></li><li><a href=/es/posts/poesia/fugaz/ title=Fugaz>Fugaz</a></li><li><a href=/es/posts/poesia/ginko/ title=Gingko>Gingko</a></li><li><a href=/es/posts/poesia/granada/ title=Granada>Granada</a></li><li><a href=/es/posts/poesia/guayaba/ title=Guayaba>Guayaba</a></li><li><a href=/es/posts/poesia/hechizo/ title=Hechizo>Hechizo</a></li><li><a href=/es/posts/poesia/hipocresia/ title=Hipocres√≠a>Hipocres√≠a</a></li><li><a href=/es/posts/poesia/horizontes/ title=Horizontes>Horizontes</a></li><li><a href=/es/posts/poesia/if-ants-rose-up/ title="If ants rose up">If ants rose up</a></li><li><a href=/es/posts/poesia/incertidumbre/ title=Incertidumbre>Incertidumbre</a></li><li><a href=/es/posts/poesia/introduccion/ title=Introducci√≥n>Introducci√≥n</a></li><li><a href=/es/posts/poesia/hermanos/ title="La Revoluci√≥n">La Revoluci√≥n</a></li><li><a href=/es/posts/poesia/metamorfosis/ title=Metamorfosis>Metamorfosis</a></li><li><a href=/es/posts/poesia/miedo/ title=Miedo>Miedo</a></li><li><a href=/es/posts/poesia/monta%C3%B1a-rusa/ title="Monta√±a Rusa">Monta√±a Rusa</a></li><li><a href=/es/posts/poesia/narciso/ title=Narciso>Narciso</a></li><li><a href=/es/posts/poesia/ojala/ title=Ojal√°>Ojal√°</a></li><li><a href=/es/posts/poesia/erase/ title="Once Upon a Time">Once Upon a Time</a></li><li><a href=/es/posts/poesia/orquesta/ title="Orquesta üéª üé∫">Orquesta üéª üé∫</a></li><li><a href=/es/posts/poesia/papel/ title=Papel>Papel</a></li><li><a href=/es/posts/poesia/philanthropy/ title=Philantropy>Philantropy</a></li><li><a href=/es/posts/poesia/pluma/ title=Plumas>Plumas</a></li><li><a href=/es/posts/poesia/pluton/ title=Plut√≥n>Plut√≥n</a></li><li><a href=/es/posts/poesia/mitologia/ title="Por fin, la Primavera">Por fin, la Primavera</a></li><li><a href=/es/posts/poesia/precipicio/ title=Precipicio>Precipicio</a></li><li><a href=/es/posts/poesia/pajaro/ title=P√°jaro>P√°jaro</a></li><li><a href=/es/posts/poesia/rarezas/ title=Rarezas>Rarezas</a></li><li><a href=/es/posts/poesia/reloj/ title=Reloj>Reloj</a></li><li><a href=/es/posts/poesia/renacer/ title=Renacer>Renacer</a></li><li><a href=/es/posts/poesia/rubato/ title=Rubato>Rubato</a></li><li><a href=/es/posts/poesia/simulaci%C3%B3n/ title=Simulaci√≥n>Simulaci√≥n</a></li><li><a href=/es/posts/poesia/soledad/ title=Soledad>Soledad</a></li><li><a href=/es/posts/poesia/sue%C3%B1o/ title=Sue√±o>Sue√±o</a></li><li><a href=/es/posts/poesia/sufragettes/ title=Sufragettes>Sufragettes</a></li><li><a href=/es/posts/poesia/tarde/ title=Tarde>Tarde</a></li><li><a href=/es/posts/poesia/teclado/ title=Teclado>Teclado</a></li><li><a href=/es/posts/poesia/tentacion/ title=Tentaci√≥n>Tentaci√≥n</a></li><li><a href=/es/posts/poesia/valor/ title=Valor>Valor</a></li><li><a href=/es/posts/poesia/verde/ title=Verde>Verde</a></li><li><a href=/es/posts/poesia/volver/ title=Volver>Volver</a></li><li><a href=/es/posts/poesia/basta/ title="Ya Basta">Ya Basta</a></li><li><a href=/es/posts/poesia/egoismo/ title=Ego√≠smo>Ego√≠smo</a></li></ul></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/seir-modified-model/hero.jpg)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/avataaars_hu2c0487d45d6329fb914c802e2a63f853_31315_120x120_fit_box_3.png alt="Author Image"><h5 class=author-name>Pablo Marcos</h5><p>mi√©rcoles, 3 de noviembre de 2021</p></div><div class=title><h1>Analysis of the efects of Lockdowns on CoViD</h1></div><div class=taxonomy-terms><ul style=padding-left:0></ul></div><div class=post-content id=post-content><div style=text-align:center><a href="./CoViD and Lockdowns.ipynb" target=_parent><img src=/posts/Imagenes/jupyter-badge.svg align=center width=20%></a></div></br><div class="cell markdown"><p>During the COVID-19 outbreak in spring 2020, preventive confinement at
day D for susceptible individuals was implemented in several countries
to ‚Äúbend‚Äù the infection curve and control the spread of the virus. Let
us assume that confined individuals are effectively protected from
infection. A ‚Äúbending‚Äù or flattening of the (log-scale) curve
representing the number of active cases was widely publicized as a sign
that the epidemics was under control.</p><ul><li>Given a set of parameters compatible with the current knowledge
about COVID-19, estimate the fraction of the susceptible population
that must be put into preventive confinement to stop the spread of
the virus. How does the delay D affect the effectiveness of the
confinement?</li><li>Explore parameter values that do and do not lead to virus remission.
Is the bending of the active cases curve a good predictor of whether
the infection will keep growing or remit?</li><li>Study the date at which the peak of the infection is reached. How
sensitive is it with respect to small changes in the parameters?
Given the uncertainty of COVID-19 epidemiological parameters, what
do you think about media predictions of the peak and end dates of
the infection?</li></ul></div><div class="cell markdown"><hr></div><div class="cell markdown"><p>To adequately model the CoViD-19 pandemic, I will use a compartimental
model approach, more concretely a SEIR model based on <a href=https://oa.upm.es/67410/>a publication by
Al√≥ndiga et al</a> which was actually created
by&mldr; hum! me!. In this work, we define a modified SEIR model with the
following compartiments:</p></div><div class="cell markdown"><p>$$\frac{dS}{dt} = -\beta I \frac{S}{N} $$<br>$$\frac{dE}{dt} = \beta I \frac{S}{N} - \delta E$$<br>$$\frac{dI}{dt} = \delta E - (1-a) \gamma I - \eta a I$$<br>$$\frac{dH}{dt} = \eta (1-a) I-\tau (1-u) H - u \sigma H$$<br>$$\frac{dC}{dt} = u \sigma H - m \rho min(UCI, C) - max(0, C-UCI) - \omega (1-m) min(UCI,C)$$<br>$$\frac{dR}{dt} = \gamma (1-a) I + \tau (1-u) H + \omega (1-m) min(UCI,C)$$<br>$$\frac{dM}{dt} = m \rho min(UCI, C) + max(0, C-UCI)$$</p></div><div class="cell markdown"><p>This compartiments can be described as:</p></div><div class="cell markdown"><ul><li>S: susceptible to contracting the disease</li><li>E: exposed to infected individuals, incubate the disease but are not
yet contagious.</li><li>I: infected, spread the disease</li><li>R: have overcome the disease, do not spread the virus and have
developed immunity to it.</li><li>C: infected individuals requiring intensive care.</li><li>M: deceased by COVID-19</li></ul></div><div class="cell markdown"><p>(For equations in LateX and Jupyter <a href=https://stackoverflow.com/questions/29505406/ipython-notebook-wont-convert-to-pdf-when-using-align>see
here</a>)</p></div><div class="cell markdown"><p>The meaning of all the parameters at large can be checked in the
extended article, <a href=https://www.pablomarcos.me/files/%20Al%c3%b3ndiga_Marcos_Pat%c3%b3n_Estudio_de_la_Pandemia_por_CoViD-19_mediante_modelos_compartimentales.pdf>availaible
here</a>,
but, to sum up the most interesting aspects of the model:</p></div><div class="cell code" data-execution_count=1><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> sympy <span style=color:#66d9ef>as</span> sp <span style=color:#75715e>#as allways, we need the imports first :D</span>
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> numpy <span style=color:#66d9ef>as</span> np
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> scipy.integrate <span style=color:#66d9ef>as</span> scint
</span></span><span style=display:flex><span><span style=color:#f92672>import</span> matplotlib.pyplot <span style=color:#66d9ef>as</span> plt
</span></span></code></pre></div></div><div class="cell markdown"><ul><li>The $R_{0} $ decreases progressively following a logistic curve
starting with the date of aplication of the measures, A. This is to
take into account natural human behaviour: people dont inmediately
cessate all activity after confinement, and they dont just start
taking preventive measures the day the confinement is decreted;
rather, fear and necessity make for a more progressive approach:</li></ul></div><div class="cell code" data-execution_count=2><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>R_0</span>(t, R_0i, R_0f, k, A):
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> (R_0i<span style=color:#f92672>-</span>R_0f) <span style=color:#f92672>/</span> (<span style=color:#ae81ff>1</span> <span style=color:#f92672>+</span> np<span style=color:#f92672>.</span>exp(<span style=color:#f92672>-</span>k<span style=color:#f92672>*</span>(<span style=color:#f92672>-</span>t<span style=color:#f92672>+</span>A))) <span style=color:#f92672>+</span> R_0f
</span></span></code></pre></div></div><div class="cell markdown"><ul><li>UCI internment is accounted for with the creation of an
<strong>Hospitalizados</strong> and a <strong>Criticos</strong> class, made up of the
individuals that require internment and of those that require
intensive care, and are thus more at risk of death. ICU occupation
rates are also accounted for, and it is estimated that, due to
political pressure, UCI capacity will also increase with time</li></ul></div><div class="cell code" data-execution_count=3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>modelo</span>(dias, N, UCI_100, R_0i, R_0f, k, A, u, m, a, s):
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>def</span> <span style=color:#a6e22e>b</span>(t):
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>return</span> R_0(t, R_0i, R_0f, k, A) <span style=color:#f92672>*</span> gamma
</span></span><span style=display:flex><span>    
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>def</span> <span style=color:#a6e22e>UCI</span>(t):
</span></span><span style=display:flex><span>        UCI_0 <span style=color:#f92672>=</span> UCI_100 <span style=color:#f92672>/</span> <span style=color:#ae81ff>100_000</span> <span style=color:#f92672>*</span> N  
</span></span><span style=display:flex><span>        <span style=color:#66d9ef>return</span> UCI_0 <span style=color:#f92672>+</span> s<span style=color:#f92672>*</span>UCI_0<span style=color:#f92672>*</span>t 
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    P0 <span style=color:#f92672>=</span> N<span style=color:#f92672>-</span><span style=color:#ae81ff>1.0</span>, <span style=color:#ae81ff>1.0</span>, <span style=color:#ae81ff>0.0</span>, <span style=color:#ae81ff>0.0</span>, <span style=color:#ae81ff>0.0</span>, <span style=color:#ae81ff>0.0</span>, <span style=color:#ae81ff>0.0</span>
</span></span><span style=display:flex><span>    tiempo <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>, dias<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>, dias, dtype<span style=color:#f92672>=</span>int)
</span></span><span style=display:flex><span>    Ps <span style=color:#f92672>=</span> scint<span style=color:#f92672>.</span>odeint(F, P0, tiempo, args<span style=color:#f92672>=</span>(N, b, m, u, a, UCI))
</span></span><span style=display:flex><span>    S <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>0</span>]
</span></span><span style=display:flex><span>    E <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>    I <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>2</span>]
</span></span><span style=display:flex><span>    H <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>3</span>]
</span></span><span style=display:flex><span>    C <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>4</span>]
</span></span><span style=display:flex><span>    R <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>5</span>]
</span></span><span style=display:flex><span>    M <span style=color:#f92672>=</span> Ps[:,<span style=color:#ae81ff>6</span>]
</span></span><span style=display:flex><span>    R_0t <span style=color:#f92672>=</span> R_0(tiempo, R_0i, R_0f, k, A)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> S, E, I, H, C, R, M, R_0t
</span></span></code></pre></div></div><div class="cell markdown"><p>Finally, we can tell python the equations we will use; the parameters,
such as delta and rho, are extracted from relevant previous literature:</p></div><div class="cell code" data-execution_count=39><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>3</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>9.0</span>; sigma <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>4.5</span>
</span></span><span style=display:flex><span>rho <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>14</span>; omega <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>6.5</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>9.0</span>
</span></span><span style=display:flex><span>tau <span style=color:#f92672>=</span> <span style=color:#ae81ff>1</span><span style=color:#f92672>/</span><span style=color:#ae81ff>14</span>; UCI_100 <span style=color:#f92672>=</span> <span style=color:#ae81ff>9.7</span> <span style=color:#75715e>#UCI_100 = the pre-existing UCI capacity per 100.000 habitants</span>
</span></span></code></pre></div></div><div class="cell code" data-execution_count=5><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>F</span>(P, t, N, b, m, u, a, UCI):
</span></span><span style=display:flex><span>    S, E, I, H, C, R, M <span style=color:#f92672>=</span> P
</span></span><span style=display:flex><span>    dS <span style=color:#f92672>=</span> <span style=color:#f92672>-</span>b(t) <span style=color:#f92672>*</span> S <span style=color:#f92672>*</span> I <span style=color:#f92672>/</span> N
</span></span><span style=display:flex><span>    dE <span style=color:#f92672>=</span> b(t) <span style=color:#f92672>*</span> S <span style=color:#f92672>*</span> I <span style=color:#f92672>/</span> N <span style=color:#f92672>-</span> delta <span style=color:#f92672>*</span> E
</span></span><span style=display:flex><span>    dI <span style=color:#f92672>=</span> delta <span style=color:#f92672>*</span> E <span style=color:#f92672>-</span> I<span style=color:#f92672>*</span>eta<span style=color:#f92672>*</span>a <span style=color:#f92672>-</span> I<span style=color:#f92672>*</span>gamma<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>a)
</span></span><span style=display:flex><span>    dH <span style=color:#f92672>=</span> eta<span style=color:#f92672>*</span>a<span style=color:#f92672>*</span>I<span style=color:#f92672>-</span>tau<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>u)<span style=color:#f92672>*</span>H<span style=color:#f92672>-</span>u<span style=color:#f92672>*</span>sigma<span style=color:#f92672>*</span>H
</span></span><span style=display:flex><span>    dC <span style=color:#f92672>=</span> u<span style=color:#f92672>*</span>sigma<span style=color:#f92672>*</span>H <span style=color:#f92672>-</span> m<span style=color:#f92672>*</span>rho<span style=color:#f92672>*</span>min(UCI(t),C) <span style=color:#f92672>-</span> max(<span style=color:#ae81ff>0</span>,C<span style=color:#f92672>-</span>UCI(t))<span style=color:#f92672>-</span>omega<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>m)<span style=color:#f92672>*</span>min(UCI(t),C)
</span></span><span style=display:flex><span>    dR <span style=color:#f92672>=</span> I<span style=color:#f92672>*</span>gamma<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>a) <span style=color:#f92672>+</span> tau<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>u)<span style=color:#f92672>*</span>H <span style=color:#f92672>+</span> omega<span style=color:#f92672>*</span>(<span style=color:#ae81ff>1</span><span style=color:#f92672>-</span>m)<span style=color:#f92672>*</span>min(UCI(t),C)
</span></span><span style=display:flex><span>    dM <span style=color:#f92672>=</span> m<span style=color:#f92672>*</span>rho<span style=color:#f92672>*</span>min(UCI(t),C) <span style=color:#f92672>+</span> max(<span style=color:#ae81ff>0</span>,C<span style=color:#f92672>-</span>UCI(t))
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> dS, dE, dI, dH, dC, dR, dM
</span></span></code></pre></div></div><div class="cell markdown"><p>In the paper, the authors fit the model to actual, real, pre-existing
data, and come up with the following values, with 2 exceptions:
population is the Spainish population in the 2020 census, according to
Wikipedia, and &ldquo;dias&rdquo; is the number of days we are studying; in this
case, 184</p></div><div class="cell code" data-execution_count=6><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>R_0i <span style=color:#f92672>=</span> <span style=color:#ae81ff>4.244</span>; R_0f<span style=color:#f92672>=</span> <span style=color:#ae81ff>0.848</span>; u <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.150</span>
</span></span><span style=display:flex><span>k <span style=color:#f92672>=</span> <span style=color:#ae81ff>5.392</span>; m <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.199</span>; a <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.125</span>
</span></span><span style=display:flex><span>s <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.012</span>; population <span style=color:#f92672>=</span> <span style=color:#ae81ff>47450795</span>; dias <span style=color:#f92672>=</span> <span style=color:#ae81ff>184</span>
</span></span></code></pre></div></div><div class="cell markdown"><p>And that&rsquo;s it! With this, we can define the plot function that we will
use to analyse the behaviour of covid (and of our model!)</p></div><div class="cell code" data-execution_count=31><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#66d9ef>def</span> <span style=color:#a6e22e>plotseircm</span>(A):
</span></span><span style=display:flex><span>    mod <span style=color:#f92672>=</span> modelo(dias, population, UCI_100, R_0i, R_0f, k, A, u, m, a, s)
</span></span><span style=display:flex><span>    tiempo <span style=color:#f92672>=</span> np<span style=color:#f92672>.</span>linspace(<span style=color:#ae81ff>0</span>,dias)
</span></span><span style=display:flex><span>    susceptibles <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>0</span>]
</span></span><span style=display:flex><span>    exposeados <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>1</span>]
</span></span><span style=display:flex><span>    infectados <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>2</span>]
</span></span><span style=display:flex><span>    hospitalizados <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>3</span>]
</span></span><span style=display:flex><span>    criticos <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>4</span>]
</span></span><span style=display:flex><span>    recuperados <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>5</span>]
</span></span><span style=display:flex><span>    muertos<span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>6</span>]
</span></span><span style=display:flex><span>    R_0t <span style=color:#f92672>=</span> mod[<span style=color:#ae81ff>7</span>]
</span></span><span style=display:flex><span>    muertes_diarias<span style=color:#f92672>=</span>[<span style=color:#ae81ff>0</span>]
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>for</span> i <span style=color:#f92672>in</span> range(<span style=color:#ae81ff>0</span>, len(muertos)<span style=color:#f92672>-</span><span style=color:#ae81ff>1</span>):
</span></span><span style=display:flex><span>        muertes_diarias<span style=color:#f92672>.</span>append(muertos[i<span style=color:#f92672>+</span><span style=color:#ae81ff>1</span>]<span style=color:#f92672>-</span>muertos[i])
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(range(dias), exposeados, <span style=color:#e6db74>&#34;-&#34;</span>, label <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Exposed&#34;</span>, color <span style=color:#f92672>=</span><span style=color:#e6db74>&#39;#ff7f0e&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(range(dias), infectados, <span style=color:#e6db74>&#34;-&#34;</span>, label <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Infected&#34;</span>, color <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;#2ca02c&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(range(dias), hospitalizados, <span style=color:#e6db74>&#34;-&#34;</span>, label <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Hospitalized&#34;</span>, color <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;#e377c2&#39;</span> )
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(range(dias), criticos, label <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Critical&#34;</span>, color <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;#d62728&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>plot(range(dias), muertos, <span style=color:#e6db74>&#34;-&#34;</span>, label <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;Muertos&#34;</span>, color <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;black&#39;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xlabel(<span style=color:#e6db74>&#34;Time (days)&#34;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ylabel(<span style=color:#e6db74>&#34;Population&#34;</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>xticks(rotation<span style=color:#f92672>=</span><span style=color:#ae81ff>40</span>)
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>legend()
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>subplots_adjust(bottom<span style=color:#f92672>=</span><span style=color:#ae81ff>0.2</span>, top<span style=color:#f92672>=</span><span style=color:#ae81ff>1.3</span>) <span style=color:#75715e>#So that plots look cute</span>
</span></span><span style=display:flex><span>    plt<span style=color:#f92672>.</span>ticklabel_format(style<span style=color:#f92672>=</span><span style=color:#e6db74>&#39;plain&#39;</span>, axis <span style=color:#f92672>=</span> <span style=color:#e6db74>&#39;y&#39;</span>)
</span></span><span style=display:flex><span>    
</span></span><span style=display:flex><span>    <span style=color:#75715e>#Guardamos como pinta el panorama</span>
</span></span><span style=display:flex><span>    panorama<span style=color:#f92672>=</span>[muertos[<span style=color:#ae81ff>123</span>],susceptibles[<span style=color:#ae81ff>123</span>], exposeados[<span style=color:#ae81ff>123</span>],infectados[<span style=color:#ae81ff>123</span>],hospitalizados[<span style=color:#ae81ff>123</span>], criticos[<span style=color:#ae81ff>123</span>],recuperados[<span style=color:#ae81ff>123</span>]]
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>return</span> panorama
</span></span></code></pre></div></div><div class="cell markdown"><p>We define a dataframe to store the raw, numeric values we will later use
to compare, and we plot the graphs:</p></div><div class="cell code" data-execution_count=34><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>totals <span style=color:#f92672>=</span> []
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Ground Truth&#39;</span>)
</span></span><span style=display:flex><span>totals<span style=color:#f92672>.</span>append(plotseircm(<span style=color:#ae81ff>73</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Scenario 1: 1 Week Before&#39;</span>)
</span></span><span style=display:flex><span>totals<span style=color:#f92672>.</span>append(plotseircm(<span style=color:#ae81ff>68</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Scenario 2: 1 Week After&#39;</span>)
</span></span><span style=display:flex><span>totals<span style=color:#f92672>.</span>append(plotseircm(<span style=color:#ae81ff>80</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Scenario 3: No confinement&#39;</span>)
</span></span><span style=display:flex><span>totals<span style=color:#f92672>.</span>append(plotseircm(<span style=color:#ae81ff>100000</span>)) <span style=color:#75715e>#Out of scope, effectively no confinement</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./CoViD%20and%20Lockdowns_21_0.png></figure></div></div><div class="cell code" data-execution_count=13 data-scrolled=true><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span><span style=color:#f92672>import</span> pandas <span style=color:#66d9ef>as</span> pd
</span></span><span style=display:flex><span>df <span style=color:#f92672>=</span> pd<span style=color:#f92672>.</span>DataFrame(columns<span style=color:#f92672>=</span>[<span style=color:#e6db74>&#39;Dead&#39;</span>,<span style=color:#e6db74>&#39;Susceptible&#39;</span>,<span style=color:#e6db74>&#39;Exposed&#39;</span>,<span style=color:#e6db74>&#39;Infected&#39;</span>,<span style=color:#e6db74>&#39;Hospitalized&#39;</span>,
</span></span><span style=display:flex><span>                      <span style=color:#e6db74>&#39;Critical&#39;</span>,<span style=color:#e6db74>&#39;Recovered&#39;</span>], data<span style=color:#f92672>=</span>totals)
</span></span><span style=display:flex><span>pd<span style=color:#f92672>.</span>set_option(<span style=color:#e6db74>&#39;display.float_format&#39;</span>, <span style=color:#66d9ef>lambda</span> x: <span style=color:#e6db74>&#39;%.f&#39;</span> <span style=color:#f92672>%</span> x)
</span></span><span style=display:flex><span>df
</span></span></code></pre></div><div class="output execute_result" data-execution_count=13><pre tabindex=0><code>     Dead  Susceptible  Exposed  Infected  Hospitalized  Critical  Recovered
0   25085     44592488    75403    268567         47499     11390    2430362
1    3747     46234518    30963    107323         18580      4946    1050718
2  201714     38982780   203888    804348        154462     15043    7088558
3 1262664      6799399   149374   2571360        895854     41298   35730846
</code></pre></div></div><div class="cell markdown"><p>As we can see, there is quite a big difference between the 4 scenarios,
so, yes, ¬°confinement is useful! Also in the table above, we can see
that, the bigger the delay &lsquo;D&rsquo;, the more ineffective the confinement is,
as the epidemic has had more time to progress; had the Spanish
government decreed confinement a week earlier, more than 20000 lives
could have been spared.</p></div><div class="cell markdown"><p>Now, given the known parameters for the model (provided in cell [4]),
we can calculate the fraction of the susceptible population that must be
put into preventive confinement to stop the spread of the virus.
<a href=https://www.pnas.org/content/pnas/118/21/e2107692118.full.pdf>According to
PNAS</a>,
mathematically, &ldquo;the herd immunity threshold can be very simply
calculated as $\omega = 1 - \frac{1}{R_{0}} $&rdquo;. If we take the model&rsquo;s
predicted $R_{0}i $ (that is, the one that corresponds to free virus
circulation) of 4.244, we get an $\omega $ of $1 - \frac{1}{4.244} $ =
0.7643, that is, 76.43 percent of the population that needs to be
virus-proof for the epidemic to cessate. If we discount the percentage
of infected population, I, from that percentage, we get the fraction of
individuals that must be confined to stop the disease from spreading.</p><p>Given that the Infected population is constantly changing, as is
$R_{0} $ according to our model, this proportion might be better
defined by creating a function.</p></div><div class="cell markdown"><p>Now, lets explore some different parameter values and see wether they
lead to virus remision or not. I will play with $\delta $, $\gamma $
and $\eta $, the original p√†rameters in the SEIR model; $\beta $
cannot be modified since it is set to dynamically change with time (our
way of affecting $R_{0}(t) $).Thus:</p></div><div class="cell code" data-execution_count=58 data-scrolled=true><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.7</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.9</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.11</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.75</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./CoViD%20and%20Lockdowns_30_0.png></figure></div></div><div class="cell markdown"><p>As was predictable, each one of the parameters does what it was assigned
to do: $\delta $ is the rate at which individuals incubating the virus
develop symptoms and become infectious, so increasing it leads to a huge
spike in the Infected department (although,in reality, it may lead to a
decrease: the big problem with CoViD are asymptomatic patients).
$\gamma $ is the inverse of the period of recuperation: if we increase
it, the Exposed department increases, as, when individuals take more to
recover, they can infect more people; finally, $\eta $, the inverse of
the period of symptom development: if it increases, the number of
infected patients decrease, while the hospitalizations increase, as do,
in turn, deaths. Exposed, however, remains constant: it seems like the
infection has become more lethal.</p></div><div class="cell markdown"><p>With regards to the bending of the curve, common sense, as well as the
analysis on Exercise 2.1, seem to indicate that, while the bending is
not a predictor of wether the infection will <strong>end</strong> or not, it is, in
fact, a good predictor of its evolution (it is, after all, a growth
curve), so it does help decide wether the infection will keep growing or
remit for the time being, helping the system cope.</p></div><div class="cell markdown"><p>Finally, we can study the peak of the infection and its sensibility to
small changes in the parametes:</p></div><div class="cell code" data-execution_count=59><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-python data-lang=python><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplots(figsize<span style=color:#f92672>=</span>(<span style=color:#ae81ff>15</span>, <span style=color:#ae81ff>10</span>))
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>1</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3333</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.1111</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.1111</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.3333</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.15</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.15</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>3</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.4</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.1111</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.1111</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>subplot(<span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>2</span>, <span style=color:#ae81ff>4</span>)
</span></span><span style=display:flex><span>delta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.5</span>; gamma <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.2</span>; eta <span style=color:#f92672>=</span> <span style=color:#ae81ff>0.2</span>
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>title(<span style=color:#e6db74>&#39;Evolution when $\delta$ = </span><span style=color:#e6db74>{0}</span><span style=color:#e6db74>, $\gamma$ = </span><span style=color:#e6db74>{1}</span><span style=color:#e6db74> and $\eta$ = </span><span style=color:#e6db74>{2}</span><span style=color:#e6db74>&#39;</span><span style=color:#f92672>.</span>format(str(delta), str(gamma), str(eta)))
</span></span><span style=display:flex><span>plotseircm(<span style=color:#ae81ff>1000</span>);
</span></span><span style=display:flex><span>plt<span style=color:#f92672>.</span>show()
</span></span></code></pre></div><div class="output display_data"><figure><img src=./CoViD%20and%20Lockdowns_26_0.png></figure></div></div><div class="cell markdown"><p>As can be seen, the impact of really, really small changes in the
parameter&rsquo;s value is neglegible, but, when we do a somewhat bigger
modification9, the thing changes: the prediction is off by more than 20
days. Given that accurate models can become more and more complex, whith
an increased number of parameters and equations, small changes in said
parameters can widely change the &ldquo;peak date&rdquo; prediction. In general, it
can be said that it is difficult to predict things in the field of
mathematical biology, and that models in general do have some degree of
uncertainty; however, even though media predictions were in fact bad,
they were the best we had, and thus modelling remains an important and
useful field to manage pandemics.</p></div><p><a href=https://apiwp.thelocal.com/wp-content/uploads/2021/01/1609846538_coronavirus-data-explorer-30.png>Post header</a> by Our World in Data via <a href=https://www.thelocal.es/20210105/spains-health-ministry-rules-out-a-new-lockdown-for-now/>thelocal.es</a> - <a href=https://creativecommons.org/licenses/by-sa/4.0/deed.en>CC BY-SA 4.0</a>.</p></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"><strong>Compartir en:</strong>
<a class="btn btn-sm facebook-btn" href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f" target=_blank><i class="fab fa-facebook"></i></a>
<a class="btn btn-sm twitter-btn" href="https://twitter.com/share?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f&text=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD&via=Pablo%20Marcos%27%20Blog" target=_blank><i class="fab fa-twitter"></i></a>
<a class="btn btn-sm reddit-btn" href="https://reddit.com/submit?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f&title=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD" target=_blank><i class="fab fa-reddit"></i></a>
<a class="btn btn-sm tumblr-btn" href="https://www.tumblr.com/share/link?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f&name=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD" target=_blank><i class="fab fa-tumblr"></i></a>
<a class="btn btn-sm pocket-btn" href="https://getpocket.com/save?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f&title=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD" target=_blank><i class="fab fa-get-pocket"></i></a>
<a class="btn btn-sm linkedin-btn" href="https://www.linkedin.com/shareArticle?url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f&title=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD" target=_blank><i class="fab fa-linkedin"></i></a>
<a class="btn btn-sm diaspora-btn" href="https://share.diasporafoundation.org/?title=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD&url=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f" rel=nofollow target=_blank><i class="fab fa-diaspora"></i></a>
<a class="btn btn-sm mastodon-btn" href="https://mastodon.social/share?text=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD - https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f" target=_blank><i class="fab fa-mastodon"></i></a>
<a class="btn btn-sm whatsapp-btn" href="https://api.whatsapp.com/send?text=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f" target=_blank><i class="fab fa-whatsapp"></i></a>
<a class="btn btn-sm email-btn" href="mailto:?subject=Analysis%20of%20the%20efects%20of%20Lockdowns%20on%20CoViD&body=https%3a%2f%2fwww.pablomarcos.me%2fes%2fposts%2fmaster-en-biolog%25C3%25ADa-computacional%2fmodelizaci%25C3%25B3n-de-sistemas-biol%25C3%25B3gicos%2fseir-modified-model%2f" target=_blank><i class="fas fa-envelope-open-text"></i></a></div><div class="col-md-6 btn-improve-page"><a href=https://codeberg.org/FlyingFlamingo/pablomarcos.me/_edit/main/content/posts/Master%20en%20Biolog%c3%ada%20Computacional/Modelizaci%c3%b3n%20de%20Sistemas%20Biol%c3%b3gicos/SEIR%20Modified%20Model/index.es.md title="Mejorar esta p√°gina" target=_blank rel=noopener><i class="fas fa-code-branch"></i>
Mejorar esta p√°gina</a></div></div><hr><div class="row next-prev-navigator"><div class="col-md-6 previous-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/celular-automata/ title="The Cellular Automata" class="btn btn-outline-info"><div><i class="fas fa-chevron-circle-left"></i> Anterior</div><div class=next-prev-text>The Cellular Automata</div></a></div><div class="col-md-6 next-article"><a href=/es/posts/master-en-biolog%C3%ADa-computacional/modelizaci%C3%B3n-de-sistemas-biol%C3%B3gicos/networks/ title="The Erd√∂s-R√©nyi model" class="btn btn-outline-info"><div>Siguiente <i class="fas fa-chevron-circle-right"></i></div><div class=next-prev-text>The Erd√∂s-R√©nyi model</div></a></div></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Contenido</h5><hr><div class=toc><nav id=TableOfContents></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navegaci√≥n</h5><ul><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#about>Sobre m√≠</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#experiences>Experiencia</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://www.pablomarcos.me/es/#projects>Proyectos</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contacto</h5><ul><li><a href=mailto:web@pablomarcos.me target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>web@pablomarcos.me</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Subscribirse</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Ingrese email"><br><input type=hidden name=btr_url value=https://www.pablomarcos.me/es/index.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">Al ingresar su direcci√≥n de correo electr√≥nico, acepta recibir el bolet√≠n de este sitio web.</small>
<button type=submit class="btn btn-info"> Enviar</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hugo-toha/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_3.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">¬© 2020 Pablo Ignacio Marcos L√≥pez</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Funcionando con
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script>
<script type=text/javascript src=/js/popper.min.js></script>
<script type=text/javascript src=/js/bootstrap.min.js></script>
<script type=text/javascript src=/js/navbar.js></script>
<script type=text/javascript src=/js/plyr.js></script>
<script type=text/javascript src=/js/main.js></script>
<script type=text/javascript src=/js/darkreader.js></script>
<script type=text/javascript src=/js/darkmode-darkreader.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script>
<script src=/js/single.js></script>
<script>hljs.initHighlightingOnLoad()</script><link rel=stylesheet href=/katex/katex.min.css><script type=text/javascript defer src=/katex/katex.min.js></script>
<script type=text/javascript defer src=/katex/auto-render.min.js onload=renderMathInElement(document.body)>renderMathInElement(document.body,{delimiters:[{left:"$$",right:"$$",display:!0},{left:"\\[",right:"\\]",display:!0},{left:"$",right:"$",display:!1},{left:"\\(",right:"\\)",display:!1}]})</script><script src=https://storage.ko-fi.com/cdn/scripts/overlay-widget.js></script>
<script>kofiWidgetOverlay.draw("pablomarcos",{type:"floating-chat","floating-chat.donateButton.text":"Ap√≥yame","floating-chat.donateButton.text-color":"#fff","floating-chat.donateButton.background-color":"#00b9fe"})</script></body></html>